<!DOCTYPE html>
<html lang="" dir="ltr">

<head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <title>Food web reconstruction through phylogenetic transfer of low-rank network representation</title>
    <script src="https://hypothes.is/embed.js" async></script>
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/normalize/8.0.1/normalize.min.css"
    integrity="sha256-l85OmPOjvil/SOvVt3HnSSjzF1TUMyT9eV0c2BzEGzU=" crossorigin="anonymous" />
    <style>
        @import url('https://fonts.googleapis.com/css2?family=Roboto:ital,wght@0,300;0,400;0,500;1,300;1,400;1,500&display=swap');
        @import url('https://fonts.googleapis.com/css2?family=Roboto+Condensed:ital,wght@0,300;0,400;0,700;1,300;1,400;1,700&display=swap');
        @import url('https://fonts.googleapis.com/css2?family=Roboto+Mono:ital,wght@0,400;0,500;1,400;1,500&display=swap');

        :root {
            --citation: #1976D2;
            --structure: hsl(210, 61%, 43%);
            
            --light: #474747;
            --pale: #a5a5a5;
            --grey: #e0e0e0;
            --nav: #fdfdfd;

            --width: 820px;
            --inner-space: 1.2em;
        }

        body {
            font-family: "Roboto", sans-serif;
            margin: 0 0;
            padding: 0 0;
        }

        article > p {
            line-height: 135%;
        }
        article > h1 {
            padding-top: 3em;
        }
        a {
            text-decoration: none;
            color: #000;
            border-bottom: 1px solid var(--structure);
        }
        a:hover {
            font-weight: 500;
        }
        .citation {
            color: var(--citation);
        }
        .container {
            margin: 0em auto;
            max-width: var(--width);
            padding: var(--inner-space);
        }
        header > .container {
            margin-top: 4em;
        }
        code, pre {
            font-family: "Roboto Mono", monospace;
        }
        pre {
            white-space: pre-wrap;
        }
        h1, h2, h3 {
            color: var(--structure);
            font-family: "Roboto Condensed", sans-serif;
            font-weight: 400;
        }
        header {
            font-family: "Roboto Condensed", sans-serif;
            z-index: 5;
            color: black;
            margin-top: 4rem;
        }
        .header--title {
            font-weight: 400;
            font-size: 2rem;
            margin-bottom: 1rem;
        }
        .header--authors > span {
            color: var(--light);
            font-weight: 400;
        }
        main {
            padding-top: 8em !important;
            padding-bottom: 8em !important;
        }
        footer {
            background-color: var(--grey);
            font-weight: 300;
        }
        .abstract {
            color: var(--light);
            margin-bottom: 5em;
        }
        figure {
            max-width: 100%;
            border: 1px solid var(--pale);
            border-radius: 2px;
            margin: 1em 0.5em;
        }
        figure > img {
            max-width: 100%;
            display: block;
            margin: 0em auto;
        }
        figure > figcaption {
            padding: var(--inner-space);
        }
        table {
            width: 100%;
            min-width: 100%;
            border-collapse: collapse;
        }
        table > caption {
            padding: var(--inner-space);
        }
        figcaption, caption {
            color: var(--light);
            font-weight: 300;
        }
        thead {
            color: var(--structure);
        }
        td, th {
            padding: 3px;
        }
        tr {
            border-top: 1px solid var(--structure);
            border-bottom: 1px solid var(--structure);
        }
        tbody > tr:nth-child(even) {background: var(--grey)};
        tbody {
            border-bottom: 1px solid var(--structure);
        }
        #refs {
            font-size: 90%;
            column-count: 2;
        }
        #refs > .csl-entry {
            padding-bottom: 0.2em;
            padding-top: 0.2em;
        }
        .hanging-indent > .csl-entry {
            padding-left: 15px ;
            text-indent: -15px ;
        }    </style>
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.10.0-rc.1/dist/katex.min.css"
    integrity="sha384-D+9gmBxUQogRLqvARvNLmA9hS2x//eK1FhVb9PiU86gmcrBrJAQT8okdJ4LMp2uv" crossorigin="anonymous" />
    <script defer src="https://cdn.jsdelivr.net/npm/katex@0.10.0-rc.1/dist/katex.min.js"
    integrity="sha384-483A6DwYfKeDa0Q52fJmxFXkcPCFfnXMoXblOkJ4JcA8zATN6Tm78UNL72AKk+0O"
    crossorigin="anonymous"></script>
    <script defer src="https://cdn.jsdelivr.net/npm/katex@0.10.0-rc.1/dist/contrib/auto-render.min.js"
    integrity="sha384-yACMu8JWxKzSp/C1YV86pzGiQ/l1YUfE8oPuahJQxzehAjEt2GiQuy/BIvl9KyeF"
    crossorigin="anonymous"></script>
    <script>
        document.addEventListener("DOMContentLoaded", function () {
            renderMathInElement(document.body, {
                delimiters: [
                { left: "$$", right: "$$", display: true },
                { left: "$", right: "$", display: false }
                ]
            });
        });
    </script>
</head>

<body>
    <header>
        <div class="container">
            <div class="header--title">Food web reconstruction through phylogenetic transfer of low-rank network representation</div>
            <div class="header--authors">
                <span class="header--author">
                <a href="https://orcid.org/0000-0001-6067-1349" target="_blank">
                    Tanya Strydom
                </a>
                <sup class="affilcode">1</sup>
                <sup class="affilcode">2</sup>
                <sup class="affilcode">‡</sup>
                </span><span class="header--author">
                <a href="https://orcid.org/0000-0003-0193-5441" target="_blank">
                    Salomé Bouskila
                </a>
                <sup class="affilcode">1</sup>
                <sup class="affilcode">‡</sup>
                </span><span class="header--author">
                <a href="https://orcid.org/0000-0001-9051-0597" target="_blank">
                    Francis Banville
                </a>
                <sup class="affilcode">1</sup>
                <sup class="affilcode">3</sup>
                <sup class="affilcode">2</sup>
                </span><span class="header--author">
                <a href="https://orcid.org/0000-0003-4036-977X" target="_blank">
                    Ceres Barros
                </a>
                <sup class="affilcode">4</sup>
                </span><span class="header--author">
                <a href="https://orcid.org/0000-0002-2151-6693" target="_blank">
                    Dominique Caron
                </a>
                <sup class="affilcode">5</sup>
                <sup class="affilcode">2</sup>
                </span><span class="header--author">
                <a href="https://orcid.org/0000-0003-0452-6993" target="_blank">
                    Maxwell J Farrell
                </a>
                <sup class="affilcode">6</sup>
                </span><span class="header--author">
                <a href="https://orcid.org/0000-0002-9935-1366" target="_blank">
                    Marie-Josée Fortin
                </a>
                <sup class="affilcode">6</sup>
                </span><span class="header--author">
                <a href="https://orcid.org/0000-0003-3220-6161" target="_blank">
                    Victoria Hemming
                </a>
                <sup class="affilcode">4</sup>
                </span><span class="header--author">
                <a href="https://orcid.org/0000-0002-4104-9463" target="_blank">
                    Benjamin Mercier
                </a>
                <sup class="affilcode">3</sup>
                <sup class="affilcode">2</sup>
                </span><span class="header--author">
                <a href="https://orcid.org/0000-0002-6004-4027" target="_blank">
                    Laura J. Pollock
                </a>
                <sup class="affilcode">5</sup>
                <sup class="affilcode">2</sup>
                </span><span class="header--author">
                <a href="https://orcid.org/0000-0001-5785-8321" target="_blank">
                    Rogini Runghen
                </a>
                <sup class="affilcode">7</sup>
                </span><span class="header--author">
                <a href="https://orcid.org/0000-0002-3454-0633" target="_blank">
                    Giulio V. Dalla Riva
                </a>
                <sup class="affilcode">8</sup>
                </span><span class="header--author">
                <a href="https://orcid.org/0000-0002-0735-5184" target="_blank">
                    Timothée Poisot
                </a>
                <sup class="affilcode">1</sup>
                <sup class="affilcode">2</sup>
                </span>
            </div>
        </div>
    </header>
    
    <main class="container">
        <section class="abstract">
            Despite their importance in many ecological processes, collecting data and information on ecological interactions is an exceedingly challenging task. For this reason, large parts of the world have a data deficit when it comes to species interactions, and how the resulting networks are structured. As data collection alone is unlikely to be sufficient, community ecologists must adopt predictive methods. Here we develop such a method, relying on graph embedding and transfer learning to assemble a predicted list of trophic interactions between Canadian mammals. This interaction list is derived from the European food web, despite sharing only 5% of common species with Canada. The results of the predictive model are compared against databases of recorded pairwise interactions, showing that we correctly recover over 95% of known interactions. We provide guidance on how this method can be adapted by substituting some approaches or predictors in order to make it more generally applicable.


            <div class="artifacts">
                <a href="draft.pdf" title="Draft PDF" target="_blank">draft</a>
                <a href="preprint.pdf" title="Preprint PDF" target="_blank">preprint</a>
            </div>
        </section>
        
        <article>
            <h1 id="introduction">Introduction</h1>
            <p>There are two core challenges we are faced with in furthering our understanding of ecological networks across space, particularly at macro-ecologically relevant scales <span class="citation" data-cites="Trojelsgaard2016EcoNet">(<em>e.g.</em> Trøjelsgaard &amp; Olesen 2016)</span>. First, networks within a location are difficult to sample properly <span class="citation" data-cites="Jordano2016ChaEco Jordano2016SamNet">(Jordano 2016a, b)</span>, resulting in a widespread “Eltonian shortfall” <span class="citation" data-cites="Hortal2015SevSho">(Hortal <em>et al.</em> 2015)</span>, <em>i.e.</em> a lack of knowledge about inter and intra specific relationships. This first challenge has been, in large part, addressed by the recent emergence of a suite of methods aiming to predict interactions within <em>existing</em> networks, many of which are reviewed in <span class="citation" data-cites="Strydom2021RoaPre">Strydom <em>et al.</em> (2021a)</span>. Second, recent analyses based on collected data <span class="citation" data-cites="Poisot2021GloKno">(Poisot <em>et al.</em> 2021a)</span> or metadata <span class="citation" data-cites="Cameron2019UneGlo">(Cameron <em>et al.</em> 2019)</span> highlight that ecological networks are currently studied in a biased subset of space and bioclimates, which impedes our ability to generalize any local understanding of network structure. Meaning that, although the framework to address incompleteness <em>within</em> networks exists, there would still be regions for which, due to a <em>lack</em> of local interaction data, we are unable to infer potential species interactions. Having a general solution for inferring a <em>plausible</em> metaweb (despite the unavailability of interaction data) could be the catalyst for significant breakthroughs in our ability to start thinking about species interaction networks over large spatial scales.</p>
            <p>Here, we present a general method for the transfer learning of network representations, relying on the similarities of species in a biologically/ecologically relevant proxy space (<em>e.g.</em> shared morphology or ancestry). Transfer learning is a machine learning methodology that uses the knowledge gained from solving one problem and applying it to a related (destination) problem <span class="citation" data-cites="Torrey2010TraLea Pan2010SurTra">(Pan &amp; Yang 2010; Torrey &amp; Shavlik 2010)</span>. In this instance, we solve the problem of predicting trophic interactions between species, based on knowledge extracted from another species pool for which interactions are known by using phylogenetic structure as a medium for transfer. This allows us to construct a <em>probabilistic</em> metaweb for a community for which we have <em>no</em> prior trophic interaction data for the desired species pool. Our methodology is outlined in fig. 1, where we provide an illustration based on learning the embedding of a metaweb of trophic interactions for European mammals <span class="citation" data-cites="Maiorano2020TetEu Maiorano2020DatTet">(known interactions; Maiorano <em>et al.</em> 2020b, a)</span> and, based on phylogenetic relationships between mammals globally <span class="citation" data-cites="Upham2019InfMam">(<em>i.e.</em>, phylogenetic tree Upham <em>et al.</em> 2019)</span>, infer a metaweb for the Canadian mammalian species pool (interactions are treated as unknown in this instance).</p>
            <figure>
            <img src="figures/figure-concept.png" id="fig:concept" alt="Figure 1: Overview of the phylogenetic transfer learning (and prediction) of species interactions networks. Starting from an initial, known, network, we learn its representation through a graph embedding step (here, a truncated Singular Value Decomposition; Step 1), yielding a series of latent traits (vulnerability traits representing species at the lower trophic-level and generality traits representing species at higher trophic-levels; sensu Schoener (1989)); second, for the destination species pool, we perform ancestral character estimation using a phylogeny (here, using a Brownian model for the latent traits; Step 2); we then sample from the reconstructed distribution of latent traits (Step 3) to generate a probabilistic metaweb at the destination (here, assuming a uniform distribution of traits), and threshold it to yield the final list of interactions (Step 4)." /><figcaption aria-hidden="true">Figure 1: Overview of the phylogenetic transfer learning (and prediction) of species interactions networks. Starting from an initial, known, network, we learn its representation through a graph embedding step (here, a truncated Singular Value Decomposition; Step 1), yielding a series of latent traits (vulnerability traits representing species at the lower trophic-level and generality traits representing species at higher trophic-levels; <em>sensu</em> <span class="citation" data-cites="Schoener1989FooWeb">Schoener (1989)</span>); second, for the destination species pool, we perform ancestral character estimation using a phylogeny (here, using a Brownian model for the latent traits; Step 2); we then sample from the reconstructed distribution of latent traits (Step 3) to generate a probabilistic metaweb at the destination (here, assuming a uniform distribution of traits), and threshold it to yield the final list of interactions (Step 4).</figcaption>
            </figure>
            <p>There is a plurality of measures of species similarities that can be used for metaweb reconstruction <span class="citation" data-cites="Morales-Castilla2015InfBio">(see <em>e.g.</em> Morales-Castilla <em>et al.</em> 2015)</span>; however, phylogenetic proximity has several desirable properties when working at large scales. <span class="citation" data-cites="Gerhold2015PhyPat">Gerhold <em>et al.</em> (2015)</span> made the point that phylogenetic signal captures diversification of characters (large macro-evolutionary process), but not necessarily community assembly (fine ecological process); <span class="citation" data-cites="Dormann2010EvoCli">Dormann <em>et al.</em> (2010)</span> previously found very similar conclusions. Interactions tend reflect a phylogenetic signal because they have a conserved pattern of evolutionary convergence that encompasses a wide range of ecological and evolutionary mechanisms <span class="citation" data-cites="Mouquet2012EcoAdv Cavender-Bares2009MerCom">(Cavender-Bares <em>et al.</em> 2009; Mouquet <em>et al.</em> 2012)</span>, and - most importantly - retain this signal even when it is not detectable at the community scale <span class="citation" data-cites="Poisot2018IntRet Hutchinson2017CopSig">(Hutchinson <em>et al.</em> 2017; Poisot &amp; Stouffer 2018)</span>. Finally, species interactions at macro-ecological scales seem to respond mostly to macro-evolutionary processes <span class="citation" data-cites="Price2003MacThe">(Price 2003)</span>; which is evidenced by the presence of conserved backbones in food webs <span class="citation" data-cites="DallaRiva2016ExpEvo">(Dalla Riva &amp; Stouffer 2016)</span>, strong evolutionary signature on prey choice <span class="citation" data-cites="Stouffer2012EvoCon">(Stouffer <em>et al.</em> 2012)</span>, and strong phylogenetic signature in food web intervality <span class="citation" data-cites="Eklof2016PhyCom">(Eklöf &amp; Stouffer 2016)</span>. Phylogenetic reconstruction has also previously been used within the context of ecological networks, namely understanding ancestral plant-insect interactions <span class="citation" data-cites="Braga2021PhyRec">(Braga <em>et al.</em> 2021)</span>. Taken together, these considerations suggest that phylogenies can reliably be used to transfer knowledge on species interactions.</p>
            <p>Our case study shows that phylogenetic transfer learning is indeed an effective approach to predict the Canadian mammalian metaweb. This showcases that although the components (species) that make up the Canadian and European communities may be <em>minimally</em> shared, if the medium (proxy space) selected in the transfer step is biologically plausible, we can still effectively learn from the known network and make biologically relevant predictions of interactions. It should be reiterated that the framework presented in fig. 1 is amenable to changes; notably, the measure of similarity may not be phylogeny, and can be replaced by information on foraging <span class="citation" data-cites="Beckerman2006ForBio">(Beckerman <em>et al.</em> 2006)</span>, cell-level mechanisms <span class="citation" data-cites="Boeckaerts2021PreBac">(Boeckaerts <em>et al.</em> 2021)</span>, or a combination of traits and phylogenetic structure <span class="citation" data-cites="Stock2021PaiLea">(Stock 2021)</span>.</p>
            <h1 id="data-used-for-the-case-study">Data used for the case study</h1>
            <p>We use data from the European metaweb assembled by <span class="citation" data-cites="Maiorano2020TetEu">Maiorano <em>et al.</em> (2020b)</span>, following the definition of the metaweb first introduced by <span class="citation" data-cites="Dunne2006NetStr">Dunne (2006)</span>, <em>i.e.</em> an inventory of all possible interactions within species from a spatially delimited pool. Notably the metaweb is not a prediction of the food web at any specific locale within the frontiers of the species pool – in fact, these local food webs are expected to have a subset of both the species and the interactions of their metaweb <span class="citation" data-cites="Poisot2012DisSpe">(Poisot <em>et al.</em> 2012)</span>. This being said, as the metaweb represents the total of functional, phylogenetic, and macroecological processes <span class="citation" data-cites="Morales-Castilla2015InfBio">(Morales-Castilla <em>et al.</em> 2015)</span>, it is thus still worthy of ecological attention. We deduced the subgraph corresponding to all mammals by matching species names in the original network to the GBIF taxonomic backbone <span class="citation" data-cites="GBIFSecretariat2021GbiBac">(GBIF Secretariat 2021)</span> and retaining all those who matched to mammals. This serves a dual purpose 1) to extract only mammals from the European network and 2) to match and standardize species names when aggregating the different data sources further downstream (which is an important consideration when combining datasets <span class="citation" data-cites="Grenie2021HarTax">(Grenié <em>et al.</em> 2021)</span>). All nodes had valid matches to GBIF at this step, and so this backbone is used for all name reconciliation steps as outlined below.</p>
            <p>The European metaweb represents the knowledge we want to learn and transfer; the phylogenetic similarity of mammals here represents the information for transfer. We used the mammalian consensus supertree by <span class="citation" data-cites="Upham2019InfMam">Upham <em>et al.</em> (2019)</span>, for which all approximatively 6000 names have been similarly matched to their GBIF valid names. This step allows us to place each node of the mammalian European metaweb in the phylogeny.</p>
            <p>The destination problem to which we want to transfer knowledge is the trophic interactions between mammals in Canada. We obtained the list of extant species from the IUCN checklist, and selected the terrestrial and semi-aquatic species (this corresponds to the same selection that was applied by <span class="citation" data-cites="Maiorano2020TetEu">Maiorano <em>et al.</em> (2020b)</span> in the European metaweb). The IUCN names were, as previously, reconciled against GBIF to have an exact match to the taxonomy.</p>
            <p>After taxonomic cleaning and reconciliation as outlined in the following sections, the mammalian European metaweb has 260 species, and the Canadian species pool has 163; of these, 17 (about 4% of the total) are shared, and 89 species from Canada (54%) had at least one congeneric species in Europe. The similarity for both species pools predictably increases with higher taxonomic order, with 19% of shared genera, 47% of shared families, and 75% of shared orders; for the last point, Canada and Europe each had a single unique order (<em>Didelphimorphia</em> for Canada, <em>Erinaceomorpha</em> for Europe).</p>
            <p>In the following sections, we describe the representational learning step applied to European data, the transfer step through phylogenetic similarity, and the generation of a probabilistic metaweb for the destination species pool.</p>
            <h1 id="method-description">Method description</h1>
            <p>The crux of the method is the transfer of knowledge of a known network, in order to predict interactions between species from another location. In fig. 1, we give a high-level overview of the approach; in the example around which this manuscript is built (leveraging detailed knowledge about binary trophic interactions between Mammalia in Europe to predict the less known trophic interactions between closely phylogenetically related Mammalia in Canada), we use a series of specific steps for network embedding, trait inference, network prediction and thresholding.</p>
            <p>Specifically, our approach can be summarized as follows: from the known network in Europe, we use a truncated Singular Value Decomposition <span class="citation" data-cites="Halko2011FinStr">(t-SVD; Halko <em>et al.</em> 2011)</span> to generate latent traits representing a low-dimensional embedding of the network; these traits give an unbiased estimate of the node’s position in the latent feature spaces. Then, we map these latent traits onto a reference phylogeny (other distance-based measures of species proximity that allow for the inference of features in the latent space can be used, for example the dissimilarity in functional traits). Based on the reconstructed latent traits for species in the destination species pool, a Random Dot Product Graph model <span class="citation" data-cites="Young2007RanDot">(hereafter RDPG; Young &amp; Scheinerman 2007)</span> predicts the interaction between species through a function of the nodes’ features through matrix multiplication. Thus, from latent traits and node position, we can infer interactions.</p>
            <h2 id="implementation-and-code-availability">Implementation and code availability</h2>
            <p>The entire pipeline is implemented in <em>Julia</em> 1.6 <span class="citation" data-cites="Bezanson2017JulFre">(Bezanson <em>et al.</em> 2017)</span> and is available under the permissive MIT License at <a href="https://osf.io/2zwqm/"><code>https://osf.io/2zwqm/</code></a>. The taxonomic cleanup steps are done using <code>GBIF.jl</code> <span class="citation" data-cites="Dansereau2021SimJl">(Dansereau &amp; Poisot 2021)</span>. The network embedding and analysis is done using <code>EcologicalNetworks.jl</code> <span class="citation" data-cites="Banville2021ManJl Poisot2019EcoJl">(Poisot <em>et al.</em> 2019; Banville <em>et al.</em> 2021)</span>. The phylogenetic simulations are done using <code>PhyloNetworks.jl</code> <span class="citation" data-cites="Solis-Lemus2017PhyPac">(Solís-Lemus <em>et al.</em> 2017)</span> and <code>Phylo.jl</code> <span class="citation" data-cites="Reeve2016HowPar">(Reeve <em>et al.</em> 2016)</span>. A complete <code>Project.toml</code> file specifying the full tree of dependencies is available alongside the code. This material also includes a fully annotated copy of the entire code required to run this project (describing both the intent of the code and discussing some technical implementation details), a vignette for every step of the process, and a series of Jupyter notebooks with the text and code. The pipeline can be executed on a laptop in a matter of minutes, and therefore does not require extensive computational power.</p>
            <h2 id="step-1-learning-the-origin-network-representation">Step 1: Learning the origin network representation</h2>
            <p>The first step in transfer learning is to learn the structure of the original dataset. In order to do so, we rely on an approach inspired from representational learning, where we learn a <em>representation</em> of the metaweb (in the form of the latent subspaces), rather than a list of interactions (species <em>a</em> eats <em>b</em>). This approach is conceptually different from other metaweb-scale predictions <span class="citation" data-cites="Albouy2019MarFis">(<em>e.g.</em> Albouy <em>et al.</em> 2019)</span>, in that the metaweb representation is easily transferable. Specifically, we use RDPG to create a number of latent variables that can be combined into an approximation of the network adjacency matrix. RDPG results are known to have strong phylogenetic signal, and to capture the evolutionary backbone of food webs <span class="citation" data-cites="DallaRiva2016ExpEvo">(Dalla Riva &amp; Stouffer 2016)</span>. In addition, recent advances show that the latent variables produced this way can be used to predict <em>de novo</em> network edges <span class="citation" data-cites="Runghen2021ExpNod">(<em>i.e.</em> interactions; Runghen <em>et al.</em> 2021)</span>.</p>
            <p>The latent variables are created by performing a truncated Singular Value Decomposition (t-SVD) on the adjacency matrix. SVD is an appropriate embedding of ecological networks, which has recently been shown to both capture their complex, emerging properties <span class="citation" data-cites="Strydom2021SvdEnt">(Strydom <em>et al.</em> 2021b)</span> and to allow highly accurate prediction of the interactions within a single network <span class="citation" data-cites="Poisot2021ImpMam">(Poisot <em>et al.</em> 2021b)</span>. Under SVD, an adjacency matrix <span class="math inline"><strong>A</strong></span> (where <span class="math inline"><strong>A</strong><sub><em>m</em>, <em>n</em></sub> ∈ 𝔹</span> where 1 indicates predation and 0 an absence thereof) is decomposed into three components resulting in <span class="math inline"><strong>A</strong> = <strong>L</strong><strong>Σ</strong><strong>R</strong><strong>′</strong>.</span> Here, <span class="math inline"><strong>Σ</strong></span> is a <span class="math inline"><em>m</em> × <em>n</em></span> diagonal matrix and contains only singular (<span class="math inline"><em>σ</em></span>) values along its diagonal, <span class="math inline"><strong>L</strong></span> is a <span class="math inline"><em>m</em> × <em>m</em></span> unitary matrix, and <span class="math inline"><strong>R</strong>′</span> a <span class="math inline"><em>n</em> × <em>n</em></span> unitary matrix. Truncating the SVD removes additional noise in the dataset by omitting non-zero and/or smaller <span class="math inline"><em>σ</em></span> values from <span class="math inline"><strong>Σ</strong></span> using the rank of the matrix. Under a t-SVD <span class="math inline"><strong>A</strong><sub><em>m</em>, <em>n</em></sub></span> is decomposed so that <span class="math inline"><strong>Σ</strong></span> is a square <span class="math inline"><em>r</em> × <em>r</em></span> diagonal matrix (whith <span class="math inline">1 ≤ <em>r</em> ≤ <em>r</em><sub><em>f</em><em>u</em><em>l</em><em>l</em></sub></span> where <span class="math inline"><em>r</em><sub><em>f</em><em>u</em><em>l</em><em>l</em></sub></span> is the full rank of <span class="math inline"><strong>A</strong></span> and <span class="math inline"><em>r</em></span> the rank at which we truncate the matrix) containing only non-zero <span class="math inline"><em>σ</em></span> values. Additionally, <span class="math inline"><strong>L</strong></span> is now a <span class="math inline"><em>m</em> × <em>r</em></span> semi unitary matrix and <span class="math inline"><strong>R</strong>′</span> a <span class="math inline"><em>n</em> × <em>r</em></span> semi-unitary matrix.</p>
            <p>The specific rank at which the SVD ought to be truncated is a difficult question. The purpose of SVD is to remove the noise (expressed at high dimensions) and to focus on the signal, (expressed at low dimensions). In datasets with a clear signal/noise demarcation, a scree plot of <span class="math inline"><strong>Σ</strong></span> can show a sharp drop at the rank where noise starts <span class="citation" data-cites="Zhu2006AutDim">(Zhu &amp; Ghodsi 2006)</span>. Because the European metaweb is almost entirely known, the amount of noise (uncertainty) is low; this is reflected in fig. 2 (left), where the scree plot shows no important drop, and in fig. 2 (right) where the proportion of variance explained increases smoothly at higher dimensions. For this reason, we default back to a threshold that explains 60% of the variance in the underlying data, corresponding to 12 dimensions - <em>i.e.</em> a tradeoff between accuracy and a reduced number of features.</p>
            <p>A RDPG estimates the probability of observing interactions between nodes (species) as a function of the nodes’ latent variables. The latent variables used for the RDPG, called the left and right subspaces, are defined as <span class="math inline">$\mathscr{L} = \mathbf{L}\sqrt{\mathbf{\Sigma}}$</span>, and <span class="math inline">$\mathscr{R} = \sqrt{\mathbf{\Sigma}}\mathbf{R}$</span> – using the full rank of <span class="math inline"><strong>A</strong></span>, <span class="math inline">ℒℛ′ = <strong>A</strong></span>, and using any smaller rank results in <span class="math inline">ℒℛ′ ≈ <strong>A</strong></span>. Using a rank of 1 for the t-SVD provides a first-order approximation of the network.</p>
            <figure>
            <img src="figures/figure-screeplot.png" id="fig:scree" alt="Figure 2: Left: representation of the screeplot of the singular values from the t-SVD on the European metaweb. The screeplot shows no obvious drop in the singular values that may be leveraged to automatically detect a minimal dimension for embedding, after e.g. Zhu &amp; Ghodsi (2006). Right: cumulative fraction of variance explained by each dimension up to the rank of the European metaweb. The grey lines represent cutoffs at 50, 60… 90% of variance explained. For the rest of the analysis, we reverted to an arbitrary threshold of 60% of variance explained, which represented a good tradeoff between accuracy and reduced number of features." /><figcaption aria-hidden="true">Figure 2: Left: representation of the screeplot of the singular values from the t-SVD on the European metaweb. The screeplot shows no obvious drop in the singular values that may be leveraged to automatically detect a minimal dimension for embedding, after <em>e.g.</em> <span class="citation" data-cites="Zhu2006AutDim">Zhu &amp; Ghodsi (2006)</span>. Right: cumulative fraction of variance explained by each dimension up to the rank of the European metaweb. The grey lines represent cutoffs at 50, 60… 90% of variance explained. For the rest of the analysis, we reverted to an arbitrary threshold of 60% of variance explained, which represented a good tradeoff between accuracy and reduced number of features.</figcaption>
            </figure>
            <p>Because RDPG relies on matrix multiplication, the higher dimensions essentially serve to make specific interactions converge towards 0 or 1; therefore, for reasonably low ranks, there is no guarantee that the values in the reconstructed network will be within the unit range. In order to determine what constitutes an appropriate threshold for probability, we performed the RDPG approach on the European metaweb, and evaluated the probability threshold by treating this as a binary classification problem, specifically assuming that both 0 and 1 in the European metaweb are all true. Given the methodological details given in <span class="citation" data-cites="Maiorano2020TetEu">Maiorano <em>et al.</em> (2020b)</span> and <span class="citation" data-cites="OConnor2020UnvFoo">O’Connor <em>et al.</em> (2020)</span>, this seems like a reasonable assumption, although one that does not hold for all metawebs. We used the thresholding approach presented in <span class="citation" data-cites="Poisot2021ImpMam">Poisot <em>et al.</em> (2021b)</span>, and picked a cutoff that maximized Youden’s <span class="math inline"><em>J</em></span> statistic (a measure of the informedness (trust) of predictions; <span class="citation" data-cites="Youden1950IndRat">Youden (1950)</span>); the resulting cutoff was 0.22, and gave an accuracy above 0.99.</p>
            <p>The left and right subspaces for the European metaweb, accompanied by the threshold for prediction, represent the knowledge we seek to transfer. In the next section, we explain how we rely on phylogenetic similarity to do so.</p>
            <h2 id="steps-2-and-3-transfer-learning-through-phylogenetic-relatedness">Steps 2 and 3: Transfer learning through phylogenetic relatedness</h2>
            <p>In order to transfer the knowledge from the European metaweb to the Canadian species pool, we performed ancestral character estimation using a Brownian motion model, which is a conservative approach in the absence of strong hypotheses about the nature of phylogenetic signal in the network decomposition <span class="citation" data-cites="Litsios2012EffPhy">(Litsios &amp; Salamin 2012)</span>. This uses the estimated feature vectors for the European mammals to create a state reconstruction for all species (conceptually something akin to a trait-based mammalian phylogeny using generality and vulnerability traits) and allows us to impute the missing (latent) trait data for the Canadian species that are not already in the European network; as we are focused on predicting contemporary interactions, we only retained the values for the tips of the tree. We assumed that all traits (<em>i.e.</em> the feature vectors for the left and right subspaces) were independent, which is a reasonable assumption as every trait/dimension added to the t-SVD has an <em>additive</em> effect to the one before it. Note that the <span class="citation" data-cites="Upham2019InfMam">Upham <em>et al.</em> (2019)</span> tree itself has some uncertainty associated to inner nodes of the phylogeny. In this case study, we have decided to not propagate this uncertainty, as it would complexify the process. The Brownian motion algorithm returns the <em>average</em> value of the trait, and its upper and lower bounds. Because we do not estimate other parameters of the traits’ distributions, we considered that every species trait is represented as a uniform distribution between these bounds; in a situation where the algorithm would return point values for all simulations, one could in theory either estimate the parameters of a distribution for each tip, or draw randomly from the outputs. In all cases, the inferred left and right sub-spaces for the Canadian species pool (<span class="math inline">$\hat{\mathscr{L}}$</span> and <span class="math inline">$\hat{\mathscr{R}}$</span>) have entries that are distributions, representing the range of values for a given species at a given dimension.</p>
            <p>These objects represent the transferred knowledge, which we can use for prediction of the Canadian metaweb.</p>
            <h2 id="step-4-probabilistic-prediction-of-the-destination-network">Step 4: Probabilistic prediction of the destination network</h2>
            <p>The phylogenetic reconstruction of <span class="math inline">$\hat{\mathscr{L}}$</span> and <span class="math inline">$\hat{\mathscr{R}}$</span> has an associated uncertainty, represented by the breadth of the uniform distribution associated to each of their entries. Therefore, we can use this information to assemble a <em>probabilistic</em> metaweb in the sense of <span class="citation" data-cites="Poisot2016StrPro">Poisot <em>et al.</em> (2016)</span>, <em>i.e.</em> in which every interaction is represented as a single, independent, Bernoulli event of probability <span class="math inline"><em>p</em></span>.</p>
            <figure>
            <img src="figures/figure-subspaces.png" id="fig:subspaces" alt="Figure 3: Visual representation of the left (green/purple) and right (green/brown) subspaces, alongside the adjacency matrix of the food web they encode (greyscale). The European metaweb is on the left, and the imputed Canadian metaweb (before data inflation) on the right. This figure illustrates how much structure the left sub-space captures. As we show in fig. 6, the species with a value of 0 in the left subspace are species without any prey." /><figcaption aria-hidden="true">Figure 3: Visual representation of the left (green/purple) and right (green/brown) subspaces, alongside the adjacency matrix of the food web they encode (greyscale). The European metaweb is on the left, and the imputed Canadian metaweb (before data inflation) on the right. This figure illustrates how much structure the left sub-space captures. As we show in fig. 6, the species with a value of 0 in the left subspace are species without any prey.</figcaption>
            </figure>
            <p>Specifically, we have adopted the following approach. For every entry in <span class="math inline">$\hat{\mathscr{L}}$</span> and <span class="math inline">$\hat{\mathscr{R}}$</span>, we draw a value from its distribution. This results in one instance of the possible left (<span class="math inline">$\hat{\mathscr{l}}$</span>) and right (<span class="math inline">$\hat{\mathscr{r}}$</span>) subspaces for the Canadian metaweb. These can be multiplied, to produce one matrix of real values. Because the entries in <span class="math inline">$\hat{\mathscr{l}}$</span> and <span class="math inline">$\hat{\mathscr{r}}$</span> are in the same space where <span class="math inline">ℒ</span> and <span class="math inline">ℛ</span> were originally predicted, it follows that the threshold <span class="math inline"><em>ρ</em></span> estimated for the European metaweb also applies. We use this information to produce one random Canadian metaweb, <span class="math inline">$N = \hat{\mathscr{L}}$</span><span class="math inline">$\hat{\mathscr{R}}' \ge \rho$</span>. As we can see in (fig. 3) the European and Canadian metawebs are structurally similar (as would be expected given the biogeographic similarities) and the two (left and right) subspaces are distinct <em>i.e.</em> capturing predation (generality) and prey (vulnerability) traits.</p>
            <p>Because the intervals around some trait values can be broad <span class="citation" data-cites="Garland1999IntPhy">(in fact, probably broader than what they would actually be, see <em>e.g.</em> Garland <em>et al.</em> 1999)</span>, we repeat the above process <span class="math inline">2 × 10<sup>5</sup></span> times, which results in a probabilistic metaweb <span class="math inline"><em>P</em></span>, where the probability of an interaction (here conveying our degree of trust that it exists given the inferred trait distributions) is given by the number of times where it appears across all random draws <span class="math inline"><em>N</em></span>, divided by the number of samples. An interaction with <span class="math inline"><em>P</em><sub><em>i</em>, <em>j</em></sub> = 1</span> means that these two species were predicted to interact in all <span class="math inline">2 × 10<sup>5</sup></span> random draws.</p>
            <h2 id="data-cleanup-discovery-validation-and-thresholding">Data cleanup, discovery, validation, and thresholding</h2>
            <p>Once the probabilistic metaweb for Canada has been produced, we followed a number of data inflation steps to finalize it. This step is external to the actual transfer learning framework but rather serves as a way to augment and validate the predicted metaweb.</p>
            <figure>
            <img src="figures/figure-validation.png" id="fig:inflation" alt="Figure 4: Left, comparison of the probabilities of interactions assigned by the model to all interactions (grey curve), the subset of interactions found in GLOBI (red), and in the Strong &amp; Leroux (2014) Newfoundland dataset (blue). The model recovers more interaction with a low probability compared to data mining, which can suggest that collected datasets are biased towards more common or easy to identify interactions. Right, distribution of the in-degree and out-degree of the mammals from Canada in the reconstructed metaweb. This figure describes a flat, relatively short food web, in which there are few predators but a large number of preys." /><figcaption aria-hidden="true">Figure 4: Left, comparison of the probabilities of interactions assigned by the model to all interactions (grey curve), the subset of interactions found in GLOBI (red), and in the <span class="citation" data-cites="Strong2014ImpNon">Strong &amp; Leroux (2014)</span> Newfoundland dataset (blue). The model recovers more interaction with a low probability compared to data mining, which can suggest that collected datasets are biased towards more common or easy to identify interactions. Right, distribution of the in-degree and out-degree of the mammals from Canada in the reconstructed metaweb. This figure describes a flat, relatively short food web, in which there are few predators but a large number of preys.</figcaption>
            </figure>
            <p>First, we extracted the subgraph corresponding to the 17 species shared between the European and Canadian pools and replaced these interactions with a probability of 0 (non-interaction) or 1 (interaction), according to their value in the European metaweb. This represents a minute modification of the inferred network (about 0.8% of all species pairs from the Canadian web), but ensures that we are directly re-using knowledge from Europe.</p>
            <p>Second, we looked for all species in the Canadian pool known to the Global Biotic Interactions (GLoBI) database <span class="citation" data-cites="Poelen2014GloBio">(Poelen <em>et al.</em> 2014)</span>, and extracted their known interactions. Because GLOBI aggregates observed interactions, it is not a <em>networks</em> data source, and therefore the only information we can reliably extract from it is that a species pair <em>was reported to interact at least once</em>. This last statement should yet be taken with caution, as some sources in GLOBI <span class="citation" data-cites="Thessen2014KnoExt">(<em>e.g.</em> Thessen &amp; Parr 2014)</span> are produced through text analysis, and therefore may not document direct evidence of the interaction. Nevertheless, should the predictive model work, we would expect that a majority of interactions known to GLOBI would also be predicted. After performing this check, we set the probability of all interactions known to GLOBI (366 in total, 33 of which were not predicted by the model, for a success rate of 91%) to 1.</p>
            <p>Finally, we downloaded the data from <span class="citation" data-cites="Strong2014ImpNon">Strong &amp; Leroux (2014)</span>, who mined various literature sources to identify trophic interactions in Newfoundland. This dataset documented 25 interactions between mammals, only two of which were not part of our (Canada-level) predictions, resulting in a success rate of 92%. These two interactions were added to our predicted metaweb with a probability of 1. A table listing all interactions in the predicted Canadian metaweb can be found in the supplementary material.</p>
            <figure>
            <img src="figures/figure-cutoffs.png" id="fig:thresholds" alt="Figure 5: Left: effect of varying the cutoff for probabilities to be considered non-zero on the number of unique links and on \hat{L}, the probabilistic estimate of the number of links assuming that all interactions are independent. Right: effect of varying the cutoff on the number of disconnected species, and on network connectance. In both panels, the grey line indicates the cutoff P(i\rightarrow j) \approx 0.08 that resulted in the first species losing all of its interactions." /><figcaption aria-hidden="true">Figure 5: Left: effect of varying the cutoff for probabilities to be considered non-zero on the number of unique links and on <span class="math inline"><em>L̂</em></span>, the probabilistic estimate of the number of links assuming that all interactions are independent. Right: effect of varying the cutoff on the number of disconnected species, and on network connectance. In both panels, the grey line indicates the cutoff <span class="math inline"><em>P</em>(<em>i</em> → <em>j</em>) ≈ 0.08</span> that resulted in the first species losing all of its interactions.</figcaption>
            </figure>
            <p>Because the confidence intervals on the inferred trait space are probably over-estimates, we decided to apply a thresholding step to the interactions after the data inflation (fig. 5). <span class="citation" data-cites="Cirtwill2021BuiFoo">Cirtwill &amp; Hambäck (2021)</span> proposed a number of strategies to threshold probabilistic networks. Their methods assume the underlying data to be tag-based sequencing, which represents interactions as co-occurrences of predator and prey within the same tags; this is conceptually identical to our Bernoulli-trial based reconstruction of a probabilistic network. We performed a full analysis of the effect of various cutoffs, and as they either resulted in removing too few interactions, or removing enough interactions that species started to be disconnected from the network, we set this threshold for a probability equivalent to 0 to the largest possible value that still allowed all species to have at least one interaction with a non-zero probability. The need for this slight deviation from the <span class="citation" data-cites="Cirtwill2021BuiFoo">Cirtwill &amp; Hambäck (2021)</span> method highlights the need for additional development on network thresholding.</p>
            <h1 id="results-and-discussion-of-the-case-study">Results and discussion of the case study</h1>
            <p>In fig. 5, we examine the effect of varying the cutoff on <span class="math inline"><em>P</em>(<em>i</em> → <em>j</em>)</span> on the number of links, species, and connectance. Determining a cutoff using the maximum curvature, or central difference approximation of the second order partial derivative, as suggested by <em>e.g.</em> <span class="citation" data-cites="Cirtwill2021BuiFoo">Cirtwill &amp; Hambäck (2021)</span>, results in species being lost, or almost all links being kept. We therefore settled on the value that allowed all species to remain with at least one interaction. This result, in and of itself, suggests that additional methodological developments for the thresholding of probabilistic networks are required.</p>
            <figure>
            <img src="figures/figure-degree.png" id="fig:degree" alt="Figure 6: Top: biological significance of the first dimension. Left: there is a linear relationship between the values on the first dimension of the left subspace and the generality, i.e. the relative number of preys, sensu Schoener (1989). Species with a value of 0 in this subspace are at the bottom-most trophic level. Right: there is, similarly, a linear relationship between the position of a species on the first dimension of the right subspace and its vulnerability, i.e. the relative number of predators. Taken together, these two figures show that the first-order representation of this network would capture its degree distribution. Bottom: topological consequences of the first dimension. Left: differences in the z-score of the actual configuration model for the reconstructed network, and the prediction based only on the first dimension. Right: distribution of the differences in the left panel." /><figcaption aria-hidden="true">Figure 6: Top: biological significance of the first dimension. Left: there is a linear relationship between the values on the first dimension of the left subspace and the generality, <em>i.e.</em> the relative number of preys, <em>sensu</em> <span class="citation" data-cites="Schoener1989FooWeb">Schoener (1989)</span>. Species with a value of 0 in this subspace are at the bottom-most trophic level. Right: there is, similarly, a linear relationship between the position of a species on the first dimension of the right subspace and its vulnerability, <em>i.e.</em> the relative number of predators. Taken together, these two figures show that the first-order representation of this network would capture its degree distribution. Bottom: topological consequences of the first dimension. Left: differences in the <span class="math inline"><em>z</em></span>-score of the actual configuration model for the reconstructed network, and the prediction based only on the first dimension. Right: distribution of the differences in the left panel.</figcaption>
            </figure>
            <p>The t-SVD embedding is able to learn relevant ecological features for the network. fig. 6 shows that the first rank correlates linearly with generality and vulnerability <span class="citation" data-cites="Schoener1989FooWeb">(Schoener 1989)</span>, <em>i.e.</em> the number of preys and predators. Importantly, this implies that a rank 1 approximation represents the configuration model for the metaweb, <em>i.e.</em> a set of random networks generated from a given degree sequence <span class="citation" data-cites="Park2004StaMec">(Park &amp; Newman 2004)</span>. Accounting for the probabilistic nature of the degrees, the rank 1 approximation also represents the <em>soft</em> configuration model <span class="citation" data-cites="vanderHoorn2018SpaMax">(van der Hoorn <em>et al.</em> 2018)</span>. Both models are maximum entropy graph models <span class="citation" data-cites="Garlaschelli2018CovStr">(Garlaschelli <em>et al.</em> 2018)</span>, with sharp (all network realizations satisfy the specified degree sequence) and soft (network realizations satisfy the degree sequence on average) local constraints, respectively. The (soft) configuration model is an unbiased random graph model widely used by ecologists in the context of null hypothesis significance testing of network structure <span class="citation" data-cites="Bascompte2003NesAss">(<em>e.g.</em> Bascompte <em>et al.</em> 2003)</span> and can provide informative priors for Bayesian inference of network structure <span class="citation" data-cites="Young2021BayInf">(<em>e.g.</em> Young <em>et al.</em> 2021)</span>. It is noteworthy that for this metaweb, the relevant information was extracted at the first rank. Because the first rank corresponds to the leading singular value of the system, the results of fig. 6 have a straightforward interpretation: degree-based processes are the most important in structuring the mammalian food web.</p>
            <h1 id="discussion">Discussion</h1>
            <p>One important aspect in which Europe and Canada differ (despite their comparable bioclimatic conditions) is the degree of the legacy of human impacts, which have been much longer in Europe. <span class="citation" data-cites="Nenzen2014Imp850">Nenzén <em>et al.</em> (2014)</span> showed that even at small scales (the Iberian peninsula), mammal food webs retain the signal of both climate change and human activity, even when this human activity was orders of magnitude less important than it is now. Similarly, <span class="citation" data-cites="Yeakel2014ColEco">Yeakel <em>et al.</em> (2014)</span> showed that changes in human occupation over several centuries can lead to food web collapse. Megafauna in particular seems to be very sensitive to human arrival <span class="citation" data-cites="Pires2015PleMeg">(Pires <em>et al.</em> 2015)</span>. In short, there is well-substantiated support for the idea that human footprint affects more than the risk of species extinction <span class="citation" data-cites="Marco2018ChaHum">(Marco <em>et al.</em> 2018)</span>, and can lead to changes in interaction structure. Yet, owing to the inherent plasticity of interactions, there have been documented instances of food webs undergoing rapid collapse/recovery cycles over short periods of time <span class="citation" data-cites="Pedersen2017SigCol">(Pedersen <em>et al.</em> 2017)</span>. The embedding of a network, in a sense, embeds its macro-evolutionary history, especially as RDPG captures ecological signal <span class="citation" data-cites="DallaRiva2016ExpEvo">(Dalla Riva &amp; Stouffer 2016)</span>; at this point, it is important to recall that a metaweb is intended as a catalogue of all possible interactions, which should then be filtered <span class="citation" data-cites="Morales-Castilla2015InfBio">(Morales-Castilla <em>et al.</em> 2015)</span>. In practice (and in this instance) the reconstructed metaweb will predict interactions that are plausible based on the species’ evolutionary history, however some interactions would not be realized due to human impact.</p>
            <p><span class="citation" data-cites="Cirtwill2019QuaFra">Cirtwill <em>et al.</em> (2019)</span> previously made the point that network inference techniques based on Bayesian approaches would perform far better in the presence of an interaction-level informative prior; the desirable properties of such a prior would be that it is expressed as a probability, preferably representing a Bernoulli event, the value of which would be representative of relevant biological processes (probability of predation in this case). We argue that the probability returned at the very last step of our framework may serve as this informative prior; indeed, the output of our analysis can be used in subsequent steps, also possibly involving expert elicitation to validate some of the most strongly recommended interactions. One important <em>caveat</em> to keep in mind when working with interaction inference is that interactions can never really be true negatives (in the current state of our methodological framework and data collection limitations); this renders the task of validating a model through the usual application of binary classification statistics very difficult <span class="citation" data-cites="Strydom2021RoaPre">(although see Strydom <em>et al.</em> 2021a for a discussion of alternative suggestions)</span>. The other way through which our framework can be improved is by substituting the predictors that are used for transfer. For example, in the presence of information on species traits that are known to be predictive of species interactions, one might want to rely on functional rather than phylogenetic distances – in food webs, body size (and allometrically related variables) has been established as such a variable <span class="citation" data-cites="Brose2006ConRes">(Brose <em>et al.</em> 2006)</span>; the identification of relevant functional traits is facilitated by recent methodological developments <span class="citation" data-cites="Rosado2013GoiBac">(Rosado <em>et al.</em> 2013)</span>. It should be noted that <span class="citation" data-cites="Xing2021RisEco">Xing &amp; Fayle (2021)</span> highlight phylogenetic relatedness as one of the core components of network comparison at the global scale. In this case study, we have embedded the original metaweb using t-SVD, because it lends itself to a RDPG reconstruction, which is known to capture the consequences of evolutionary processes <span class="citation" data-cites="DallaRiva2016ExpEvo">(Dalla Riva &amp; Stouffer 2016)</span>; this being said, there are others ways to embed graphs <span class="citation" data-cites="Arsov2019NetEmb Cai2017ComSur Cao2019NetEmb">(Cai <em>et al.</em> 2017; Arsov &amp; Mirceva 2019; Cao <em>et al.</em> 2019)</span>, which can be used as alternatives.</p>
            <p>As <span class="citation" data-cites="Herbert1965Dun">Herbert (1965)</span> rightfully pointed out, “[y]ou can’t draw neat lines around planet-wide problems”; in this regard, our approach must contend with two interesting problems. The first is the limit of the metaweb to embed and transfer. If the initial metaweb is too narrow in scope, notably from a taxonomic point of view, the chances of finding another area with enough related species to make a reliable inference decrease. This is notably true if the metaweb is assembled in an area with mostly endemic species. Conversely, the metaweb should be reliably filled, which assumes that the <span class="math inline"><em>S</em><sup>2</sup></span> interactions in a pool of <span class="math inline"><em>S</em></span> species have been examined, either through literature surveys or expert elicitation. The second problem is to determine which area should be used to infer the new metaweb in, as this determines the species pool that must be used. In our application, we focused on the mammals of Canada. The upside of this approach is that information at the country level is likely to be required by policy makers and stakeholders for their biodiversity assessment, as each country tends to set goals at the national level <span class="citation" data-cites="Buxton2021KeyInf">(Buxton <em>et al.</em> 2021)</span> for which quantitative instruments are designed <span class="citation" data-cites="Turak2017UsiEss">(Turak <em>et al.</em> 2017)</span>, with specific strategies often enacted at smaller scales <span class="citation" data-cites="Ray2021BioCri">(Ray <em>et al.</em> 2021)</span>. Yet these national divisions, in large parts of the world, reflect nothing except for the legacy of settler colonialism, and operating under them must be done under the clear realization that they contributed to the ongoing biodiversity crisis <span class="citation" data-cites="Adam2014EleTre">(Adam 2014)</span>, can reinforce environmental injustice <span class="citation" data-cites="Choudry2013SavBio Dominguez2020DecCon">(Choudry 2013; Domínguez &amp; Luoma 2020)</span>, and on Turtle Island especially, will probably end up being replaced by Indigenous principles of land management <span class="citation" data-cites="Eichhorn2019SteDec Nokmaq2021AwaSle">(Eichhorn <em>et al.</em> 2019; No’kmaq <em>et al.</em> 2021)</span>.</p>
            <p><strong>Acknowledgements:</strong> We acknowledge that this study was conducted on land within the traditional unceded territory of the Saint Lawrence Iroquoian, Anishinabewaki, Mohawk, Huron-Wendat, and Omàmiwininiwak nations. TP, TS, DC, and LP received funding from the Canadian Institue for Ecology &amp; Evolution. FB is funded by the Institut de Valorisation des Données. TS, SB, and TP are funded by a donation from the Courtois Foundation. CB was awarded a Mitacs Elevate Fellowship no. IT12391, in partnership with fRI Research, and also acknowledges funding from Alberta Innovates and the Forest Resources Improvement Association of Alberta. M-JF acknowledges funding from NSERC Discovery Grant and NSERC CRC. RR is funded by New Zealand’s Biological Heritage Ngā Koiora Tuku Iho National Science Challenge, administered by New Zealand Ministry of Business, Innovation, and Employment. BM is funded by the NSERC Alexander Graham Bell Canada Graduate Scholarship and the FRQNT master’s scholarship. LP acknowledges funding from NSERC Discovery Grant (NSERC RGPIN-2019-05771). TP acknowledges financial support from NSERC through the Discovery Grants and Discovery Accelerator Supplement programs.</p>
            <h1 class="unnumbered" id="references">References</h1>
            <div id="refs" class="references csl-bib-body hanging-indent" role="doc-bibliography">
            <div id="ref-Adam2014EleTre" class="csl-entry" role="doc-biblioentry">
            Adam, R. (2014). <em>Elephant treaties: The Colonial legacy of the biodiversity crisis</em>. UPNE.
            </div>
            <div id="ref-Albouy2019MarFis" class="csl-entry" role="doc-biblioentry">
            Albouy, C., Archambault, P., Appeltans, W., Araújo, M.B., Beauchesne, D., Cazelles, K., <em>et al.</em> (2019). The marine fish food web is globally connected. <em>Nature Ecology &amp; Evolution</em>, 3, 1153–1161.
            </div>
            <div id="ref-Arsov2019NetEmb" class="csl-entry" role="doc-biblioentry">
            Arsov, N. &amp; Mirceva, G. (2019). Network Embedding: An Overview. <em>arXiv:1911.11726 [cs, stat]</em>.
            </div>
            <div id="ref-Banville2021ManJl" class="csl-entry" role="doc-biblioentry">
            Banville, F., Vissault, S. &amp; Poisot, T. (2021). Mangal.jl and EcologicalNetworks.jl: Two complementary packages for analyzing ecological networks in Julia. <em>Journal of Open Source Software</em>, 6, 2721.
            </div>
            <div id="ref-Bascompte2003NesAss" class="csl-entry" role="doc-biblioentry">
            Bascompte, J., Jordano, P., Melian, C.J. &amp; Olesen, J.M. (2003). The nested assembly of plant-animal mutualistic networks. <em>Proceedings of the National Academy of Sciences</em>, 100, 9383–9387.
            </div>
            <div id="ref-Beckerman2006ForBio" class="csl-entry" role="doc-biblioentry">
            Beckerman, A.P., Petchey, O.L. &amp; Warren, P.H. (2006). Foraging biology predicts food web complexity. <em>Proceedings of the National Academy of Sciences</em>, 103, 13745–13749.
            </div>
            <div id="ref-Bezanson2017JulFre" class="csl-entry" role="doc-biblioentry">
            Bezanson, J., Edelman, A., Karpinski, S. &amp; Shah, V. (2017). Julia: A Fresh Approach to Numerical Computing. <em>SIAM Review</em>, 59, 65–98.
            </div>
            <div id="ref-Boeckaerts2021PreBac" class="csl-entry" role="doc-biblioentry">
            Boeckaerts, D., Stock, M., Criel, B., Gerstmans, H., De Baets, B. &amp; Briers, Y. (2021). Predicting bacteriophage hosts based on sequences of annotated receptor-binding proteins. <em>Scientific Reports</em>, 11, 1467.
            </div>
            <div id="ref-Braga2021PhyRec" class="csl-entry" role="doc-biblioentry">
            Braga, M.P., Janz, N., Nylin, S., Ronquist, F. &amp; Landis, M.J. (2021). Phylogenetic reconstruction of ancestral ecological networks through time for pierid butterflies and their host plants. <em>Ecology Letters</em>, n/a.
            </div>
            <div id="ref-Brose2006ConRes" class="csl-entry" role="doc-biblioentry">
            Brose, U., Jonsson, T., Berlow, E.L., Warren, P., Banasek-Richter, C., Bersier, L.-F., <em>et al.</em> (2006). ConsumerResource Body-Size Relationships in Natural Food Webs. <em>Ecology</em>, 87, 2411–2417.
            </div>
            <div id="ref-Buxton2021KeyInf" class="csl-entry" role="doc-biblioentry">
            Buxton, R.T., Bennett, J.R., Reid, A.J., Shulman, C., Cooke, S.J., Francis, C.M., <em>et al.</em> (2021). Key information needs to move from knowledge to action for biodiversity conservation in Canada. <em>Biological Conservation</em>, 256, 108983.
            </div>
            <div id="ref-Cai2017ComSur" class="csl-entry" role="doc-biblioentry">
            Cai, H., Zheng, V.W. &amp; Chang, K.C.-C. (2017). A Comprehensive Survey of Graph Embedding: Problems, Techniques and Applications. <em>arXiv preprint arXiv:1709.07604</em>.
            </div>
            <div id="ref-Cameron2019UneGlo" class="csl-entry" role="doc-biblioentry">
            Cameron, E.K., Sundqvist, M.K., Keith, S.A., CaraDonna, P.J., Mousing, E.A., Nilsson, K.A., <em>et al.</em> (2019). Uneven global distribution of food web studies under climate change. <em>Ecosphere</em>, 10, e02645.
            </div>
            <div id="ref-Cao2019NetEmb" class="csl-entry" role="doc-biblioentry">
            Cao, R.-M., Liu, S.-Y. &amp; Xu, X.-K. (2019). Network embedding for link prediction: The pitfall and improvement. <em>Chaos: An Interdisciplinary Journal of Nonlinear Science</em>, 29, 103102.
            </div>
            <div id="ref-Cavender-Bares2009MerCom" class="csl-entry" role="doc-biblioentry">
            Cavender-Bares, J., Kozak, K.H., Fine, P.V.A. &amp; Kembel, S.W. (2009). The merging of community ecology and phylogenetic biology. <em>Ecology Letters</em>, 12, 693–715.
            </div>
            <div id="ref-Choudry2013SavBio" class="csl-entry" role="doc-biblioentry">
            Choudry, A. (2013). Saving biodiversity, for whom and for what? Conservation NGOs, complicity, colonialism and conquest in an era of capitalist globalization. In: <em>NGOization: Complicity, contradictions and prospects</em>. Bloomsbury Publishing, pp. 24–44.
            </div>
            <div id="ref-Cirtwill2019QuaFra" class="csl-entry" role="doc-biblioentry">
            Cirtwill, A.R., Eklf, A., Roslin, T., Wootton, K. &amp; Gravel, D. (2019). A quantitative framework for investigating the reliability of empirical network construction. <em>Methods in Ecology and Evolution</em>, 0.
            </div>
            <div id="ref-Cirtwill2021BuiFoo" class="csl-entry" role="doc-biblioentry">
            Cirtwill, A.R. &amp; Hambäck, P. (2021). Building food networks from molecular data: Bayesian or fixed-number thresholds for including links. <em>Basic and Applied Ecology</em>, 50, 67–76.
            </div>
            <div id="ref-DallaRiva2016ExpEvo" class="csl-entry" role="doc-biblioentry">
            Dalla Riva, G.V. &amp; Stouffer, D.B. (2016). Exploring the evolutionary signature of food webs’ backbones using functional traits. <em>Oikos</em>, 125, 446–456.
            </div>
            <div id="ref-Dansereau2021SimJl" class="csl-entry" role="doc-biblioentry">
            Dansereau, G. &amp; Poisot, T. (2021). SimpleSDMLayers.jl and GBIF.jl: A Framework for Species Distribution Modeling in Julia. <em>Journal of Open Source Software</em>, 6, 2872.
            </div>
            <div id="ref-Dominguez2020DecCon" class="csl-entry" role="doc-biblioentry">
            Domínguez, L. &amp; Luoma, C. (2020). Decolonising Conservation Policy: How Colonial Land and Conservation Ideologies Persist and Perpetuate Indigenous Injustices at the Expense of the Environment. <em>Land</em>, 9, 65.
            </div>
            <div id="ref-Dormann2010EvoCli" class="csl-entry" role="doc-biblioentry">
            Dormann, C.F., Gruber, B., Winter, M. &amp; Herrmann, D. (2010). Evolution of climate niches in European mammals? <em>Biology Letters</em>, 6, 229–232.
            </div>
            <div id="ref-Dunne2006NetStr" class="csl-entry" role="doc-biblioentry">
            Dunne, J.A. (2006). The Network Structure of Food Webs. In: <em>Ecological networks: Linking structure and dynamics</em> (eds. Dunne, J.A. &amp; Pascual, M.). Oxford University Press, pp. 27–86.
            </div>
            <div id="ref-Eichhorn2019SteDec" class="csl-entry" role="doc-biblioentry">
            Eichhorn, M.P., Baker, K. &amp; Griffiths, M. (2019). Steps towards decolonising biogeography. <em>Frontiers of Biogeography</em>, 12, 1–7.
            </div>
            <div id="ref-Eklof2016PhyCom" class="csl-entry" role="doc-biblioentry">
            Eklöf, A. &amp; Stouffer, D.B. (2016). The phylogenetic component of food web structure and intervality. <em>Theoretical Ecology</em>, 9, 107–115.
            </div>
            <div id="ref-Garland1999IntPhy" class="csl-entry" role="doc-biblioentry">
            Garland, T., JR., Midford, P.E. &amp; Ives, A.R. (1999). An Introduction to Phylogenetically Based Statistical Methods, with a New Method for Confidence Intervals on Ancestral Values1. <em>American Zoologist</em>, 39, 374–388.
            </div>
            <div id="ref-Garlaschelli2018CovStr" class="csl-entry" role="doc-biblioentry">
            Garlaschelli, D., Hollander, F. den &amp; Roccaverde, A. (2018). Covariance structure behind breaking of ensemble equivalence in random graphs. <em>Journal of Statistical Physics</em>, 173, 644–662.
            </div>
            <div id="ref-GBIFSecretariat2021GbiBac" class="csl-entry" role="doc-biblioentry">
            GBIF Secretariat. (2021). GBIF Backbone Taxonomy.
            </div>
            <div id="ref-Gerhold2015PhyPat" class="csl-entry" role="doc-biblioentry">
            Gerhold, P., Cahill, J.F., Winter, M., Bartish, I.V. &amp; Prinzing, A. (2015). Phylogenetic patterns are not proxies of community assembly mechanisms (they are far better). <em>Functional Ecology</em>, 29, 600–614.
            </div>
            <div id="ref-Grenie2021HarTax" class="csl-entry" role="doc-biblioentry">
            Grenié, M., Berti, E., Carvajal-Quintero, J.D., Winter, M. &amp; Sagouis, A. (2021). Harmonizing taxon names in biodiversity data: A review of tools, databases, and best practices.
            </div>
            <div id="ref-Halko2011FinStr" class="csl-entry" role="doc-biblioentry">
            Halko, N., Martinsson, P.G. &amp; Tropp, J.A. (2011). Finding Structure with Randomness: Probabilistic Algorithms for Constructing Approximate Matrix Decompositions. <em>SIAM Review</em>, 53, 217–288.
            </div>
            <div id="ref-Herbert1965Dun" class="csl-entry" role="doc-biblioentry">
            Herbert, F. (1965). <em>Dune</em>. First. Chilton Book Company, Philadelphia.
            </div>
            <div id="ref-Hortal2015SevSho" class="csl-entry" role="doc-biblioentry">
            Hortal, J., de Bello, F., Diniz-Filho, J.A.F., Lewinsohn, T.M., Lobo, J.M. &amp; Ladle, R.J. (2015). Seven Shortfalls that Beset Large-Scale Knowledge of Biodiversity. <em>Annual Review of Ecology, Evolution, and Systematics</em>, 46, 523–549.
            </div>
            <div id="ref-Hutchinson2017CopSig" class="csl-entry" role="doc-biblioentry">
            Hutchinson, M.C., Cagua, E.F. &amp; Stouffer, D.B. (2017). Cophylogenetic signal is detectable in pollination interactions across ecological scales. <em>Ecology</em>, n/a–n/a.
            </div>
            <div id="ref-Jordano2016ChaEco" class="csl-entry" role="doc-biblioentry">
            Jordano, P. (2016a). Chasing Ecological Interactions. <em>PLOS Biol</em>, 14, e1002559.
            </div>
            <div id="ref-Jordano2016SamNet" class="csl-entry" role="doc-biblioentry">
            Jordano, P. (2016b). Sampling networks of ecological interactions. <em>Functional Ecology</em>, 30, 1883–1893.
            </div>
            <div id="ref-Litsios2012EffPhy" class="csl-entry" role="doc-biblioentry">
            Litsios, G. &amp; Salamin, N. (2012). Effects of Phylogenetic Signal on Ancestral State Reconstruction. <em>Systematic Biology</em>, 61, 533–538.
            </div>
            <div id="ref-Maiorano2020DatTet" class="csl-entry" role="doc-biblioentry">
            Maiorano, L., Montemaggiori, A., Ficetola, G.F., O’Connor, L. &amp; Thuiller, W. (2020a). Data from: Tetra-EU 1.0: A species-level trophic meta-web of European tetrapods.
            </div>
            <div id="ref-Maiorano2020TetEu" class="csl-entry" role="doc-biblioentry">
            Maiorano, L., Montemaggiori, A., Ficetola, G.F., O’Connor, L. &amp; Thuiller, W. (2020b). TETRA-EU 1.0: A species-level trophic metaweb of European tetrapods. <em>Global Ecology and Biogeography</em>, 29, 1452–1457.
            </div>
            <div id="ref-Marco2018ChaHum" class="csl-entry" role="doc-biblioentry">
            Marco, M.D., Venter, O., Possingham, H.P. &amp; Watson, J.E.M. (2018). Changes in human footprint drive changes in species extinction risk. <em>Nature Communications</em>, 9, 4621.
            </div>
            <div id="ref-Morales-Castilla2015InfBio" class="csl-entry" role="doc-biblioentry">
            Morales-Castilla, I., Matias, M.G., Gravel, D. &amp; Araújo, M.B. (2015). Inferring biotic interactions from proxies. <em>Trends in Ecology &amp; Evolution</em>, 30, 347–356.
            </div>
            <div id="ref-Mouquet2012EcoAdv" class="csl-entry" role="doc-biblioentry">
            Mouquet, N., Devictor, V., Meynard, C.N., Munoz, F., Bersier, L.-F., Chave, J., <em>et al.</em> (2012). Ecophylogenetics: Advances and perspectives. <em>Biological Reviews</em>, 87, 769–785.
            </div>
            <div id="ref-Nenzen2014Imp850" class="csl-entry" role="doc-biblioentry">
            Nenzén, H.K., Montoya, D. &amp; Varela, S. (2014). The Impact of 850,000 Years of Climate Changes on the Structure and Dynamics of Mammal Food Webs. <em>PLOS ONE</em>, 9, e106651.
            </div>
            <div id="ref-Nokmaq2021AwaSle" class="csl-entry" role="doc-biblioentry">
            No’kmaq, M., Marshall, A., Beazley, K.F., Hum, J., joudry, shalan, Papadopoulos, A., <em>et al.</em> (2021). <span>“Awakening the sleeping giant”</span>: Re-Indigenization principles for transforming biodiversity conservation in Canada and beyond. <em>FACETS</em>, 6, 839–869.
            </div>
            <div id="ref-OConnor2020UnvFoo" class="csl-entry" role="doc-biblioentry">
            O’Connor, L.M.J., Pollock, L.J., Braga, J., Ficetola, G.F., Maiorano, L., Martinez-Almoyna, C., <em>et al.</em> (2020). Unveiling the food webs of tetrapods across Europe through the prism of the Eltonian niche. <em>Journal of Biogeography</em>, 47, 181–192.
            </div>
            <div id="ref-Pan2010SurTra" class="csl-entry" role="doc-biblioentry">
            Pan, S.J. &amp; Yang, Q. (2010). A Survey on Transfer Learning. <em>IEEE Transactions on Knowledge and Data Engineering</em>, 22, 1345–1359.
            </div>
            <div id="ref-Park2004StaMec" class="csl-entry" role="doc-biblioentry">
            Park, J. &amp; Newman, M.E.J. (2004). Statistical mechanics of networks. <em>Physical Review E</em>, 70, 066117.
            </div>
            <div id="ref-Pedersen2017SigCol" class="csl-entry" role="doc-biblioentry">
            Pedersen, E.J., Thompson, P.L., Ball, R.A., Fortin, M.-J., Gouhier, T.C., Link, H., <em>et al.</em> (2017). Signatures of the collapse and incipient recovery of an overexploited marine ecosystem. <em>Royal Society Open Science</em>, 4, 170215.
            </div>
            <div id="ref-Pires2015PleMeg" class="csl-entry" role="doc-biblioentry">
            Pires, M.M., Koch, P.L., Fariña, R.A., de Aguiar, M.A.M., dos Reis, S.F. &amp; Guimarães, P.R. (2015). Pleistocene megafaunal interaction networks became more vulnerable after human arrival. <em>Proceedings of the Royal Society B: Biological Sciences</em>, 282, 20151367.
            </div>
            <div id="ref-Poelen2014GloBio" class="csl-entry" role="doc-biblioentry">
            Poelen, J.H., Simons, J.D. &amp; Mungall, C.J. (2014). Global biotic interactions: An open infrastructure to share and analyze species-interaction datasets. <em>Ecological Informatics</em>, 24, 148–159.
            </div>
            <div id="ref-Poisot2019EcoJl" class="csl-entry" role="doc-biblioentry">
            Poisot, T., Belisle, Z., Hoebeke, L., Stock, M. &amp; Szefer, P. (2019). EcologicalNetworks.jl - analysing ecological networks. <em>Ecography</em>.
            </div>
            <div id="ref-Poisot2021GloKno" class="csl-entry" role="doc-biblioentry">
            Poisot, T., Bergeron, G., Cazelles, K., Dallas, T., Gravel, D., MacDonald, A., <em>et al.</em> (2021a). Global knowledge gaps in species interaction networks data. <em>Journal of Biogeography</em>, n/a.
            </div>
            <div id="ref-Poisot2012DisSpe" class="csl-entry" role="doc-biblioentry">
            Poisot, T., Canard, E., Mouillot, D., Mouquet, N. &amp; Gravel, D. (2012). The dissimilarity of species interaction networks. <em>Ecology Letters</em>, 15, 1353–1361.
            </div>
            <div id="ref-Poisot2016StrPro" class="csl-entry" role="doc-biblioentry">
            Poisot, T., Cirtwill, A.R., Cazelles, K., Gravel, D., Fortin, M.-J. &amp; Stouffer, D.B. (2016). The structure of probabilistic networks. <em>Methods in Ecology and Evolution</em>, 7, 303–312.
            </div>
            <div id="ref-Poisot2021ImpMam" class="csl-entry" role="doc-biblioentry">
            Poisot, T., Ouellet, M.-A., Mollentze, N., Farrell, M.J., Becker, D.J., Albery, G.F., <em>et al.</em> (2021b). Imputing the mammalian virome with linear filtering and singular value decomposition. <em>arXiv:2105.14973 [q-bio]</em>.
            </div>
            <div id="ref-Poisot2018IntRet" class="csl-entry" role="doc-biblioentry">
            Poisot, T. &amp; Stouffer, D.B. (2018). Interactions retain the co-phylogenetic matching that communities lost. <em>Oikos</em>, 127, 230–238.
            </div>
            <div id="ref-Price2003MacThe" class="csl-entry" role="doc-biblioentry">
            Price, P.W. (2003). <em>Macroevolutionary theory on macroecological patterns</em>. Cambridge University Press.
            </div>
            <div id="ref-Ray2021BioCri" class="csl-entry" role="doc-biblioentry">
            Ray, J.C., Grimm, J. &amp; Olive, A. (2021). The biodiversity crisis in Canada: Failures and challenges of federal and sub-national strategic and legal frameworks. <em>FACETS</em>, 6, 1044–1068.
            </div>
            <div id="ref-Reeve2016HowPar" class="csl-entry" role="doc-biblioentry">
            Reeve, R., Leinster, T., Cobbold, C.A., Thompson, J., Brummitt, N., Mitchell, S.N., <em>et al.</em> (2016). How to partition diversity. <em>arXiv:1404.6520 [q-bio]</em>.
            </div>
            <div id="ref-Rosado2013GoiBac" class="csl-entry" role="doc-biblioentry">
            Rosado, B.H.P., Dias, A. &amp; de Mattos, E. (2013). Going Back to Basics: Importance of Ecophysiology when Choosing Functional Traits for Studying Communities and Ecosystems. <em>Natureza &amp; conservaç~ao revista brasileira de conservaç~ao da natureza</em>, 11, 15–22.
            </div>
            <div id="ref-Runghen2021ExpNod" class="csl-entry" role="doc-biblioentry">
            Runghen, R., Stouffer, D.B. &amp; Dalla Riva, G.V. (2021). Exploiting node metadata to predict interactions in large networks using graph embedding and neural networks.
            </div>
            <div id="ref-Schoener1989FooWeb" class="csl-entry" role="doc-biblioentry">
            Schoener, T.W. (1989). Food webs from the small to the large. <em>Ecology</em>, 70, 1559–1589.
            </div>
            <div id="ref-Solis-Lemus2017PhyPac" class="csl-entry" role="doc-biblioentry">
            Solís-Lemus, C., Bastide, P. &amp; Ané, C. (2017). PhyloNetworks: A Package for Phylogenetic Networks. <em>Molecular Biology and Evolution</em>, 34, 3292–3298.
            </div>
            <div id="ref-Stock2021PaiLea" class="csl-entry" role="doc-biblioentry">
            Stock, M. (2021). Pairwise learning for predicting pollination interactions based on traits and phylogeny. <em>Ecological Modelling</em>, 14.
            </div>
            <div id="ref-Stouffer2012EvoCon" class="csl-entry" role="doc-biblioentry">
            Stouffer, D.B., Sales-Pardo, M., Sirer, M.I. &amp; Bascompte, J. (2012). Evolutionary Conservation of Species’ Roles in Food Webs. <em>Science</em>, 335, 1489–1492.
            </div>
            <div id="ref-Strong2014ImpNon" class="csl-entry" role="doc-biblioentry">
            Strong, J.S. &amp; Leroux, S.J. (2014). Impact of Non-Native Terrestrial Mammals on the Structure of the Terrestrial Mammal Food Web of Newfoundland, Canada. <em>PLOS ONE</em>, 9, e106264.
            </div>
            <div id="ref-Strydom2021RoaPre" class="csl-entry" role="doc-biblioentry">
            Strydom, T., Catchen, M.D., Banville, F., Caron, D., Dansereau, G., Desjardins-Proulx, P., <em>et al.</em> (2021a). <em>A Roadmap Toward Predicting Species Interaction Networks (Across Space and Time)</em> (Preprint). EcoEvoRxiv.
            </div>
            <div id="ref-Strydom2021SvdEnt" class="csl-entry" role="doc-biblioentry">
            Strydom, T., Dalla Riva, G.V. &amp; Poisot, T. (2021b). SVD Entropy Reveals the High Complexity of Ecological Networks. <em>Frontiers in Ecology and Evolution</em>, 9.
            </div>
            <div id="ref-Thessen2014KnoExt" class="csl-entry" role="doc-biblioentry">
            Thessen, A.E. &amp; Parr, C.S. (2014). Knowledge extraction and semantic annotation of text from the encyclopedia of life. <em>PloS one</em>, 9, e89550.
            </div>
            <div id="ref-Torrey2010TraLea" class="csl-entry" role="doc-biblioentry">
            Torrey, L. &amp; Shavlik, J. (2010). Transfer learning. In: <em>Handbook of research on machine learning applications and trends: Algorithms, methods, and techniques</em>. IGI global, pp. 242–264.
            </div>
            <div id="ref-Trojelsgaard2016EcoNet" class="csl-entry" role="doc-biblioentry">
            Trøjelsgaard, K. &amp; Olesen, J.M. (2016). Ecological networks in motion: Micro- and macroscopic variability across scales. <em>Functional Ecology</em>, 30, 1926–1935.
            </div>
            <div id="ref-Turak2017UsiEss" class="csl-entry" role="doc-biblioentry">
            Turak, E., Brazill-Boast, J., Cooney, T., Drielsma, M., DelaCruz, J., Dunkerley, G., <em>et al.</em> (2017). Using the essential biodiversity variables framework to measure biodiversity change at national scale. <em>Biological Conservation</em>, SI:measures of biodiversity, 213, 264–271.
            </div>
            <div id="ref-Upham2019InfMam" class="csl-entry" role="doc-biblioentry">
            Upham, N.S., Esselstyn, J.A. &amp; Jetz, W. (2019). Inferring the mammal tree: Species-level sets of phylogenies for questions in ecology, evolution, and conservation. <em>PLOS Biology</em>, 17, e3000494.
            </div>
            <div id="ref-vanderHoorn2018SpaMax" class="csl-entry" role="doc-biblioentry">
            van der Hoorn, P., Lippner, G. &amp; Krioukov, D. (2018). Sparse Maximum-Entropy Random Graphs with a Given Power-Law Degree Distribution. <em>Journal of Statistical Physics</em>, 173, 806–844.
            </div>
            <div id="ref-Xing2021RisEco" class="csl-entry" role="doc-biblioentry">
            Xing, S. &amp; Fayle, T.M. (2021). The rise of ecological network meta-analyses: Problems and prospects. <em>Global Ecology and Conservation</em>, 30, e01805.
            </div>
            <div id="ref-Yeakel2014ColEco" class="csl-entry" role="doc-biblioentry">
            Yeakel, J.D., Pires, M.M., Rudolf, L., Dominy, N.J., Koch, P.L., Guimarães, P.R., <em>et al.</em> (2014). Collapse of an ecological network in Ancient Egypt. <em>PNAS</em>, 111, 14472–14477.
            </div>
            <div id="ref-Youden1950IndRat" class="csl-entry" role="doc-biblioentry">
            Youden, W.J. (1950). Index for rating diagnostic tests. <em>Cancer</em>, 3, 32–35.
            </div>
            <div id="ref-Young2021BayInf" class="csl-entry" role="doc-biblioentry">
            Young, J.-G., Cantwell, G.T. &amp; Newman, M.E.J. (2021). Bayesian inference of network structure from unreliable data. <em>Journal of Complex Networks</em>, 8.
            </div>
            <div id="ref-Young2007RanDot" class="csl-entry" role="doc-biblioentry">
            Young, S.J. &amp; Scheinerman, E.R. (2007). Random Dot Product Graph Models for Social Networks. In: <em>Algorithms and Models for the Web-Graph</em>, Lecture Notes in Computer Science (eds. Bonato, A. &amp; Chung, F.R.K.). Springer, Berlin, Heidelberg, pp. 138–149.
            </div>
            <div id="ref-Zhu2006AutDim" class="csl-entry" role="doc-biblioentry">
            Zhu, M. &amp; Ghodsi, A. (2006). Automatic dimensionality selection from the scree plot via the use of profile likelihood. <em>Computational Statistics &amp; Data Analysis</em>, 51, 918–930.
            </div>
            </div>
        </article>
    </main>

    <footer>
        <div class="container">
            <h2>Affiliations</h2> 
            <sup>1</sup>&nbsp;Département de Sciences Biologiques, Université de Montréal, Montréal, Canada; <sup>2</sup>&nbsp;Québec Centre for Biodiversity Sciences, Montréal, Canada; <sup>3</sup>&nbsp;Université de Sherbrooke, Sherbrooke, Canada; <sup>4</sup>&nbsp;Department of Forest Resources Management, University of British Columbia, Vancouver, Canada; <sup>5</sup>&nbsp;Department of Biology, McGill University, Montréal, Canada; <sup>6</sup>&nbsp;Department of Ecology &amp; Evolutionary Biology, University of Toronto, Toronto, Canada; <sup>7</sup>&nbsp;Centre for Integrative Ecology, School of Biological Sciences, University of Canterbury, Canterbury, New Zealand; <sup>8</sup>&nbsp;School of Mathematics and Statistics, University of Canterbury, Canterbury, New Zealand
            <br />
            <sup>‡</sup>&nbsp;These authors contributed equally to the work
        </div>
    </footer>
    
</body>
</html>