<!DOCTYPE html>
<html lang="" dir="ltr">

<head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <title>Food web reconstruction through phylogenetic transfer of low-rank network representation</title>
    <script src="https://hypothes.is/embed.js" async></script>
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/normalize/8.0.1/normalize.min.css"
    integrity="sha256-l85OmPOjvil/SOvVt3HnSSjzF1TUMyT9eV0c2BzEGzU=" crossorigin="anonymous" />
    <style>
        @import url('https://fonts.googleapis.com/css2?family=Roboto:ital,wght@0,300;0,400;0,500;1,300;1,400;1,500&display=swap');
        @import url('https://fonts.googleapis.com/css2?family=Roboto+Condensed:ital,wght@0,300;0,400;0,700;1,300;1,400;1,700&display=swap');
        @import url('https://fonts.googleapis.com/css2?family=Roboto+Mono:ital,wght@0,400;0,500;1,400;1,500&display=swap');

        :root {
            --citation: #1976D2;
            --structure: hsl(210, 61%, 43%);
            
            --light: #474747;
            --pale: #a5a5a5;
            --grey: #e0e0e0;
            --nav: #fdfdfd;

            --width: 820px;
            --inner-space: 1.2em;
        }

        body {
            font-family: "Roboto", sans-serif;
            margin: 0 0;
            padding: 0 0;
        }

        article > p {
            line-height: 135%;
        }
        article > h1 {
            padding-top: 3em;
        }
        a {
            text-decoration: none;
            color: #000;
            border-bottom: 1px solid var(--structure);
        }
        a:hover {
            font-weight: 500;
        }
        .citation {
            color: var(--citation);
        }
        .container {
            margin: 0em auto;
            max-width: var(--width);
            padding: var(--inner-space);
        }
        header > .container {
            margin-top: 4em;
        }
        code, pre {
            font-family: "Roboto Mono", monospace;
        }
        pre {
            white-space: pre-wrap;
        }
        h1, h2, h3 {
            color: var(--structure);
            font-family: "Roboto Condensed", sans-serif;
            font-weight: 400;
        }
        header {
            font-family: "Roboto Condensed", sans-serif;
            z-index: 5;
            color: black;
            margin-top: 4rem;
        }
        .header--title {
            font-weight: 400;
            font-size: 2rem;
            margin-bottom: 1rem;
        }
        .header--authors > span {
            color: var(--light);
            font-weight: 400;
        }
        main {
            padding-top: 8em !important;
            padding-bottom: 8em !important;
        }
        footer {
            background-color: var(--grey);
            font-weight: 300;
        }
        .abstract {
            color: var(--light);
            margin-bottom: 5em;
        }
        figure {
            max-width: 100%;
            border: 1px solid var(--pale);
            border-radius: 2px;
            margin: 1em 0.5em;
        }
        figure > img {
            max-width: 100%;
            display: block;
            margin: 0em auto;
        }
        figure > figcaption {
            padding: var(--inner-space);
        }
        table {
            width: 100%;
            min-width: 100%;
            border-collapse: collapse;
        }
        table > caption {
            padding: var(--inner-space);
        }
        figcaption, caption {
            color: var(--light);
            font-weight: 300;
        }
        thead {
            color: var(--structure);
        }
        td, th {
            padding: 3px;
        }
        tr {
            border-top: 1px solid var(--structure);
            border-bottom: 1px solid var(--structure);
        }
        tbody > tr:nth-child(even) {background: var(--grey)};
        tbody {
            border-bottom: 1px solid var(--structure);
        }
        #refs {
            font-size: 90%;
            column-count: 2;
        }
        #refs > .csl-entry {
            padding-bottom: 0.2em;
            padding-top: 0.2em;
        }
        .hanging-indent > .csl-entry {
            padding-left: 15px ;
            text-indent: -15px ;
        }    </style>
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.10.0-rc.1/dist/katex.min.css"
    integrity="sha384-D+9gmBxUQogRLqvARvNLmA9hS2x//eK1FhVb9PiU86gmcrBrJAQT8okdJ4LMp2uv" crossorigin="anonymous" />
    <script defer src="https://cdn.jsdelivr.net/npm/katex@0.10.0-rc.1/dist/katex.min.js"
    integrity="sha384-483A6DwYfKeDa0Q52fJmxFXkcPCFfnXMoXblOkJ4JcA8zATN6Tm78UNL72AKk+0O"
    crossorigin="anonymous"></script>
    <script defer src="https://cdn.jsdelivr.net/npm/katex@0.10.0-rc.1/dist/contrib/auto-render.min.js"
    integrity="sha384-yACMu8JWxKzSp/C1YV86pzGiQ/l1YUfE8oPuahJQxzehAjEt2GiQuy/BIvl9KyeF"
    crossorigin="anonymous"></script>
    <script>
        document.addEventListener("DOMContentLoaded", function () {
            renderMathInElement(document.body, {
                delimiters: [
                { left: "$$", right: "$$", display: true },
                { left: "$", right: "$", display: false }
                ]
            });
        });
    </script>
</head>

<body>
    <header>
        <div class="container">
            <div class="header--title">Food web reconstruction through phylogenetic transfer of low-rank network representation</div>
            <div class="header--authors">
                <span class="header--author">
                <a href="https://orcid.org/0000-0001-6067-1349" target="_blank">
                    Tanya Strydom
                </a>
                <sup class="affilcode">1</sup>
                <sup class="affilcode">2</sup>
                <sup class="affilcode">‚Ä°</sup>
                </span><span class="header--author">
                <a href="https://orcid.org/0000-0003-0193-5441" target="_blank">
                    Salom√© Bouskila
                </a>
                <sup class="affilcode">1</sup>
                <sup class="affilcode">‚Ä°</sup>
                </span><span class="header--author">
                <a href="https://orcid.org/0000-0001-9051-0597" target="_blank">
                    Francis Banville
                </a>
                <sup class="affilcode">1</sup>
                <sup class="affilcode">3</sup>
                <sup class="affilcode">2</sup>
                </span><span class="header--author">
                <a href="https://orcid.org/0000-0003-4036-977X" target="_blank">
                    Ceres Barros
                </a>
                <sup class="affilcode">4</sup>
                </span><span class="header--author">
                <a href="https://orcid.org/0000-0002-2151-6693" target="_blank">
                    Dominique Caron
                </a>
                <sup class="affilcode">5</sup>
                <sup class="affilcode">2</sup>
                </span><span class="header--author">
                <a href="https://orcid.org/0000-0003-0452-6993" target="_blank">
                    Maxwell J Farrell
                </a>
                <sup class="affilcode">6</sup>
                </span><span class="header--author">
                <a href="https://orcid.org/0000-0002-9935-1366" target="_blank">
                    Marie-Jos√©e Fortin
                </a>
                <sup class="affilcode">6</sup>
                </span><span class="header--author">
                <a href="https://orcid.org/0000-0003-3220-6161" target="_blank">
                    Victoria Hemming
                </a>
                <sup class="affilcode">7</sup>
                </span><span class="header--author">
                <a href="https://orcid.org/0000-0002-4104-9463" target="_blank">
                    Benjamin Mercier
                </a>
                <sup class="affilcode">3</sup>
                <sup class="affilcode">2</sup>
                </span><span class="header--author">
                <a href="https://orcid.org/0000-0002-6004-4027" target="_blank">
                    Laura J. Pollock
                </a>
                <sup class="affilcode">5</sup>
                <sup class="affilcode">2</sup>
                </span><span class="header--author">
                <a href="https://orcid.org/0000-0001-5785-8321" target="_blank">
                    Rogini Runghen
                </a>
                <sup class="affilcode">8</sup>
                </span><span class="header--author">
                <a href="https://orcid.org/0000-0002-3454-0633" target="_blank">
                    Giulio V. Dalla Riva
                </a>
                <sup class="affilcode">9</sup>
                </span><span class="header--author">
                <a href="https://orcid.org/0000-0002-0735-5184" target="_blank">
                    Timoth√©e Poisot
                </a>
                <sup class="affilcode">1</sup>
                <sup class="affilcode">2</sup>
                </span>
            </div>
        </div>
    </header>
    
    <main class="container">
        <section class="abstract">
            
            <ol>
                <li>Despite their importance in many ecological processes, collecting data and information on ecological interactions is an exceedingly challenging task. For this reason, large parts of the world have a data deficit when it comes to species interactions, and how the resulting networks are structured. As data collection alone is unlikely to be sufficient, community ecologists must adopt predictive methods.</li>
                <li>We present a methodological framework that uses graph embedding and transfer learning to assemble a predicted list of trophic interactions of a species pool for which their interactions are unknown. Specifically, we ‚Äòlearn‚Äô the information from a known interaction network by inferring the latent traits of species and infer the latent traits of a species pool for which we have no <em>a priori</em> interaction data based on their phylogenetic relatedness to species from the known network. The latent traits can then be used to predict interactions and construct an interaction network.</li>
                <li>Here we assembled a metaweb for Canadian mammals derived from interactions in the European food web, despite only 4% of common species being shared between the two locations. The results of the predictive model are compared against databases of recorded pairwise interactions, showing that we correctly recover 91% of known interactions.</li>
                <li>The framework itself is robust even when the known network is incomplete or contains spurious interactions making it an ideal candidate as a tool for filling gaps when it comes to species interactions. We provide guidance on how this framework can be adapted by substituting some approaches or predictors in order to make it more generally applicable.</li>
            </ol>

            <div class="artifacts">
                <a href="draft.pdf" title="Draft PDF" target="_blank">draft</a>
                <a href="preprint.pdf" title="Preprint PDF" target="_blank">preprint</a>
            </div>
        </section>
        
        <article>
            <h1 id="introduction">Introduction</h1>
            <p>There are two core challenges we are faced with in furthering our understanding of ecological networks across space, particularly at macro-ecologically relevant scales <span class="citation" data-cites="Trojelsgaard2016EcoNet">(<em>e.g.</em> Tr√∏jelsgaard &amp; Olesen, 2016)</span>. First, ecological networks within a location are difficult to sample properly <span class="citation" data-cites="Jordano2016ChaEco Jordano2016SamNet">(Jordano, 2016a, 2016b)</span>, resulting in a widespread ‚ÄúEltonian shortfall‚Äù <span class="citation" data-cites="Hortal2015SevSho">(Hortal et al., 2015)</span>, <em>i.e.</em> a lack of knowledge about inter- and intra- specific relationships. This first challenge has been, in large part, addressed by the recent emergence of a suite of methods aiming to predict interactions within <em>existing</em> networks, many of which are reviewed in <span class="citation" data-cites="Strydom2021RoaPre">Strydom, Catchen, et al. (2021)</span>. Second, recent analyses based on collected data <span class="citation" data-cites="Poisot2021GloKno">(Poisot, Bergeron, et al., 2021)</span> or metadata <span class="citation" data-cites="Cameron2019UneGlo">(Cameron et al., 2019)</span> highlight that ecological networks are currently studied in a biased subset of space and bioclimates, which impedes our ability to generalize any local understanding of network structure. Meaning that, although the framework to address incompleteness <em>within</em> networks exists, there would still be regions for which, due to a <em>lack</em> of local interaction data, we are unable to infer potential species interactions.</p>
            <p>Here, we present a general method to infer potential trophic interactions, relying on the transfer learning of network representations, specifically by using similarities of species in a biologically/ecologically relevant proxy space (<em>e.g.</em> shared morphology or ancestry). Transfer learning is a machine learning methodology that uses the knowledge gained from solving one problem and applying it to a related (destination) problem <span class="citation" data-cites="Torrey2010TraLea Pan2010SurTra">(Pan &amp; Yang, 2010; Torrey &amp; Shavlik, 2010)</span>. In this instance, we solve the problem of predicting trophic interactions between species, based on knowledge extracted from another species pool for which interactions are known by using phylogenetic structure as a medium for transfer. There is a plurality of measures of species similarities that can be used for inferring <em>potential</em> species interactions <em>i.e.</em> metaweb reconstruction <span class="citation" data-cites="Morales-Castilla2015InfBio">(see <em>e.g.</em> Morales-Castilla et al., 2015)</span>; however, phylogenetic proximity has several desirable properties when working at large scales. <span class="citation" data-cites="Gerhold2015PhyPat">Gerhold et al. (2015)</span> made the point that phylogenetic signal captures diversification of characters (large macro-evolutionary process), but not necessarily community assembly (fine ecological process); <span class="citation" data-cites="Dormann2010EvoCli">Dormann et al. (2010)</span> previously found very similar conclusions. Interactions tend to reflect a phylogenetic signal because they have a conserved pattern of evolutionary convergence that encompasses a wide range of ecological and evolutionary mechanisms <span class="citation" data-cites="Mouquet2012EcoAdv Cavender-Bares2009MerCom">(Cavender-Bares et al., 2009; Mouquet et al., 2012)</span>, and - most importantly - retain this signal even if it is obscured at the community scale due to <em>e.g.</em> local conditions <span class="citation" data-cites="Poisot2018IntRet Hutchinson2017CopSig">(Hutchinson et al., 2017; Poisot &amp; Stouffer, 2018)</span>. Finally, species interactions at macro-ecological scales seem to respond mostly to macro-evolutionary processes <span class="citation" data-cites="Price2003MacThe">(Price, 2003)</span>; which is evidenced by the presence of conserved backbones in food webs <span class="citation" data-cites="Mora2018IdeCom DallaRiva2016ExpEvo">(Dalla Riva &amp; Stouffer, 2016; Mora et al., 2018)</span>, strong evolutionary signature on prey choice <span class="citation" data-cites="Stouffer2012EvoCon">(Stouffer et al., 2012)</span>, and strong phylogenetic signature in food web intervality <span class="citation" data-cites="Eklof2016PhyCom">(Ekl√∂f &amp; Stouffer, 2016)</span>. Phylogenetic reconstruction has also previously been used within the context of ecological networks, namely understanding ancestral plant-insect interactions <span class="citation" data-cites="Braga2021PhyRec">(Braga et al., 2021)</span>. Taken together, these considerations suggest that phylogenies can reliably be used to transfer knowledge on species interactions.</p>
            <figure>
            <img src="figures/figure-concept_v2.png" id="fig:concept" alt="Figure 1: Overview of the phylogenetic transfer learning (and prediction) of species interactions networks. Starting from an initial, known, network, we learn its representation through a graph embedding step (here, a truncated Singular Value Decomposition; Step 1), yielding a series of latent traits (latent vulnerability traits are more representative of species at the lower trophic-level and latent generality traits are more representative of species at higher trophic-levels; sensu Schoener (1989)); second, for the destination species pool, we perform ancestral character estimation using a phylogeny (here, using a Brownian model for the latent traits; Step 2); we then sample from the reconstructed distribution of latent traits (Step 3) to generate a probabilistic metaweb at the destination (here, assuming a uniform distribution of traits), and threshold it to yield the final list of interactions (Step 4)." /><figcaption aria-hidden="true">Figure 1: Overview of the phylogenetic transfer learning (and prediction) of species interactions networks. Starting from an initial, known, network, we learn its representation through a graph embedding step (here, a truncated Singular Value Decomposition; Step 1), yielding a series of latent traits (latent vulnerability traits are more representative of species at the lower trophic-level and latent generality traits are more representative of species at higher trophic-levels; <em>sensu</em> <span class="citation" data-cites="Schoener1989FooWeb">Schoener (1989)</span>); second, for the destination species pool, we perform ancestral character estimation using a phylogeny (here, using a Brownian model for the latent traits; Step 2); we then sample from the reconstructed distribution of latent traits (Step 3) to generate a probabilistic metaweb at the destination (here, assuming a uniform distribution of traits), and threshold it to yield the final list of interactions (Step 4).</figcaption>
            </figure>
            <p>In fig.¬†1, we provide a methodological overview based on learning the embedding of a metaweb of trophic interactions for European mammals <span class="citation" data-cites="Maiorano2020Tet10 Maiorano2020DatTet">(known interactions; Maiorano et al., 2020a, 2020b)</span> and, based on phylogenetic relationships between mammals globally <span class="citation" data-cites="Upham2019InfMam">(<em>i.e.</em>, phylogenetic tree Upham et al., 2019)</span>, infer a metaweb for the Canadian mammalian species pool (using only a species list <em>i.e.</em> interactions are treated as unknown in this instance). Our case study shows that phylogenetic transfer learning is an effective approach to the generation of probabilistic metawebs. This showcases that although the components (species) that make up the Canadian and European communities may be <em>minimally</em> shared (the overall species overlap is less than 4%), if the medium (proxy space) selected in the transfer step is biologically plausible, we can still effectively learn from the known network and make biologically relevant predictions of interactions. Indeed, as we detail in the results, when validated against known but fractional data of trophic interactions between Canadian mammals, our model achieves a predictive accuracy of approximately 91%.</p>
            <h1 id="method-description">Method description</h1>
            <p>The core point of our method is the transfer of knowledge of a known ecological network to predict interactions between species for another location for which the network is unknown (or partially known) and is summarized in the grey text boxes in fig.¬†1. The method we develop is, ecologically speaking, a ‚Äúblack box,‚Äù <em>i.e.</em> an algorithm that can be understood mathematically, but whose component parts are not always directly tied to ecological processes. There is a growing realization in machine learning that (unintentional) black box algorithms are not necessarily a bad thing <span class="citation" data-cites="Holm2019DefBla">(Holm, 2019)</span>, as long as their constituent parts can be examined (which is the case with our method). But more importantly, data hold more information than we might think; as such, even algorithms that are disconnected from a model can make correct guesses most of the time <span class="citation" data-cites="Halevy2009UnrEff">(Halevy et al., 2009)</span>; in fact, in an instance of ecological forecasting of spatio-temporal systems, model-free approaches (<em>i.e.</em> drawing all of their information from the data) outperformed model-informed ones <span class="citation" data-cites="Perretti2013ModFor">(Perretti et al., 2013)</span>.</p>
            <h2 id="data-used-for-the-case-study">Data used for the case study</h2>
            <p>We use data from the European metaweb assembled by <span class="citation" data-cites="Maiorano2020Tet10">Maiorano et al. (2020a)</span>. This was assembled using data extracted from scientific literature (including published papers, books, and grey literature) from the last 50 years and includes all terrestrial tetrapods (mammals, breeding birds, reptiles and amphibians) occurring on the European sub-continent (and Turkey) - with the caveat that only species introduced in historical times and currently naturalized being included. The European metaweb was filtered using the Global Biodiversity Information Facility (GBIF) taxonomic backbone <span class="citation" data-cites="GBIFSecretariat2021GbiBac">(GBIF Secretariat, 2021)</span> so as to contain only terrestrial and semi-aquatic mammals. As all species had valid matches to the GBIF taxonomy it was used as the backbone for the remaining reconciliation steps namely, the mammalian consensus supertree by <span class="citation" data-cites="Upham2019InfMam">Upham et al. (2019)</span> (which is used for the knowledge transfer step) and for the Canadian species list‚Äîwhich was extracted from the International Union for Conservation of Nature (IUCN) checklist, and corresponds to the same selection criteria that was applied by <span class="citation" data-cites="Maiorano2020Tet10">Maiorano et al. (2020a)</span> in the European metaweb. After taxonomic cleaning and reconciliation the European metaweb has 260 species, and the Canadian species pool 163; of these, 17 (about 4% of the total) are shared, and 89 species from Canada (54%) had at least one congeneric species in Europe. The similarity for both species pools predictably increases with higher taxonomic order, with 19% of shared genera, 47% of shared families, and 75% of shared orders; for the last point, Canada and Europe each had a single unique order (<em>Didelphimorphia</em> for Canada, <em>Erinaceomorpha</em> for Europe).</p>
            <h2 id="implementation-and-code-availability">Implementation and code availability</h2>
            <p>The entire pipeline is implemented in <em>Julia</em> 1.6 <span class="citation" data-cites="Bezanson2017JulFre">(Bezanson et al., 2017)</span> and is available under the permissive MIT License at <a href="https://osf.io/2zwqm/"><code>https://osf.io/2zwqm/</code></a>. The taxonomic cleanup steps are done using <code>GBIF.jl</code> <span class="citation" data-cites="Dansereau2021SimJl">(Dansereau &amp; Poisot, 2021)</span>. The network embedding and analysis is done using <code>EcologicalNetworks.jl</code> <span class="citation" data-cites="Banville2021ManJl Poisot2019EcoJl">(Banville et al., 2021; Poisot et al., 2019)</span>. The phylogenetic simulations are done using <code>PhyloNetworks.jl</code> <span class="citation" data-cites="Solis-Lemus2017PhyPac">(Sol√≠s-Lemus et al., 2017)</span> and <code>Phylo.jl</code> <span class="citation" data-cites="Reeve2016HowPar">(Reeve et al., 2016)</span>. A complete <code>Project.toml</code> file specifying the full tree of dependencies is available alongside the code. This material also includes a fully annotated copy of the entire code required to run this project (describing both the intent of the code and discussing some technical implementation details), a vignette for every step of the process, and a series of Jupyter notebooks with the text and code. The pipeline can be executed on a laptop in a matter of minutes, and therefore does not require extensive computational power.</p>
            <h2 id="step-1-learning-the-origin-network-representation">Step 1: Learning the origin network representation</h2>
            <p>The first step in transfer learning is to learn the structure of the original dataset. In order to do so, we rely on an approach inspired from representational learning, where we learn a <em>representation</em> of the metaweb (in the form of the latent subspaces), rather than a list of interactions (species <em>a</em> eats <em>b</em>). This approach is conceptually different from other metaweb-scale predictions <span class="citation" data-cites="Albouy2019MarFis">(<em>e.g.</em> Albouy et al., 2019)</span>, in that the metaweb representation is easily transferable. Specifically, we use a Random Dot Product Graph model <span class="citation" data-cites="Young2007RanDot">(hereafter RDPG; S. J. Young &amp; Scheinerman, 2007)</span> to create a number of latent variables that can be combined into an approximation of the network adjacency matrix. RDPG is known to capture the evolutionary backbone of food webs <span class="citation" data-cites="DallaRiva2016ExpEvo">(Dalla Riva &amp; Stouffer, 2016)</span>, resulting in strong phylogenetic signal in RDPG results; in other words, the latent variables of an RDPG can be mapped onto a phylogenetic tree, and phylogenetically similar predators should share phylogenetically similar preys. In addition, recent advances show that the latent variables produced this way can be used to predict <em>de novo</em> interactions. Interestingly, the latent variables do not need to be produced by decomposing the network itself; in a recent contribution, <span class="citation" data-cites="Runghen2021ExpNod">Runghen et al. (2021)</span> showed that deep artificial neural networks are able to reconstruct the left and right subspaces of an RDPG, in order to predict human movement networks from individual/location metadata and opens up the possibility of using additional metadata as predictors.</p>
            <p>The latent variables are created by performing a truncated Singular Value Decomposition <span class="citation" data-cites="Halko2011FinStr">(t-SVD; Halko et al., 2011)</span> on the adjacency matrix. SVD is an appropriate embedding of ecological networks, which has recently been shown to both capture their complex, emerging properties <span class="citation" data-cites="Strydom2021SvdEnt">(Strydom, Dalla Riva, et al., 2021)</span> and to allow highly accurate prediction of the interactions within a single network <span class="citation" data-cites="Poisot2021ImpMam">(Poisot, Ouellet, et al., 2021)</span>. Under SVD, an adjacency matrix <span class="math inline"><strong>A</strong></span> (where <span class="math inline"><strong>A</strong><sub><em>m</em>,‚ÄÜ<em>n</em></sub>‚ÄÑ‚àà‚ÄÑùîπ</span> where 1 indicates predation and 0 an absence thereof) is decomposed into three components resulting in <span class="math inline"><strong>A</strong>‚ÄÑ=‚ÄÑ<strong>U</strong><strong>Œ£</strong><strong>V</strong><strong>‚Ä≤</strong>.</span> Here, <span class="math inline"><strong>Œ£</strong></span> is a <span class="math inline"><em>m</em>‚ÄÖ√ó‚ÄÖ<em>n</em></span> diagonal matrix and contains only singular (<span class="math inline"><em>œÉ</em></span>) values along its diagonal, <span class="math inline"><strong>U</strong></span> is a <span class="math inline"><em>m</em>‚ÄÖ√ó‚ÄÖ<em>m</em></span> unitary matrix, and <span class="math inline"><strong>V</strong>‚Ä≤</span> a <span class="math inline"><em>n</em>‚ÄÖ√ó‚ÄÖ<em>n</em></span> unitary matrix. Truncating the SVD removes additional noise in the dataset by omitting non-zero and/or smaller <span class="math inline"><em>œÉ</em></span> values from <span class="math inline"><strong>Œ£</strong></span> using the rank of the matrix. Under a t-SVD <span class="math inline"><strong>A</strong><sub><em>m</em>,‚ÄÜ<em>n</em></sub></span> is decomposed so that <span class="math inline"><strong>Œ£</strong></span> is a square <span class="math inline"><em>r</em>‚ÄÖ√ó‚ÄÖ<em>r</em></span> diagonal matrix (with <span class="math inline">1‚ÄÑ‚â§‚ÄÑ<em>r</em>‚ÄÑ‚â§‚ÄÑ<em>r</em><sub><em>f</em><em>u</em><em>l</em><em>l</em></sub></span> where <span class="math inline"><em>r</em><sub><em>f</em><em>u</em><em>l</em><em>l</em></sub></span> is the full rank of <span class="math inline"><strong>A</strong></span> and <span class="math inline"><em>r</em></span> the rank at which we truncate the matrix) containing only non-zero <span class="math inline"><em>œÉ</em></span> values. Additionally, <span class="math inline"><strong>U</strong></span> is now a <span class="math inline"><em>m</em>‚ÄÖ√ó‚ÄÖ<em>r</em></span> semi unitary matrix and <span class="math inline"><strong>V</strong>‚Ä≤</span> a <span class="math inline"><em>n</em>‚ÄÖ√ó‚ÄÖ<em>r</em></span> semi-unitary matrix.</p>
            <p>The specific rank at which the SVD ought to be truncated is a difficult question. The purpose of SVD is to remove the noise (expressed at high dimensions) and to focus on the signal, (expressed at low dimensions). In datasets with a clear signal/noise demarcation, a scree plot of <span class="math inline"><strong>Œ£</strong></span> can show a sharp drop at the rank where noise starts <span class="citation" data-cites="Zhu2006AutDim">(Zhu &amp; Ghodsi, 2006)</span>. Because the European metaweb is almost entirely known, the amount of noise (uncertainty) is low; this is reflected in fig.¬†2 (left), where the scree plot shows no important drop, and in fig.¬†2 (right) where the proportion of variance explained increases smoothly at higher dimensions. For this reason, we default back to a threshold that explains 60% of the variance in the underlying data, corresponding to 12 dimensions - <em>i.e.</em> a tradeoff between accuracy and a reduced number of features.</p>
            <p>An RDPG estimates the probability of observing interactions between nodes (species) as a function of the nodes‚Äô latent variables, and is a way to turn an SVD (which decompose one matrix into three) into two matrices that can be multiplied to provide an approximation of the network. The latent variables used for the RDPG, called the left and right subspaces, are defined as <span class="math inline">$\mathscr{L} = \mathbf{U}\sqrt{\mathbf{\Sigma}}$</span>, and <span class="math inline">$\mathscr{R} = \sqrt{\mathbf{\Sigma}}\mathbf{V}'$</span> ‚Äì using the full rank of <span class="math inline"><strong>A</strong></span>, <span class="math inline">‚Ñí‚Ñõ‚ÄÑ=‚ÄÑ<strong>A</strong></span>, and using any smaller rank results in <span class="math inline">‚Ñí‚Ñõ‚ÄÑ‚âà‚ÄÑ<strong>A</strong></span>. Using a rank of 1 for the t-SVD provides a first-order approximation of the network. One advantage of using an RDPG rather than an SVD is that the number of components to estimate decreases; notably, one does not have to estimate the singular values of the SVD. Furthermore, the two subspaces can be directly multiplied to yield a network.</p>
            <figure>
            <img src="figures/figure-screeplot.png" id="fig:scree" alt="Figure 2: Left: representation of the scree plot of the singular values from the t-SVD on the European metaweb. The scree plot shows no obvious drop in the singular values that may be leveraged to automatically detect a minimal dimension for embedding, after e.g. Zhu &amp; Ghodsi (2006). Right: cumulative fraction of variance explained by each dimension up to the rank of the European metaweb. The grey lines represent cutoffs at 50, 60, ‚Ä¶, 90% of variance explained. For the rest of the analysis, we reverted to an arbitrary threshold of 60% of variance explained, which represented a good tradeoff between accuracy and reduced number of features." /><figcaption aria-hidden="true">Figure 2: Left: representation of the scree plot of the singular values from the t-SVD on the European metaweb. The scree plot shows no obvious drop in the singular values that may be leveraged to automatically detect a minimal dimension for embedding, after <em>e.g.</em> <span class="citation" data-cites="Zhu2006AutDim">Zhu &amp; Ghodsi (2006)</span>. Right: cumulative fraction of variance explained by each dimension up to the rank of the European metaweb. The grey lines represent cutoffs at 50, 60, ‚Ä¶, 90% of variance explained. For the rest of the analysis, we reverted to an arbitrary threshold of 60% of variance explained, which represented a good tradeoff between accuracy and reduced number of features.</figcaption>
            </figure>
            <p>Because RDPG relies on matrix multiplication, the higher dimensions essentially serve to make specific interactions converge towards 0 or 1; therefore, for reasonably low ranks, there is no guarantee that the values in the reconstructed network will be within the unit range. In order to determine what constitutes an appropriate threshold for probability, we performed the RDPG approach on the European metaweb, and evaluated the probability threshold by treating this as a binary classification problem, specifically assuming that both 0 and 1 in the European metaweb are all true. Given the methodological details given in <span class="citation" data-cites="Maiorano2020Tet10">Maiorano et al. (2020a)</span> and <span class="citation" data-cites="OConnor2020UnvFoo">O‚ÄôConnor et al. (2020)</span>, this seems like a reasonable assumption, although one that does not hold for all metawebs. We used the thresholding approach presented in <span class="citation" data-cites="Poisot2021ImpMam">Poisot, Ouellet, et al. (2021)</span>, and picked a cutoff that maximized Youden‚Äôs <span class="math inline"><em>J</em></span> statistic (a measure of the informedness (trust) of predictions; <span class="citation" data-cites="Youden1950IndRat">Youden (1950)</span>); the resulting cutoff was 0.22, and gave an accuracy above 0.99. In Supp. Mat. 1, we provide several lines of evidence that using the entire network to estimate the threshold does not lead to overfitting; that using a subset of species would yield the same threshold; that decreasing the quality of the original data by adding or removing interactions would minimally affect the predictive accuracy of RDPG applied to the European metaweb; and that the networks reconstructed from artificially modified data are reconstructed with the correct ecological properties.</p>
            <p>The left and right subspaces for the European metaweb, accompanied by the threshold for prediction, represent the knowledge we seek to transfer. In the next section, we explain how we rely on phylogenetic similarity to do so.</p>
            <h2 id="steps-2-and-3-transfer-learning-through-phylogenetic-relatedness">Steps 2 and 3: Transfer learning through phylogenetic relatedness</h2>
            <p>In order to transfer the knowledge from the European metaweb to the Canadian species pool, we performed ancestral character estimation using a Brownian motion model, which is a conservative approach in the absence of strong hypotheses about the nature of phylogenetic signal in the network decomposition <span class="citation" data-cites="Litsios2012EffPhy">(Litsios &amp; Salamin, 2012)</span>. This uses the estimated feature vectors for the European mammals to create a state reconstruction for all species (conceptually something akin to a trait-based mammalian phylogeny using latent generality and vulnerability traits) and allows us to impute the missing (latent) trait data for the Canadian species that are not already in the European network; as we are focused on predicting contemporary interactions, we only retained the values for the tips of the tree. We assumed that all traits (<em>i.e.</em> the feature vectors for the left and right subspaces) were independent, which is a reasonable assumption as every trait/dimension added to the t-SVD has an <em>additive</em> effect to the one before it. Note that the <span class="citation" data-cites="Upham2019InfMam">Upham et al. (2019)</span> tree itself has some uncertainty associated to inner nodes of the phylogeny. In this case study we have decided to not propagate this uncertainty as it would complexify the process. The Brownian motion algorithm returns the <em>average</em> value of the trait, and its upper and lower bounds. Because we do not estimate other parameters of the traits‚Äô distributions, we considered that every species trait is represented as a uniform distribution between these bounds. The choice of the uniform distribution was made because the algorithm returns a minimum and maximum point estimate for the value, and given this information, the uniform distribution is the one with maximum entropy. Had all mean parameters estimates been positive, the exponential distribution would have been an alternative, but this is not the case for the subspaces of an RDPG. In order to examine the consequences of the choice of distribution, we estimated the variance per latent variable per node to use a Normal distribution; as we show in Supp. Mat. 2, this decision results in dramatically over-estimating the number and probability of interactions, and therefore we keep the discussions in the main text to the uniform case. The inferred left and right subspaces for the Canadian species pool (<span class="math inline">$\hat{\mathscr{L}}$</span> and <span class="math inline">$\hat{\mathscr{R}}$</span>) have entries that are distributions, representing the range of values for a given species at a given dimension. These objects represent the transferred knowledge, which we can use for prediction of the Canadian metaweb.</p>
            <h2 id="step-4-probabilistic-prediction-of-the-destination-network">Step 4: Probabilistic prediction of the destination network</h2>
            <p>The phylogenetic reconstruction of <span class="math inline">$\hat{\mathscr{L}}$</span> and <span class="math inline">$\hat{\mathscr{R}}$</span> has an associated uncertainty, represented by the breadth of the uniform distribution associated to each of their entries. Therefore, we can use this information to assemble a <em>probabilistic</em> metaweb in the sense of <span class="citation" data-cites="Poisot2016StrPro">Poisot et al. (2016)</span>, <em>i.e.</em> in which every interaction is represented as a single, independent, Bernoulli event of probability <span class="math inline"><em>p</em></span>.</p>
            <figure>
            <img src="figures/figure-subspaces.png" id="fig:subspaces" alt="Figure 3: Visual representation of the left (green/purple) and right (green/brown) subspaces, alongside the adjacency matrix of the food web they encode (greyscale). The European metaweb is on the left, and the imputed Canadian metaweb (before data inflation) on the right. This figure illustrates how much structure the left subspace captures. As we show in fig.¬†6, the species with a value of 0 in the left subspace are species without any prey." /><figcaption aria-hidden="true">Figure 3: Visual representation of the left (green/purple) and right (green/brown) subspaces, alongside the adjacency matrix of the food web they encode (greyscale). The European metaweb is on the left, and the imputed Canadian metaweb (before data inflation) on the right. This figure illustrates how much structure the left subspace captures. As we show in fig.¬†6, the species with a value of 0 in the left subspace are species without any prey.</figcaption>
            </figure>
            <p>Specifically, we have adopted the following approach. For every entry in <span class="math inline">$\hat{\mathscr{L}}$</span> and <span class="math inline">$\hat{\mathscr{R}}$</span>, we draw a value from its distribution. This results in one instance of the possible left (<span class="math inline">$\hat{\mathscr{l}}$</span>) and right (<span class="math inline">$\hat{\mathscr{r}}$</span>) subspaces for the Canadian metaweb. These can be multiplied, to produce one matrix of real values. Because the entries in <span class="math inline">$\hat{\mathscr{l}}$</span> and <span class="math inline">$\hat{\mathscr{r}}$</span> are in the same space where <span class="math inline">‚Ñí</span> and <span class="math inline">‚Ñõ</span> were originally predicted, it follows that the threshold <span class="math inline"><em>œÅ</em></span> estimated for the European metaweb also applies. We use this information to produce one random Canadian metaweb, <span class="math inline">$N = \hat{\mathscr{L}}\hat{\mathscr{R}}' \ge \rho$</span>. As we can see in (fig.¬†3), the European and Canadian metawebs are structurally similar (as would be expected given the biogeographic similarities) and the two (left and right) subspaces are distinct <em>i.e.</em> capturing predation (generality) and prey (vulnerability) latent traits.</p>
            <p>Because the intervals around some trait values can be broad <span class="citation" data-cites="Garland1999IntPhy">(in fact, probably broader than what they would actually be, see <em>e.g.</em> Garland et al., 1999)</span>, we repeat the above process <span class="math inline">2‚ÄÖ√ó‚ÄÖ10<sup>5</sup></span> times, which results in a probabilistic metaweb <span class="math inline"><em>P</em></span>, where the probability of an interaction (here conveying our degree of trust that it exists given the inferred trait distributions) is given by the number of times where it appears across all random draws <span class="math inline"><em>N</em></span>, divided by the number of samples. An interaction with <span class="math inline"><em>P</em><sub><em>i</em>,‚ÄÜ<em>j</em></sub>‚ÄÑ=‚ÄÑ1</span> means that these two species were predicted to interact in all <span class="math inline">2‚ÄÖ√ó‚ÄÖ10<sup>5</sup></span> random draws.</p>
            <p>It must be noted that despite bringing in a large amount of information from the European species pool and interactions, the Canadian metaweb has distinct structural properties. Following an approach similar to <span class="citation" data-cites="Vermaat2009MajDim">Vermaat et al. (2009)</span>, we show in Supp. Mat. 3 that not only can we observe differences in a multivariate space between the European and Canadian metaweb, we can also observe differences in the same space between random subgraphs from these networks. These results line up with the studies spatializing metawebs that have been discussed in the introduction: changes in the species pool are driving local structural changes in the networks.</p>
            <h2 id="data-cleanup-discovery-validation-and-thresholding">Data cleanup, discovery, validation, and thresholding</h2>
            <p>Once the probabilistic metaweb for Canada has been produced, we followed a number of data inflation steps to finalize it. This step is external to the actual transfer learning framework but rather serves as a way to augment and validate the predicted metaweb.</p>
            <figure>
            <img src="figures/figure-validation.png" id="fig:inflation" alt="Figure 4: Left, comparison of the probabilities of interactions assigned by the model to all interactions (grey curve), the subset of interactions found in GLOBI (red), and in the Strong &amp; Leroux (2014) Newfoundland dataset (blue). The model recovers more interactions with a low probability compared to data mining, which can suggest that collected datasets are biased towards more common or easy to identify interactions. Right, distribution of the in-degree and out-degree of the mammals from Canada in the reconstructed metaweb. This figure describes a flat, relatively short food web, in which there are few predators but a large number of preys." /><figcaption aria-hidden="true">Figure 4: Left, comparison of the probabilities of interactions assigned by the model to all interactions (grey curve), the subset of interactions found in GLOBI (red), and in the <span class="citation" data-cites="Strong2014ImpNon">Strong &amp; Leroux (2014)</span> Newfoundland dataset (blue). The model recovers more interactions with a low probability compared to data mining, which can suggest that collected datasets are biased towards more common or easy to identify interactions. Right, distribution of the in-degree and out-degree of the mammals from Canada in the reconstructed metaweb. This figure describes a flat, relatively short food web, in which there are few predators but a large number of preys.</figcaption>
            </figure>
            <p>First, we extracted the network corresponding to the 17 species shared between the European and Canadian pools and replaced these interactions with a probability of 0 (non-interaction) or 1 (interaction), according to their value in the European metaweb. This represents a minute modification of the inferred network (about 0.8% of all species pairs from the Canadian web), but ensures that we are directly re-using knowledge from Europe.</p>
            <p>Second, we looked for all species in the Canadian pool known to the Global Biotic Interactions (GLoBI) database <span class="citation" data-cites="Poelen2014GloBio">(Poelen et al., 2014)</span>, and extracted their known interactions. Because GLoBI aggregates observed interactions, it is not a <em>networks</em> data source, and therefore the only information we can reliably extract from it is that a species pair <em>was reported to interact at least once</em>. This last statement should yet be taken with caution, as some sources in GLoBI <span class="citation" data-cites="Thessen2014KnoExt">(<em>e.g.</em> Thessen &amp; Parr, 2014)</span> are produced through text analysis, and therefore may not document direct evidence of the interaction. Nevertheless, should the predictive model work, we would expect that a majority of interactions known to GLoBI would also be predicted. We retrieved 366 interactions between mammals from the Canadian species pool from GLoBI, 33 of which were not predicted by the model; this results in a success rate of 91%. After performing this check, we set the probability of all interactions known to GLoBI to 1.</p>
            <p>Finally, we downloaded the data from <span class="citation" data-cites="Strong2014ImpNon">Strong &amp; Leroux (2014)</span>, who mined various literature sources to identify trophic interactions in Newfoundland. This dataset documented 25 interactions between mammals, only two of which were not part of our (Canada-level) predictions, resulting in a success rate of 92%. These two interactions were added to our predicted metaweb with a probability of 1. A table listing all interactions in the predicted Canadian metaweb can be found in the supplementary material.</p>
            <figure>
            <img src="figures/figure-cutoffs.png" id="fig:thresholds" alt="Figure 5: Left: effect of varying the cutoff for probabilities to be considered non-zero on the number of unique links and on \hat{L}, the probabilistic estimate of the number of links assuming that all interactions are independent. Right: effect of varying the cutoff on the number of disconnected species, and on network connectance. In both panels, the grey line indicates the cutoff P(i\rightarrow j) \approx 0.08 that resulted in the first species losing all of its interactions." /><figcaption aria-hidden="true">Figure 5: Left: effect of varying the cutoff for probabilities to be considered non-zero on the number of unique links and on <span class="math inline"><em>LÃÇ</em></span>, the probabilistic estimate of the number of links assuming that all interactions are independent. Right: effect of varying the cutoff on the number of disconnected species, and on network connectance. In both panels, the grey line indicates the cutoff <span class="math inline"><em>P</em>(<em>i</em>‚ÄÑ‚Üí‚ÄÑ<em>j</em>)‚ÄÑ‚âà‚ÄÑ0.08</span> that resulted in the first species losing all of its interactions.</figcaption>
            </figure>
            <p>Because the confidence intervals on the inferred trait space are probably over-estimates, we decided to apply a thresholding step to the interactions after data inflation (see fig.¬†5 showing the effect of varying the cutoff on <span class="math inline"><em>P</em>(<em>i</em>‚ÄÑ‚Üí‚ÄÑ<em>j</em>)</span>). <span class="citation" data-cites="Cirtwill2021BuiFoo">Cirtwill &amp; Hamb√§ck (2021)</span> proposed a number of strategies to threshold probabilistic networks. Their methods assume the underlying data to be tag-based sequencing, which represents interactions as co-occurrences of predator and prey within the same tags; this is conceptually identical to our Bernoulli-trial based reconstruction of a probabilistic network. We performed a full analysis of the effect of various cutoffs, and as they either resulted in removing too few interactions, or removing enough interactions that species started to be disconnected from the network, we set this threshold for a probability equivalent to 0 to the largest possible value that still allowed all species to have at least one interaction with a non-zero probability. The need for this slight deviation from the <span class="citation" data-cites="Cirtwill2021BuiFoo">Cirtwill &amp; Hamb√§ck (2021)</span> method highlights the need for additional development on network thresholding.</p>
            <h1 id="results-and-discussion-of-the-case-study">Results and discussion of the case study</h1>
            <figure>
            <img src="figures/figure-degree.png" id="fig:degree" alt="Figure 6: Top: biological significance of the first dimension. Left: there is a linear relationship between the values on the first dimension of the left subspace and the generality, i.e. the relative number of preys, sensu Schoener (1989). Species with a value of 0 in this subspace are at the bottom-most trophic level. Right: there is, similarly, a linear relationship between the position of a species on the first dimension of the right subspace and its vulnerability, i.e. the relative number of predators. Taken together, these two figures show that the first-order representation of this network would capture its degree distribution. Bottom: topological consequences of the first dimension. Left: differences in the z-score of the actual configuration model for the reconstructed network, and the prediction based only on the first dimension. Right: distribution of the differences in the left panel." /><figcaption aria-hidden="true">Figure 6: Top: biological significance of the first dimension. Left: there is a linear relationship between the values on the first dimension of the left subspace and the generality, <em>i.e.</em> the relative number of preys, <em>sensu</em> <span class="citation" data-cites="Schoener1989FooWeb">Schoener (1989)</span>. Species with a value of 0 in this subspace are at the bottom-most trophic level. Right: there is, similarly, a linear relationship between the position of a species on the first dimension of the right subspace and its vulnerability, <em>i.e.</em> the relative number of predators. Taken together, these two figures show that the first-order representation of this network would capture its degree distribution. Bottom: topological consequences of the first dimension. Left: differences in the <span class="math inline"><em>z</em></span>-score of the actual configuration model for the reconstructed network, and the prediction based only on the first dimension. Right: distribution of the differences in the left panel.</figcaption>
            </figure>
            <p>Using a transfer learning framework we were able to construct a probabilistic metaweb and <span class="citation" data-cites="Dunne2006NetStr">(as as per Dunne, 2006)</span> it is a list of potential interactions and does not mean that they will exist wherever the two species co-occur. The t-SVD embedding is able to learn relevant ecological features for the network. fig.¬†6 shows that the first rank correlates linearly with generality and vulnerability <span class="citation" data-cites="Schoener1989FooWeb">(Schoener, 1989)</span>, <em>i.e.</em> the number of preys and predators for each species. Importantly, this implies that a rank 1 approximation represents the configuration model for the metaweb, <em>i.e.</em> a set of random networks generated from a given degree sequence <span class="citation" data-cites="Park2004StaMec">(Park &amp; Newman, 2004)</span>. Accounting for the probabilistic nature of the degrees, the rank 1 approximation also represents the <em>soft</em> configuration model <span class="citation" data-cites="vanderHoorn2018SpaMax">(van der Hoorn et al., 2018)</span>. Both models are maximum entropy graph models <span class="citation" data-cites="Garlaschelli2018CovStr">(Garlaschelli et al., 2018)</span>, with sharp (all network realizations satisfy the specified degree sequence) and soft (network realizations satisfy the degree sequence on average) local constraints, respectively. The (soft) configuration model is an unbiased random graph model widely used by ecologists in the context of null hypothesis significance testing of network structure <span class="citation" data-cites="Bascompte2003NesAss">(<em>e.g.</em> Bascompte et al., 2003)</span> and can provide informative priors for Bayesian inference of network structure <span class="citation" data-cites="Young2021BayInf">(<em>e.g.</em> J.-G. Young et al., 2021)</span>. It is noteworthy that for this metaweb, the relevant information was extracted at the first rank. Because the first rank corresponds to the leading singular value of the system, the results of fig.¬†6 have a straightforward interpretation: degree-based processes are the most important in structuring the mammalian food web.</p>
            <h1 id="discussion">Discussion</h1>
            <p>One important aspect in which Europe and Canada differ (despite their comparable bioclimatic conditions) is the degree of the legacy of human impacts, which have been much longer in Europe. <span class="citation" data-cites="Nenzen2014Imp850">Nenz√©n et al. (2014)</span> showed that even at small scales (the Iberian peninsula), mammal food webs retain the signal of both past climate change and human activity, even when this human activity was orders of magnitude less important than it is now. Similarly, <span class="citation" data-cites="Yeakel2014ColEco">Yeakel et al. (2014)</span> showed that changes in human occupation over several centuries can lead to food web collapse. Megafauna in particular seems to be very sensitive to human arrival <span class="citation" data-cites="Pires2015PleMeg">(Pires et al., 2015)</span>. In short, there is well-substantiated support for the idea that human footprint affects more than the risk of species extinction <span class="citation" data-cites="Marco2018ChaHum">(Marco et al., 2018)</span>, and can lead to changes in interaction structure.</p>
            <p><span class="citation" data-cites="Cirtwill2019QuaFra">Cirtwill et al. (2019)</span> showed that network inference techniques based on Bayesian approaches would perform far better in the presence of an interaction-level informative prior; the desirable properties of such a prior would be that it is expressed as a probability, preferably representing a Bernoulli event, the value of which would be representative of relevant biological processes (probability of predation in this case). We argue that the probability returned at the very last step of our framework may serve as this informative prior; indeed, the output of our analysis can be used in subsequent steps, also possibly involving expert elicitation to validate some of the most strongly recommended interactions. One important <em>caveat</em> to keep in mind when working with interaction inference is that interactions can never really be true negatives (in the current state of our methodological framework and data collection limitations); this renders the task of validating a model through the usual application of binary classification statistics very difficult <span class="citation" data-cites="Strydom2021RoaPre">(although see Strydom, Catchen, et al., 2021 for a discussion of alternative suggestions)</span>. The other way through which our framework can be improved is by substituting the predictors that are used for transfer. For example, in the presence of information on species traits that are known to be predictive of species interactions, one might want to rely on functional rather than phylogenetic distances ‚Äì in food webs, body size (and allometrically related variables) has been established as such a variable <span class="citation" data-cites="Brose2006ConRes">(Brose et al., 2006)</span>; the identification of relevant functional traits is facilitated by recent methodological developments <span class="citation" data-cites="Rosado2013GoiBac">(Rosado et al., 2013)</span>.</p>
            <p>Finally, it should be noted that the framework we have presented is amenable to changes lending to applicability to a broad range of potential scenarios. For example in this case study we have embedded the original metaweb using t-SVD, because it lends itself to an RDPG reconstruction, which is known to capture the consequences of evolutionary processes <span class="citation" data-cites="DallaRiva2016ExpEvo">(Dalla Riva &amp; Stouffer, 2016)</span>; this being said, there are other ways to embed graphs <span class="citation" data-cites="Arsov2019NetEmb Cai2017ComSur Cao2019NetEmb">(Arsov &amp; Mirceva, 2019; Cai et al., 2017; Cao et al., 2019)</span>, which can be used as alternatives. Regarding the transfer step it is possible to use distinct trees if working with distinct clades (such as pollination networks) or an alternative measure of similarity (transfer medium) such as information on foraging <span class="citation" data-cites="Beckerman2006ForBio">(Beckerman et al., 2006)</span>, cell-level mechanisms <span class="citation" data-cites="Boeckaerts2021PreBac">(Boeckaerts et al., 2021)</span>, or a combination of traits and phylogenetic structure <span class="citation" data-cites="Stock2021PaiLea">(Stock, 2021)</span>. Most importantly, although we focus on a trophic system, it is an established fact that different (non-trophic) interactions do themselves interact with and influence the outcome of trophic interactions <span class="citation" data-cites="Kawatsu2021AreNet Kefi2012MorMea">(see <em>e.g.</em> Kawatsu et al., 2021; K√©fi et al., 2012)</span>. Future development of metaweb inference techniques should cover the prediction of multiple interaction types.</p>
            <p><strong>Acknowledgements:</strong> We acknowledge that this study was conducted on land within the traditional unceded territory of the Saint Lawrence Iroquoian, Anishinabewaki, Mohawk, Huron-Wendat, and Om√†miwininiwak nations. TP, TS, DC, and LP received funding from the Canadian Institue for Ecology &amp; Evolution. FB is funded by the Institute for Data Valorization (IVADO). TS, SB, and TP are funded by a donation from the Courtois Foundation. CB was awarded a Mitacs Elevate Fellowship no. IT12391, in partnership with fRI Research, and also acknowledges funding from Alberta Innovates and the Forest Resources Improvement Association of Alberta. M-JF acknowledges funding from NSERC Discovery Grant and NSERC CRC. RR is funded by New Zealand‚Äôs Biological Heritage NgƒÅ Koiora Tuku Iho National Science Challenge, administered by New Zealand Ministry of Business, Innovation, and Employment. BM is funded by the NSERC Alexander Graham Bell Canada Graduate Scholarship and the FRQNT master‚Äôs scholarship. LP acknowledges funding from NSERC Discovery Grant (NSERC RGPIN-2019-05771). TP acknowledges financial support from NSERC through the Discovery Grants and Discovery Accelerator Supplement programs.</p>
            <p><strong>Conflict of interest:</strong> The authors have no conflict interests to disclose</p>
            <p><strong>Authors‚Äô contributions:</strong> TS, SB, and TP designed the study and performed the analysis; GVDR, MF, and RR provided additional feedback on the analyses. DC, BM, and FB helped with data collection. All authors contributed to writing and editing the manuscript.</p>
            <p><strong>Data availability:</strong> All code and data used in this manuscript is publicly available and archived on OSF <a href="https://osf.io/2zwqm/">https://osf.io/2zwqm/</a> and is currently referenced in the manuscript.</p>
            <h1 class="unnumbered" id="references">References</h1>
            <div id="refs" class="references csl-bib-body hanging-indent" data-line-spacing="2" role="doc-bibliography">
            <div id="ref-Albouy2019MarFis" class="csl-entry" role="doc-biblioentry">
            Albouy, C., Archambault, P., Appeltans, W., Ara√∫jo, M. B., Beauchesne, D., Cazelles, K., Cirtwill, A. R., Fortin, M.-J., Galiana, N., Leroux, S. J., Pellissier, L., Poisot, T., Stouffer, D. B., Wood, S. A., &amp; Gravel, D. (2019). The marine fish food web is globally connected. <em>Nature Ecology &amp; Evolution</em>, <em>3</em>(8, 8), 1153‚Äì1161. <a href="https://doi.org/10.1038/s41559-019-0950-y">https://doi.org/10.1038/s41559-019-0950-y</a>
            </div>
            <div id="ref-Arsov2019NetEmb" class="csl-entry" role="doc-biblioentry">
            Arsov, N., &amp; Mirceva, G. (2019, November 26). <em>Network Embedding: An Overview</em>. <a href="http://arxiv.org/abs/1911.11726">http://arxiv.org/abs/1911.11726</a>
            </div>
            <div id="ref-Banville2021ManJl" class="csl-entry" role="doc-biblioentry">
            Banville, F., Vissault, S., &amp; Poisot, T. (2021). Mangal.jl and EcologicalNetworks.jl: Two complementary packages for analyzing ecological networks in Julia. <em>Journal of Open Source Software</em>, <em>6</em>(61), 2721. <a href="https://doi.org/10.21105/joss.02721">https://doi.org/10.21105/joss.02721</a>
            </div>
            <div id="ref-Bascompte2003NesAss" class="csl-entry" role="doc-biblioentry">
            Bascompte, J., Jordano, P., Melian, C. J., &amp; Olesen, J. M. (2003). The nested assembly of plant-animal mutualistic networks. <em>Proceedings of the National Academy of Sciences</em>, <em>100</em>(16), 9383‚Äì9387. <a href="https://doi.org/10.1073/pnas.1633576100">https://doi.org/10.1073/pnas.1633576100</a>
            </div>
            <div id="ref-Beckerman2006ForBio" class="csl-entry" role="doc-biblioentry">
            Beckerman, A. P., Petchey, O. L., &amp; Warren, P. H. (2006). Foraging biology predicts food web complexity. <em>Proceedings of the National Academy of Sciences</em>, <em>103</em>(37), 13745‚Äì13749. <a href="https://doi.org/10.1073/pnas.0603039103">https://doi.org/10.1073/pnas.0603039103</a>
            </div>
            <div id="ref-Bezanson2017JulFre" class="csl-entry" role="doc-biblioentry">
            Bezanson, J., Edelman, A., Karpinski, S., &amp; Shah, V. (2017). Julia: A Fresh Approach to Numerical Computing. <em>SIAM Review</em>, <em>59</em>(1), 65‚Äì98. <a href="https://doi.org/10.1137/141000671">https://doi.org/10.1137/141000671</a>
            </div>
            <div id="ref-Boeckaerts2021PreBac" class="csl-entry" role="doc-biblioentry">
            Boeckaerts, D., Stock, M., Criel, B., Gerstmans, H., De Baets, B., &amp; Briers, Y. (2021). Predicting bacteriophage hosts based on sequences of annotated receptor-binding proteins. <em>Scientific Reports</em>, <em>11</em>(1, 1), 1467. <a href="https://doi.org/10.1038/s41598-021-81063-4">https://doi.org/10.1038/s41598-021-81063-4</a>
            </div>
            <div id="ref-Braga2021PhyRec" class="csl-entry" role="doc-biblioentry">
            Braga, M. P., Janz, N., Nylin, S., Ronquist, F., &amp; Landis, M. J. (2021). Phylogenetic reconstruction of ancestral ecological networks through time for pierid butterflies and their host plants. <em>Ecology Letters</em>, <em>n/a</em>(n/a). <a href="https://doi.org/10.1111/ele.13842">https://doi.org/10.1111/ele.13842</a>
            </div>
            <div id="ref-Brose2006ConRes" class="csl-entry" role="doc-biblioentry">
            Brose, U., Jonsson, T., Berlow, E. L., Warren, P., Banasek-Richter, C., Bersier, L.-F., Blanchard, J. L., Brey, T., Carpenter, S. R., Blandenier, M.-F. C., Cushing, L., Dawah, H. A., Dell, T., Edwards, F., Harper-Smith, S., Jacob, U., Ledger, M. E., Martinez, N. D., Memmott, J., ‚Ä¶ Cohen, J. E. (2006). ConsumerResource Body-Size Relationships in Natural Food Webs. <em>Ecology</em>, <em>87</em>(10), 2411‚Äì2417. <a href="https://doi.org/10.1890/0012-9658(2006)87[2411:CBRINF]2.0.CO;2">https://doi.org/10.1890/0012-9658(2006)87[2411:CBRINF]2.0.CO;2</a>
            </div>
            <div id="ref-Cai2017ComSur" class="csl-entry" role="doc-biblioentry">
            Cai, H., Zheng, V. W., &amp; Chang, K. C.-C. (2017). <em>A Comprehensive Survey of Graph Embedding: Problems, Techniques and Applications</em>. <a href="http://arxiv.org/abs/1709.07604">http://arxiv.org/abs/1709.07604</a>
            </div>
            <div id="ref-Cameron2019UneGlo" class="csl-entry" role="doc-biblioentry">
            Cameron, E. K., Sundqvist, M. K., Keith, S. A., CaraDonna, P. J., Mousing, E. A., Nilsson, K. A., Metcalfe, D. B., &amp; Classen, A. T. (2019). Uneven global distribution of food web studies under climate change. <em>Ecosphere</em>, <em>10</em>(3), e02645. <a href="https://doi.org/10.1002/ecs2.2645">https://doi.org/10.1002/ecs2.2645</a>
            </div>
            <div id="ref-Cao2019NetEmb" class="csl-entry" role="doc-biblioentry">
            Cao, R.-M., Liu, S.-Y., &amp; Xu, X.-K. (2019). Network embedding for link prediction: The pitfall and improvement. <em>Chaos: An Interdisciplinary Journal of Nonlinear Science</em>, <em>29</em>(10), 103102. <a href="https://doi.org/10.1063/1.5120724">https://doi.org/10.1063/1.5120724</a>
            </div>
            <div id="ref-Cavender-Bares2009MerCom" class="csl-entry" role="doc-biblioentry">
            Cavender-Bares, J., Kozak, K. H., Fine, P. V. A., &amp; Kembel, S. W. (2009). The merging of community ecology and phylogenetic biology. <em>Ecology Letters</em>, <em>12</em>(7), 693‚Äì715. <a href="https://doi.org/10.1111/j.1461-0248.2009.01314.x">https://doi.org/10.1111/j.1461-0248.2009.01314.x</a>
            </div>
            <div id="ref-Cirtwill2019QuaFra" class="csl-entry" role="doc-biblioentry">
            Cirtwill, A. R., Ekl, A., Roslin, T., Wootton, K., &amp; Gravel, D. (2019). A quantitative framework for investigating the reliability of empirical network construction. <em>Methods in Ecology and Evolution</em>, <em>0</em>. <a href="https://doi.org/10.1111/2041-210X.13180">https://doi.org/10.1111/2041-210X.13180</a>
            </div>
            <div id="ref-Cirtwill2021BuiFoo" class="csl-entry" role="doc-biblioentry">
            Cirtwill, A. R., &amp; Hamb√§ck, P. (2021). Building food networks from molecular data: Bayesian or fixed-number thresholds for including links. <em>Basic and Applied Ecology</em>, <em>50</em>, 67‚Äì76. <a href="https://doi.org/10.1016/j.baae.2020.11.007">https://doi.org/10.1016/j.baae.2020.11.007</a>
            </div>
            <div id="ref-DallaRiva2016ExpEvo" class="csl-entry" role="doc-biblioentry">
            Dalla Riva, G. V., &amp; Stouffer, D. B. (2016). Exploring the evolutionary signature of food webs‚Äô backbones using functional traits. <em>Oikos</em>, <em>125</em>(4), 446‚Äì456. <a href="https://doi.org/10.1111/oik.02305">https://doi.org/10.1111/oik.02305</a>
            </div>
            <div id="ref-Dansereau2021SimJl" class="csl-entry" role="doc-biblioentry">
            Dansereau, G., &amp; Poisot, T. (2021). SimpleSDMLayers.jl and GBIF.jl: A Framework for Species Distribution Modeling in Julia. <em>Journal of Open Source Software</em>, <em>6</em>(57), 2872. <a href="https://doi.org/10.21105/joss.02872">https://doi.org/10.21105/joss.02872</a>
            </div>
            <div id="ref-Dormann2010EvoCli" class="csl-entry" role="doc-biblioentry">
            Dormann, C. F., Gruber, B., Winter, M., &amp; Herrmann, D. (2010). Evolution of climate niches in European mammals? <em>Biology Letters</em>, <em>6</em>(2), 229‚Äì232. <a href="https://doi.org/10.1098/rsbl.2009.0688">https://doi.org/10.1098/rsbl.2009.0688</a>
            </div>
            <div id="ref-Dunne2006NetStr" class="csl-entry" role="doc-biblioentry">
            Dunne, J. A. (2006). The Network Structure of Food Webs. In J. A. Dunne &amp; M. Pascual (Eds.), <em>Ecological networks: Linking structure and dynamics</em> (pp. 27‚Äì86). Oxford University Press.
            </div>
            <div id="ref-Eklof2016PhyCom" class="csl-entry" role="doc-biblioentry">
            Ekl√∂f, A., &amp; Stouffer, D. B. (2016). The phylogenetic component of food web structure and intervality. <em>Theoretical Ecology</em>, <em>9</em>(1), 107‚Äì115. <a href="https://doi.org/10.1007/s12080-015-0273-9">https://doi.org/10.1007/s12080-015-0273-9</a>
            </div>
            <div id="ref-Garland1999IntPhy" class="csl-entry" role="doc-biblioentry">
            Garland, T., JR., Midford, P. E., &amp; Ives, A. R. (1999). An Introduction to Phylogenetically Based Statistical Methods, with a New Method for Confidence Intervals on Ancestral Values1. <em>American Zoologist</em>, <em>39</em>(2), 374‚Äì388. <a href="https://doi.org/10.1093/icb/39.2.374">https://doi.org/10.1093/icb/39.2.374</a>
            </div>
            <div id="ref-Garlaschelli2018CovStr" class="csl-entry" role="doc-biblioentry">
            Garlaschelli, D., Hollander, F. den, &amp; Roccaverde, A. (2018). Covariance structure behind breaking of ensemble equivalence in random graphs. <em>Journal of Statistical Physics</em>, <em>173</em>(3-4), 644‚Äì662. <a href="https://doi.org/10.1007/s10955-018-2114-x">https://doi.org/10.1007/s10955-018-2114-x</a>
            </div>
            <div id="ref-GBIFSecretariat2021GbiBac" class="csl-entry" role="doc-biblioentry">
            GBIF Secretariat. (2021). <em>GBIF Backbone Taxonomy</em>. <a href="https://doi.org/10.15468/39omei">https://doi.org/10.15468/39omei</a>
            </div>
            <div id="ref-Gerhold2015PhyPat" class="csl-entry" role="doc-biblioentry">
            Gerhold, P., Cahill, J. F., Winter, M., Bartish, I. V., &amp; Prinzing, A. (2015). Phylogenetic patterns are not proxies of community assembly mechanisms (they are far better). <em>Functional Ecology</em>, <em>29</em>(5), 600‚Äì614. <a href="https://doi.org/10.1111/1365-2435.12425">https://doi.org/10.1111/1365-2435.12425</a>
            </div>
            <div id="ref-Halevy2009UnrEff" class="csl-entry" role="doc-biblioentry">
            Halevy, A., Norvig, P., &amp; Pereira, F. (2009). The Unreasonable Effectiveness of Data. <em>IEEE Intelligent Systems</em>, <em>24</em>(2), 8‚Äì12. <a href="https://doi.org/10.1109/MIS.2009.36">https://doi.org/10.1109/MIS.2009.36</a>
            </div>
            <div id="ref-Halko2011FinStr" class="csl-entry" role="doc-biblioentry">
            Halko, N., Martinsson, P. G., &amp; Tropp, J. A. (2011). Finding Structure with Randomness: Probabilistic Algorithms for Constructing Approximate Matrix Decompositions. <em>SIAM Review</em>, <em>53</em>(2), 217‚Äì288. <a href="https://doi.org/10.1137/090771806">https://doi.org/10.1137/090771806</a>
            </div>
            <div id="ref-Holm2019DefBla" class="csl-entry" role="doc-biblioentry">
            Holm, E. A. (2019). In defense of the black box. <em>Science</em>, <em>364</em>(6435), 26‚Äì27. <a href="https://doi.org/10.1126/science.aax0162">https://doi.org/10.1126/science.aax0162</a>
            </div>
            <div id="ref-Hortal2015SevSho" class="csl-entry" role="doc-biblioentry">
            Hortal, J., de Bello, F., Diniz-Filho, J. A. F., Lewinsohn, T. M., Lobo, J. M., &amp; Ladle, R. J. (2015). Seven Shortfalls that Beset Large-Scale Knowledge of Biodiversity. <em>Annual Review of Ecology, Evolution, and Systematics</em>, <em>46</em>(1), 523‚Äì549. <a href="https://doi.org/10.1146/annurev-ecolsys-112414-054400">https://doi.org/10.1146/annurev-ecolsys-112414-054400</a>
            </div>
            <div id="ref-Hutchinson2017CopSig" class="csl-entry" role="doc-biblioentry">
            Hutchinson, M. C., Cagua, E. F., &amp; Stouffer, D. B. (2017). Cophylogenetic signal is detectable in pollination interactions across ecological scales. <em>Ecology</em>, n/a‚Äìn/a. <a href="https://doi.org/10.1002/ecy.1955">https://doi.org/10.1002/ecy.1955</a>
            </div>
            <div id="ref-Jordano2016ChaEco" class="csl-entry" role="doc-biblioentry">
            Jordano, P. (2016a). Chasing Ecological Interactions. <em>PLOS Biol</em>, <em>14</em>(9), e1002559. <a href="https://doi.org/10.1371/journal.pbio.1002559">https://doi.org/10.1371/journal.pbio.1002559</a>
            </div>
            <div id="ref-Jordano2016SamNet" class="csl-entry" role="doc-biblioentry">
            Jordano, P. (2016b). Sampling networks of ecological interactions. <em>Functional Ecology</em>, <em>30</em>(12), 1883‚Äì1893. <a href="https://doi.org/10.1111/1365-2435.12763">https://doi.org/10.1111/1365-2435.12763</a>
            </div>
            <div id="ref-Kawatsu2021AreNet" class="csl-entry" role="doc-biblioentry">
            Kawatsu, K., Ushio, M., van Veen, F. J. F., &amp; Kondoh, M. (2021). Are networks of trophic interactions sufficient for understanding the dynamics of multi-trophic communities? Analysis of a tri-trophic insect food-web time-series. <em>Ecology Letters</em>, <em>24</em>(3), 543‚Äì552. <a href="https://doi.org/10.1111/ele.13672">https://doi.org/10.1111/ele.13672</a>
            </div>
            <div id="ref-Kefi2012MorMea" class="csl-entry" role="doc-biblioentry">
            K√©fi, S., Berlow, E. L., Wieters, E. A., Navarrete, S. A., Petchey, O. L., Wood, S. A., Boit, A., Joppa, L. N., Lafferty, K. D., Williams, R. J., Martinez, N. D., Menge, B. A., Blanchette, C. A., Iles, A. C., &amp; Brose, U. (2012). More than a meal‚Ä¶ integrating non-feeding interactions into food webs: More than a meal ‚Ä¶. <em>Ecology Letters</em>, <em>15</em>(4), 291‚Äì300. <a href="https://doi.org/10.1111/j.1461-0248.2011.01732.x">https://doi.org/10.1111/j.1461-0248.2011.01732.x</a>
            </div>
            <div id="ref-Litsios2012EffPhy" class="csl-entry" role="doc-biblioentry">
            Litsios, G., &amp; Salamin, N. (2012). Effects of Phylogenetic Signal on Ancestral State Reconstruction. <em>Systematic Biology</em>, <em>61</em>(3), 533‚Äì538. <a href="https://doi.org/10.1093/sysbio/syr124">https://doi.org/10.1093/sysbio/syr124</a>
            </div>
            <div id="ref-Maiorano2020Tet10" class="csl-entry" role="doc-biblioentry">
            Maiorano, L., Montemaggiori, A., Ficetola, G. F., O‚ÄôConnor, L., &amp; Thuiller, W. (2020a). TETRA-EU 1.0: A species-level trophic metaweb of European tetrapods. <em>Global Ecology and Biogeography</em>, <em>29</em>(9), 1452‚Äì1457. <a href="https://doi.org/10.1111/geb.13138">https://doi.org/10.1111/geb.13138</a>
            </div>
            <div id="ref-Maiorano2020DatTet" class="csl-entry" role="doc-biblioentry">
            Maiorano, L., Montemaggiori, A., Ficetola, G. F., O‚ÄôConnor, L., &amp; Thuiller, W. (2020b). <em>Data from: Tetra-EU 1.0: A species-level trophic meta-web of European tetrapods</em> (Version 3, pp. 16596876 bytes) [Data set]. Dryad. <a href="https://doi.org/10.5061/DRYAD.JM63XSJ7B">https://doi.org/10.5061/DRYAD.JM63XSJ7B</a>
            </div>
            <div id="ref-Marco2018ChaHum" class="csl-entry" role="doc-biblioentry">
            Marco, M. D., Venter, O., Possingham, H. P., &amp; Watson, J. E. M. (2018). Changes in human footprint drive changes in species extinction risk. <em>Nature Communications</em>, <em>9</em>(1), 4621. <a href="https://doi.org/10.1038/s41467-018-07049-5">https://doi.org/10.1038/s41467-018-07049-5</a>
            </div>
            <div id="ref-Mora2018IdeCom" class="csl-entry" role="doc-biblioentry">
            Mora, B. B., Gravel, D., Gilarranz, L. J., Poisot, T., &amp; Stouffer, D. B. (2018). Identifying a common backbone of interactions underlying food webs from different ecosystems. <em>Nature Communications</em>, <em>9</em>(1), 2603. <a href="https://doi.org/10.1038/s41467-018-05056-0">https://doi.org/10.1038/s41467-018-05056-0</a>
            </div>
            <div id="ref-Morales-Castilla2015InfBio" class="csl-entry" role="doc-biblioentry">
            Morales-Castilla, I., Matias, M. G., Gravel, D., &amp; Ara√∫jo, M. B. (2015). Inferring biotic interactions from proxies. <em>Trends in Ecology &amp; Evolution</em>, <em>30</em>(6), 347‚Äì356. <a href="https://doi.org/10.1016/j.tree.2015.03.014">https://doi.org/10.1016/j.tree.2015.03.014</a>
            </div>
            <div id="ref-Mouquet2012EcoAdv" class="csl-entry" role="doc-biblioentry">
            Mouquet, N., Devictor, V., Meynard, C. N., Munoz, F., Bersier, L.-F., Chave, J., Couteron, P., Dalecky, A., Fontaine, C., Gravel, D., Hardy, O. J., Jabot, F., Lavergne, S., Leibold, M., Mouillot, D., M√ºnkem√ºller, T., Pavoine, S., Prinzing, A., Rodrigues, A. S. L., ‚Ä¶ Thuiller, W. (2012). Ecophylogenetics: Advances and perspectives. <em>Biological Reviews</em>, <em>87</em>(4), 769‚Äì785. <a href="https://doi.org/10.1111/j.1469-185X.2012.00224.x">https://doi.org/10.1111/j.1469-185X.2012.00224.x</a>
            </div>
            <div id="ref-Nenzen2014Imp850" class="csl-entry" role="doc-biblioentry">
            Nenz√©n, H. K., Montoya, D., &amp; Varela, S. (2014). The Impact of 850,000 Years of Climate Changes on the Structure and Dynamics of Mammal Food Webs. <em>PLOS ONE</em>, <em>9</em>(9), e106651. <a href="https://doi.org/10.1371/journal.pone.0106651">https://doi.org/10.1371/journal.pone.0106651</a>
            </div>
            <div id="ref-OConnor2020UnvFoo" class="csl-entry" role="doc-biblioentry">
            O‚ÄôConnor, L. M. J., Pollock, L. J., Braga, J., Ficetola, G. F., Maiorano, L., Martinez-Almoyna, C., Montemaggiori, A., Ohlmann, M., &amp; Thuiller, W. (2020). Unveiling the food webs of tetrapods across Europe through the prism of the Eltonian niche. <em>Journal of Biogeography</em>, <em>47</em>(1), 181‚Äì192. <a href="https://doi.org/10.1111/jbi.13773">https://doi.org/10.1111/jbi.13773</a>
            </div>
            <div id="ref-Pan2010SurTra" class="csl-entry" role="doc-biblioentry">
            Pan, S. J., &amp; Yang, Q. (2010). A Survey on Transfer Learning. <em>IEEE Transactions on Knowledge and Data Engineering</em>, <em>22</em>(10), 1345‚Äì1359. <a href="https://doi.org/10.1109/TKDE.2009.191">https://doi.org/10.1109/TKDE.2009.191</a>
            </div>
            <div id="ref-Park2004StaMec" class="csl-entry" role="doc-biblioentry">
            Park, J., &amp; Newman, M. E. J. (2004). Statistical mechanics of networks. <em>Physical Review E</em>, <em>70</em>(6), 066117. <a href="https://doi.org/10.1103/PhysRevE.70.066117">https://doi.org/10.1103/PhysRevE.70.066117</a>
            </div>
            <div id="ref-Perretti2013ModFor" class="csl-entry" role="doc-biblioentry">
            Perretti, C. T., Munch, S. B., &amp; Sugihara, G. (2013). Model-free forecasting outperforms the correct mechanistic model for simulated and experimental data. <em>Proceedings of the National Academy of Sciences</em>, <em>110</em>(13), 5253‚Äì5257. <a href="https://doi.org/10.1073/pnas.1216076110">https://doi.org/10.1073/pnas.1216076110</a>
            </div>
            <div id="ref-Pires2015PleMeg" class="csl-entry" role="doc-biblioentry">
            Pires, M. M., Koch, P. L., Fari√±a, R. A., de Aguiar, M. A. M., dos Reis, S. F., &amp; Guimar√£es, P. R. (2015). Pleistocene megafaunal interaction networks became more vulnerable after human arrival. <em>Proceedings of the Royal Society B: Biological Sciences</em>, <em>282</em>(1814), 20151367. <a href="https://doi.org/10.1098/rspb.2015.1367">https://doi.org/10.1098/rspb.2015.1367</a>
            </div>
            <div id="ref-Poelen2014GloBio" class="csl-entry" role="doc-biblioentry">
            Poelen, J. H., Simons, J. D., &amp; Mungall, C. J. (2014). Global biotic interactions: An open infrastructure to share and analyze species-interaction datasets. <em>Ecological Informatics</em>, <em>24</em>, 148‚Äì159. <a href="https://doi.org/10.1016/j.ecoinf.2014.08.005">https://doi.org/10.1016/j.ecoinf.2014.08.005</a>
            </div>
            <div id="ref-Poisot2019EcoJl" class="csl-entry" role="doc-biblioentry">
            Poisot, T., Belisle, Z., Hoebeke, L., Stock, M., &amp; Szefer, P. (2019). EcologicalNetworks.jl - analysing ecological networks. <em>Ecography</em>. <a href="https://doi.org/10.1111/ecog.04310">https://doi.org/10.1111/ecog.04310</a>
            </div>
            <div id="ref-Poisot2021GloKno" class="csl-entry" role="doc-biblioentry">
            Poisot, T., Bergeron, G., Cazelles, K., Dallas, T., Gravel, D., MacDonald, A., Mercier, B., Violet, C., &amp; Vissault, S. (2021). Global knowledge gaps in species interaction networks data. <em>Journal of Biogeography</em>, <em>n/a</em>(n/a). <a href="https://doi.org/10.1111/jbi.14127">https://doi.org/10.1111/jbi.14127</a>
            </div>
            <div id="ref-Poisot2016StrPro" class="csl-entry" role="doc-biblioentry">
            Poisot, T., Cirtwill, A. R., Cazelles, K., Gravel, D., Fortin, M.-J., &amp; Stouffer, D. B. (2016). The structure of probabilistic networks. <em>Methods in Ecology and Evolution</em>, <em>7</em>(3), 303‚Äì312. <a href="https://doi.org/10.1111/2041-210X.12468">https://doi.org/10.1111/2041-210X.12468</a>
            </div>
            <div id="ref-Poisot2021ImpMam" class="csl-entry" role="doc-biblioentry">
            Poisot, T., Ouellet, M.-A., Mollentze, N., Farrell, M. J., Becker, D. J., Albery, G. F., Gibb, R. J., Seifert, S. N., &amp; Carlson, C. J. (2021, May 31). <em>Imputing the mammalian virome with linear filtering and singular value decomposition</em>. <a href="http://arxiv.org/abs/2105.14973">http://arxiv.org/abs/2105.14973</a>
            </div>
            <div id="ref-Poisot2018IntRet" class="csl-entry" role="doc-biblioentry">
            Poisot, T., &amp; Stouffer, D. B. (2018). Interactions retain the co-phylogenetic matching that communities lost. <em>Oikos</em>, <em>127</em>(2), 230‚Äì238. <a href="https://doi.org/10.1111/oik.03788">https://doi.org/10.1111/oik.03788</a>
            </div>
            <div id="ref-Price2003MacThe" class="csl-entry" role="doc-biblioentry">
            Price, P. W. (2003). <em>Macroevolutionary theory on macroecological patterns</em>. Cambridge University Press.
            </div>
            <div id="ref-Reeve2016HowPar" class="csl-entry" role="doc-biblioentry">
            Reeve, R., Leinster, T., Cobbold, C. A., Thompson, J., Brummitt, N., Mitchell, S. N., &amp; Matthews, L. (2016, December 8). <em>How to partition diversity</em>. <a href="http://arxiv.org/abs/1404.6520">http://arxiv.org/abs/1404.6520</a>
            </div>
            <div id="ref-Rosado2013GoiBac" class="csl-entry" role="doc-biblioentry">
            Rosado, B. H. P., Dias, A., &amp; de Mattos, E. (2013). Going Back to Basics: Importance of Ecophysiology when Choosing Functional Traits for Studying Communities and Ecosystems. <em>Natureza &amp; Conserva√ß~ao Revista Brasileira de Conserva√ß~ao Da Natureza</em>, <em>11</em>, 15‚Äì22. <a href="https://doi.org/10.4322/natcon.2013.002">https://doi.org/10.4322/natcon.2013.002</a>
            </div>
            <div id="ref-Runghen2021ExpNod" class="csl-entry" role="doc-biblioentry">
            Runghen, R., Stouffer, D. B., &amp; Dalla Riva, G. V. (2021). <em>Exploiting node metadata to predict interactions in large networks using graph embedding and neural networks</em>. <a href="https://doi.org/10.1101/2021.06.10.447991">https://doi.org/10.1101/2021.06.10.447991</a>
            </div>
            <div id="ref-Schoener1989FooWeb" class="csl-entry" role="doc-biblioentry">
            Schoener, T. W. (1989). Food webs from the small to the large. <em>Ecology</em>, <em>70</em>(6), 1559‚Äì1589.
            </div>
            <div id="ref-Solis-Lemus2017PhyPac" class="csl-entry" role="doc-biblioentry">
            Sol√≠s-Lemus, C., Bastide, P., &amp; An√©, C. (2017). PhyloNetworks: A Package for Phylogenetic Networks. <em>Molecular Biology and Evolution</em>, <em>34</em>(12), 3292‚Äì3298. <a href="https://doi.org/10.1093/molbev/msx235">https://doi.org/10.1093/molbev/msx235</a>
            </div>
            <div id="ref-Stock2021PaiLea" class="csl-entry" role="doc-biblioentry">
            Stock, M. (2021). Pairwise learning for predicting pollination interactions based on traits and phylogeny. <em>Ecological Modelling</em>, 14.
            </div>
            <div id="ref-Stouffer2012EvoCon" class="csl-entry" role="doc-biblioentry">
            Stouffer, D. B., Sales-Pardo, M., Sirer, M. I., &amp; Bascompte, J. (2012). Evolutionary Conservation of Species‚Äô Roles in Food Webs. <em>Science</em>, <em>335</em>(6075), 1489‚Äì1492. <a href="https://doi.org/10.1126/science.1216556">https://doi.org/10.1126/science.1216556</a>
            </div>
            <div id="ref-Strong2014ImpNon" class="csl-entry" role="doc-biblioentry">
            Strong, J. S., &amp; Leroux, S. J. (2014). Impact of Non-Native Terrestrial Mammals on the Structure of the Terrestrial Mammal Food Web of Newfoundland, Canada. <em>PLOS ONE</em>, <em>9</em>(8), e106264. <a href="https://doi.org/10.1371/journal.pone.0106264">https://doi.org/10.1371/journal.pone.0106264</a>
            </div>
            <div id="ref-Strydom2021RoaPre" class="csl-entry" role="doc-biblioentry">
            Strydom, T., Catchen, M. D., Banville, F., Caron, D., Dansereau, G., Desjardins-Proulx, P., Forero-Mu√±oz, N. R., Higino, G., Mercier, B., Gonzalez, A., Gravel, D., Pollock, L., &amp; Poisot, T. (2021). A roadmap towards predicting species interaction networks (across space and time). <em>Philosophical Transactions of the Royal Society B: Biological Sciences</em>, <em>376</em>(1837), 20210063. <a href="https://doi.org/10.1098/rstb.2021.0063">https://doi.org/10.1098/rstb.2021.0063</a>
            </div>
            <div id="ref-Strydom2021SvdEnt" class="csl-entry" role="doc-biblioentry">
            Strydom, T., Dalla Riva, G. V., &amp; Poisot, T. (2021). SVD Entropy Reveals the High Complexity of Ecological Networks. <em>Frontiers in Ecology and Evolution</em>, <em>9</em>. <a href="https://doi.org/10.3389/fevo.2021.623141">https://doi.org/10.3389/fevo.2021.623141</a>
            </div>
            <div id="ref-Thessen2014KnoExt" class="csl-entry" role="doc-biblioentry">
            Thessen, A. E., &amp; Parr, C. S. (2014). Knowledge extraction and semantic annotation of text from the encyclopedia of life. <em>PloS One</em>, <em>9</em>(3), e89550.
            </div>
            <div id="ref-Torrey2010TraLea" class="csl-entry" role="doc-biblioentry">
            Torrey, L., &amp; Shavlik, J. (2010). Transfer learning. In <em>Handbook of research on machine learning applications and trends: Algorithms, methods, and techniques</em> (pp. 242‚Äì264). IGI global.
            </div>
            <div id="ref-Trojelsgaard2016EcoNet" class="csl-entry" role="doc-biblioentry">
            Tr√∏jelsgaard, K., &amp; Olesen, J. M. (2016). Ecological networks in motion: Micro- and macroscopic variability across scales. <em>Functional Ecology</em>, <em>30</em>(12), 1926‚Äì1935. <a href="https://doi.org/10.1111/1365-2435.12710">https://doi.org/10.1111/1365-2435.12710</a>
            </div>
            <div id="ref-Upham2019InfMam" class="csl-entry" role="doc-biblioentry">
            Upham, N. S., Esselstyn, J. A., &amp; Jetz, W. (2019). Inferring the mammal tree: Species-level sets of phylogenies for questions in ecology, evolution, and conservation. <em>PLOS Biology</em>, <em>17</em>(12), e3000494. <a href="https://doi.org/10.1371/journal.pbio.3000494">https://doi.org/10.1371/journal.pbio.3000494</a>
            </div>
            <div id="ref-vanderHoorn2018SpaMax" class="csl-entry" role="doc-biblioentry">
            van der Hoorn, P., Lippner, G., &amp; Krioukov, D. (2018). Sparse Maximum-Entropy Random Graphs with a Given Power-Law Degree Distribution. <em>Journal of Statistical Physics</em>, <em>173</em>(3-4), 806‚Äì844. <a href="https://doi.org/10.1007/s10955-017-1887-7">https://doi.org/10.1007/s10955-017-1887-7</a>
            </div>
            <div id="ref-Vermaat2009MajDim" class="csl-entry" role="doc-biblioentry">
            Vermaat, J. E., Dunne, J. A., &amp; Gilbert, A. J. (2009). Major dimensions in food-web structure properties. <em>Ecology</em>, <em>90</em>(1), 278‚Äì282. <a href="http://www.ncbi.nlm.nih.gov/pubmed/19294932">http://www.ncbi.nlm.nih.gov/pubmed/19294932</a>
            </div>
            <div id="ref-Yeakel2014ColEco" class="csl-entry" role="doc-biblioentry">
            Yeakel, J. D., Pires, M. M., Rudolf, L., Dominy, N. J., Koch, P. L., Guimar√£es, P. R., &amp; Gross, T. (2014). Collapse of an ecological network in Ancient Egypt. <em>PNAS</em>, <em>111</em>(40), 14472‚Äì14477. <a href="https://doi.org/10.1073/pnas.1408471111">https://doi.org/10.1073/pnas.1408471111</a>
            </div>
            <div id="ref-Youden1950IndRat" class="csl-entry" role="doc-biblioentry">
            Youden, W. J. (1950). Index for rating diagnostic tests. <em>Cancer</em>, <em>3</em>(1), 32‚Äì35. <a href="https://doi.org/10.1002/1097-0142(1950)3:1&lt;32::AID-CNCR2820030106&gt;3.0.CO;2-3">https://doi.org/10.1002/1097-0142(1950)3:1&lt;32::AID-CNCR2820030106&gt;3.0.CO;2-3</a>
            </div>
            <div id="ref-Young2021BayInf" class="csl-entry" role="doc-biblioentry">
            Young, J.-G., Cantwell, G. T., &amp; Newman, M. E. J. (2021). Bayesian inference of network structure from unreliable data. <em>Journal of Complex Networks</em>, <em>8</em>(6). <a href="https://doi.org/10.1093/comnet/cnaa046">https://doi.org/10.1093/comnet/cnaa046</a>
            </div>
            <div id="ref-Young2007RanDot" class="csl-entry" role="doc-biblioentry">
            Young, S. J., &amp; Scheinerman, E. R. (2007). Random Dot Product Graph Models for Social Networks. In A. Bonato &amp; F. R. K. Chung (Eds.), <em>Algorithms and Models for the Web-Graph</em> (pp. 138‚Äì149). Springer. <a href="https://doi.org/10.1007/978-3-540-77004-6_11">https://doi.org/10.1007/978-3-540-77004-6_11</a>
            </div>
            <div id="ref-Zhu2006AutDim" class="csl-entry" role="doc-biblioentry">
            Zhu, M., &amp; Ghodsi, A. (2006). Automatic dimensionality selection from the scree plot via the use of profile likelihood. <em>Computational Statistics &amp; Data Analysis</em>, <em>51</em>(2), 918‚Äì930. <a href="https://doi.org/10.1016/j.csda.2005.09.010">https://doi.org/10.1016/j.csda.2005.09.010</a>
            </div>
            </div>
        </article>
    </main>

    <footer>
        <div class="container">
            <h2>Affiliations</h2> 
            <sup>1</sup>&nbsp;D√©partement de Sciences Biologiques, Universit√© de Montr√©al, Montr√©al, Canada; <sup>2</sup>&nbsp;Quebec Centre for Biodiversity Science, Montr√©al, Canada; <sup>3</sup>&nbsp;D√©partement de Biologie, Universit√© de Sherbrooke, Sherbrooke, Canada; <sup>4</sup>&nbsp;Department of Forest Resources Management, University of British Columbia, Vancouver, B.C., Canada; <sup>5</sup>&nbsp;Department of Biology, McGill University, Montr√©al, Canada; <sup>6</sup>&nbsp;Department of Ecology &amp; Evolutionary Biology, University of Toronto, Toronto, Canada; <sup>7</sup>&nbsp;Department of Forest and Conservation Sciences, University of British Columbia, Vancouver, Canada; <sup>8</sup>&nbsp;Centre for Integrative Ecology, School of Biological Sciences, University of Canterbury, Canterbury, New Zealand; <sup>9</sup>&nbsp;School of Mathematics and Statistics, University of Canterbury, Canterbury, New Zealand
            <br />
            <sup>‚Ä°</sup>&nbsp;These authors contributed equally to the work
        </div>
    </footer>
    
</body>
</html>