# Responses to reviewers comments

## Editor in chief

Three reviewers and an AE have looked at your manuscript. One reviewer and the
AE recommend rejection, because any revision would essentially result in quite a
different paper. Two other reviewers are critical, but suspect the manuscript
could be revised.  On balance, i have decided to keep the door open a jar, and
will allow a very significantly revised manuscript to be resubmitted. You may
decide the ask is too much, and try publishing elsewhere.

In particular, it is important that the major criticisms of the AE and reviewer
2 are all addressed. These get to the heart of the issue. The method is elegant,
but we just don't know how well it will work in real life situations. We need
some validation. The manuscript also requires additional methodological details,
and some new model runs. Some of these can be included in Supplementary
Information, but they are important.

## Associate editor

The manuscript presents an interesting study with promising results directed
towards important future research. However, based on the following limitations,
it would fit better to a methodological/theoretical journal, and an essentially
different paper would only be acceptable for a top journal: (1) the presentation
could be improved by providing a richer context of the problem,

> **TK**

(2) some major details of the approach need to be re-thought, like trait
distribution,

> **TK**

(3) one or several case studies should be used for
testing/validating the approach,

> **TK**

(4) beyond (3), also a discussion should clarify the generality of the approach
(whether its use is of more tactical or more strategic nature).

> **TK**

I see a change to transform the manuscript to a very good paper but also in its
present form it is useful and appropriate to a minor journal.

> **TK**

## Referee 1

The Manuscript ELE-01066-2021 addresses the important question of how to infer
species interactions and inherent networks. Authors used machine learning
algorithms to this end, using a foodweb of European mammals as training datasets
to make predictions on potential foodweb involving Canadian mammals. Results
indicate that the algorithm has a good predictive power, although it largely
tends to overestimate uncommon interactions. Authors conclude that colonialism
and nationalism are detrimental to biodiversity.

> We are glad that the reviewer recognizes the good predictive power of our
> approach. We would like to challenge the claim that we over-estimate uncommon
> interactions, for at least  two reasons. First, the output of our analysis
> (the metaweb) is probabilistic by design; this implies that we can describe an
> interaction as "being here", but we can moderate this statement by
> (essentially) saying that it is present with probability $p$, which strictly
> speaking represents the chance of success of a Bernoulli trial. We have
> modified most of the introduction to expand on the original definition of a
> metaweb (**TK**), explain how it has been used in other systems (**TK**), and
> why adopting a probabilistic vision of it is a more cautious approach
> (**TK**). Second, it is expected that uncommon interactions (by which we mean,
> more difficult to observe in nature) are going to also be less represented in
> interaction data stemming from observation, despite possibly being
> over-represented in nature. We have added a paragraph to the discussion, at
> **TK**, to expand on this point.

This is a nice study that sheds some new light into a fascinating problem in
ecology, that of inferring species interactions. Although the phenomenological
nature of the study and the black box of machine learning do not allow us to
understand WHY nor HOW, the algorithm made the correct inference most of the
time. I enjoyed reading the paper and I admire its concise way in highlighting
the main methodologies and their relevance. This paper will certainly be of
interest to ELE readers.

> We are glad that the reviewer foud the article concise and enjoyable to read;
> we certainly worked hard at making the text accessible. We do recognize that
> the algorithm is, in a sense, a black box. Not so much in that it relies on
> mathematical concepts that are obscure, but in that the mechanisms used for
> inference are not necessarily ecological. This being said, there have been a
> number of interesting results in the ecological modeling forecasting that show
> mechanisms-free models outpeforming models based on ecological knowledge: this
> has notably be shown for forecasting of spatio-temporal data; we have update
> the text at **TK** to reflect this. Taking a broader perspective, a common
> saying in machine learning is that data are "unreasonably effective"; we
> discus this statement at the place cited above, as it has an important
> implication: accurate algorithms are very often able to "learn" the
> mechanisms, even if they do not do so in a way that is epistemologically
> transparent. Finally, we discuss the role of AI/ML algorithms as data
> generators in the discussion, specifically **TK**.

I have some suggestions that shall improve the clarity of the manuscript and the
strength of its claims.

> We have addressed all of these suggestions, and thank the reviewer for them.

It sounds like authors are reinventing the wheel. Please provide a more thorough
introduction to current methods and knowledge on predicting ecological networks.

> The field of ecological networks prediction is relatively novel (at least when
> it comes to producing interaction-level predictions). It has been very
> recently reviewed by Strydom et al., now cited in the introduction at **TK**;
> this review indentified a number of challenges for the field to solve in order
> to move forward, which we discuss at **TK**, thereby outlining how our
> approach can be useful. We are confident that this will provide readers with a
> stronger justification for why our approach is solving problems that are not
> currently solvable.

Especially, the submitted manuscript looks scarily similar to that of Runghen,
R. et al. (incomplete reference in the list). Such similarity poses some doubts
regarding its novelty and originality. Also, the literature cited is quite
self-referential.

> We have clarified the Runghen et al. citation at **TK** --- their aim was to
> predict human movement from external metadata. We hope that this clarifies how
> different our method is. We understand that the literature is in part
> self-referential; this is because this manuscript is part of a larger research
> program; we are pleased to note that the reviewer has not identified citations
> that were not justifiable, or citations that were missing, suggesting that we
> are crediting ideas appropriately.

Some fundamental information is missing. This current paper must be
self-contained. Please provide more information about the European network:
which community, species interaction type, ecosystem, biome, climate, spatial
extent, time scale. Authors cannot force readers to go get such information
somewhere else.

> In addition to keeping the original citation to boht the data and the
> manuscript describing them, we have expanded the description of the European
> metaweb further, by listing **TK**; these changes have been made in the
> manuscript as **TK**.

Trait distribution. Please run the model using a normal distribution instead of
a uniform one. Why did you chose a UD?

> We picked a uniform distribution because it is the maximal entropy
> distribution available when the two parameters known are upper and lower
> bounds for a series of random deviates, which is the case for the ancestral
> character estimation model we used; we now mention this information in the
> text at **TK**.

“Traits” are NOT uniformly distributed at all. How does this affect the
estimates? Please provide new results.

> As per the previous suggestion, we added a series of simulations using a
> node-level normal distribution. The inference of the normal distribution
> parameters was done by converting the lower and upper bounds of the random
> uniform estimates at a 95% confidence interval, and turning this into a
> standard deviation (the mean is simply half the distance between the bounds).
> As we explain in Supp. Mat. **TK**, the normal distribution model makes wildly
> unrealistic predictions. First, it over-estimates the number of interactions,
> but it also assigns higher probabilities to all possible interactions. We
> share the concern expressed by the reviewer of not over-estimating uncommon
> interactions, and using a normal distribution model would do just that. We
> have mentionned the existence of the Supp. Mat. in the text at **TK**, but
> have not discussed the normal model further.

Furthermore, please do not create false expectation in the reader by continuing
referring to traits when, in fact, those are just node topological properties.
Please mind the difference between a node degree and a species trait; here,
there are no actual functional nor phenotypic traits, being node degree rank a
purely topological property.

> There is some semantic confusion for which we apologize; we use "traits" to
> refer to latent variables. For an epistemic standpoint, latent variables have
> no meaningful differences from empirical measurements, but in order to
> minimise the chances of mis-understanding, we have replaced "traits" in this
> context by "latent variables" throughout, and given a short rationale of why
> at **TK**. We nevertheless would like to gently push back on the idea that we
> are *creating* false expectations: we would like to assure the reviewer that
> no word in this manuscript has been written in bad faith.

Manuscript structure. Intro and discussion are quite sterile, they miss the
point. The state-of-the-art is not presented thoroughly as it does not build on
what we do and know so far.

> The changes to the introduction, linking it to the recent review on prediction
> (at **TK**), as well as suggested by other reviewers, will hopefully prove
> sufficient to alleviate this concern. We would like to note that we are
> operating under strict word count guidelines, and have prefered to add details
> to the results and methods when required.

Furthermore, the discussion is a complete separate piece, a random digression
detached from the rest of the manuscript. Linked to that, the conclusion,
although I truly love it, is quite out of context. Please seriously improve
those parts.

> We have modified the flow of the discussion according to the very helpful
> comments the reviewer made in the annotated PDF of the article, notably by
> **TK** at **TK**, **TK** at **TK**, and **TK** at **TK**. We are confident
> that the flow is now much better.

Please discuss your results in light of your large overestimations.

> As we answered previously, and as Supp. Mat. **TK** shows, we do not think we
> are over-estimating by a large amount; this being said, we have clarified why
> some overestimation is appropriate in the context of a metaweb, especially a
> probabilistic one (**TK**), and revisit this question in the discussion
> (**TK**).

Figures 3 and Fig. 6c are of bad quality. Please improve them.

> We have re-drawn a number of figures in the process of this revision; lacking
> substance advice as to what made the figures bad, we can only hope that these
> changes will satisfy reviewer 1.

Please find attached minor, detailed comments.

## Referee 2

This is a novel paper which uses phylogenetic signal as a way of constructing
food webs from community composition data. The approach is illustrated
constructing a Canadian mammal food web by drawing on European data. The paper
is well written and of an appropriate length. The approach is well described in
a clear way.

> We are glad that referee 2 recognizes the novelty of our approach, and found
> the manuscript easy to follow.

My concern with these modelling approaches (and the broader use of metawebs) is
that is that the approach itself is a very conservative one which will tend to
'average out' differences across locations.

> This is a valid concern; but getting back to the definition of a metaweb as
> introduced originally by Dunne (2006; cited in the text at **TK**), a metaweb
> *by definition* averages out differences across location. Functionally, it
> serves as a "regional interaction pool", and so captures the $\gamma$
> diversity of interactions, whereas local food webs (termed "metaweb
> realizations", following **TK**) represent the $\alpha$ diversity of
> interactions. We have clarified this distinction when introducing the concept
> of a metaweb at **TK**. Furthermore, drawing on a recent review of network
> prediction challenges, we discuss at **TK** the fact that having a metaweb can
> be used to facilitate the prediction of local food webs; this last point,
> indeed, remains a formidable challenge that our method is not intended to
> solve, and we indicate as much at **TK**. Nevertheless, we provide some
> elements of answers to the legitimate question of "Why then do we need a
> metaweb?", specifically at **TK** and **TK**.

Because the paper does not have an associated case study showing that it can
actually detect differences in food web structure between places or times, the
risk is that this is a method which allows an increase in the number of food
webs - while at the same time predisposing them to all look identical.

> We would like to respectfully push back on the statement that the paper lacks
> a case  study; indeed, the bulk of the manuscript *is* a case study,
> specifically of producing a list of trophic interactions in a species pool for
> which we do not have this information. A case study like the reviewer asks,
> while undoubtedly interesting, would require several metawebs of mammals,
> which we do not have at the moment. The concern about all networks becoming
> identical is an intersting one. On one hand, we suspect this to be the case
> (see **TK** and **TK** showing the existence of various food web structural
> backnones): they are variations around a common motif, and we now mention this
> as **TK**. On the other hand, studies using metawebs have shown that they
> exhibit some degree of variation when spatially downscaled; this is the case
> for example of **TK**, using a global fish food web, and we now mention this
> result at **TK**. To check that different species pools would not result in
> networks with the same structure, we provide a simple simulation as Supp. Mat.
> **TK**, wherein we sub-sample the Canadian species list, and show that there
> exists a distribution of network structure in a multi-variate space, as was
> used by **TK** to measure network structural dissimilarity.

The second issue is to do with when and how the phylogenetic signal would be
most robust at predicting trophic interactions. Presumably systems dominated by
generalists  and with high phylogenetic overlap with the region generating the
predictions will be very reliable in predicting food webs. However for areas
with high endemism and limited species pools which may have favoured
specialisation I am less convinced. Is it possible that some metric of
phylogenetic relatedness across the metaweb should be reported in carrying out
this kind of modelling .

> We produced this information when preparing this manuscript, but forgot to
> include it in the main text. We do apologize for this omission, which has now
> been fixed at **TK**. Furthermore, we do briefly discuss the question of
> phylogenetic signal in the discussion, in the context of estimating it from
> phylogenetic regression, at **TK**.

As with the use of metawebs, my concern is that some of the most interesting
ecology would be obscured by constructing food webs in this way.

> We hope that the clarification of the metaweb concept and its usefulness,
> respectively at **TK** and **TK**, will have alleviated this concern.

What I would really liked to have seen was a case study which showed were a set
of food webs had been able to show a clear ecological pattern (e.g. David Post's
lake food webs showing relationships with size and productivity), reconstruct
those food webs using the modelling approach and show that the pattern was still
evident, and not averaged away by the process of inferring the food webs.

> An important step of our pipeline is to compare our prediction to data from
> Strong & Leroux (**TK**), which is exactly similar to what the reviewer
> suggests. We reach a very good predictive accuracy on this dataset, suggesting
> that our method is able to make correct inference on the structure of the food
> web. In addition to Supp. Mat. **TK** on structural variation, and Supp. Mat.
> **TK** on the ability to estimate network properties from a SVD, we are
> confident that this concern has been addressed.

## Referee 3

In this study, the authors developed a data-driven framework for the prediction
of unknown ecological networks. Specifically, as far as my understanding, the
analytical procedure is as the following three steps. First, for a known
ecological network (food web of European mammalian in this case), the latent
generality trait (i.e., prey species composition) and vulnerability trait
(predator species composition) are extracted by performing t-SVD of the
interaction (adjacency) matrix. Then, the generality/vulnerability traits of
species in the other community with an unknown network structure are estimated
by mapping the extracted traits of the known community onto global phylogeny.
Finally, left and right subspaces of the target community are probabilistically
obtained based on the estimated generality/vulnerability traits: multiplying
these subspaces yield the probabilistic network realization of the unknown
community. Applying this framework to the established European food web and
mammalian phylogeny, the authors reconstructed the hitherto unknown trophic
network of Canadian mammals.

> We are grateful to the referee for a very succint summary of our manuscript;
> although this information was present in the caption of Figure 1, we
> understand how valuable it can be to guide readers, and have adapted parts of
> it to add to the abstract and introduction, notably at **TK** and **TK**.

Overall, I agree with the ecological importance of the topic covered in this
study (i.e., how to infer the network structure of an ecological community where
there is poor knowledge or lack of direct observations of species interaction).
However, the text is hard to follow, and I feel there is room to revise,
especially for its presentation.

> Based on comments by all three reviewers, we have greatly improved the
> presentation of the manuscript, and we hope that it will prove appropriate in
> its revised form.

For example, the authors used RDPG (random dot product graph) to obtain the
latent feature subspaces, but I didn’t know anything about this method, so I
couldn’t understand how it relates to SVD from the text alone. Thus, I suggest
the authors add the text that explains the concept of the RDPG (e.g., this
assumes that interaction occurs more often between the species with similar
functional traits) and that SVD is a promising method to provide these
functional traits from the observed network data. Similar incomprehensible
points may be found in the Methods.

> We thank the reviewer for pointing out this issue; RDPG is a simple trick to
> transform the three matrices $\mathbf{U}$, $\mathbf{\Sigma}$, and
> $\mathbf{V}'$ of the SVD into two, defined as
> $\mathbf{L}=\mathbf{U}\sqrt{\mathbf{\Sigma}}$ and
> $\mathbf{R}=\sqrt{\mathbf{\Sigma}}\mathbf{V}'$, so that the underlying
> original matrix can be reconstructed through the matrix multiplication
> $\mathbf{L}\mathbf{U}$. We have added this information at **TK**. As we
> clarified at **TK** following comments from reviewer 1, the "traits" here are
> more accurately described at latent variables, but their similarity is indeed
> a predictor of the expected similarity of interactions.
 
In addition, I am wondering if all the results are described correctly.
Specifically, the authors assert that the reconstructed Canadian food web
recovers over 95% of known interactions by comparing databases of recorded
pairwise interactions. However, I found no corresponding description in “Results
and discussion of the case study” section (and also “Data cleanup, discovery,
validation, and thresholding” section). Thus, if the result exists, please add
it to the appropriate part and otherwise remove the description from the
abstract.

> These results were already present in the text, but we have clarified their
> presentation at **TK** and **TK**. In addition, following comments from
> reviewer 2, we provide a full taxonomic breakdown of which taxa are shared
> (see **TK**).

Minor comments:

L119–120: The logic is unclear. Is there any assumption that a list of prey and
predator species tend to be similar among closely related species? Please
clarify.

> **TK**

L143 & 296: 'a' RDPG -> an RDPG?

> actually yes - the R in RDPG starts with the sound ä as in ar [TS aside] I
> learnt something new today **TK FIX**

L150–153: Did you use all the data in the European metaweb for an interaction
presence/absence classification analysis? If this is true, overfitting may occur
in determining the threshold for interaction probability. Thus, I suggest the
authors adopt some approaches to avoid overfitting, such as the cross-validation
methods.

> We did indeed use all the data from the European metaweb in order to produce
> the left and right sub-spaces for the transfer step. Based on this comment, we
> have produced Supp. Mat. **TK**, in which we show that even with up to 40% of
> interactions withheld, the binary classifier remains highly accurate;
> furthermore, we show that the threshold identified responds to network
> completeness of sampling, and that for reasonably well sampled networks (as
> the European metaweb is, see **TK**), the structural properties of the
> reconstructed network are within a very narrow range of what they actually
> are. We are therefore confident that the risk of overfitting is minimal.

L247–249 & top panels in Fig. 6. I consider these results are algebraically
self-evident. Specifically, 1st left and right singular vectors correspond to
the direction with the highest variance for the row- and column-wise elements,
respectively. That is, elements with greater contribution to these 1st singular
vectors tend to have larger values for the row or column vector in the original
matrix. Therefore, I think these results provide no novel information, and a
simple statement of this should suffice.

> We do feel like these results can guide the reading of the manuscript;
> furthermore, the situation would look a lot different from an entirely modular
> network (for example), and so we have made no changes to the manuscript.

L301–304. In addition to the authors' mention, I think biased information of
species interactions toward specific behaviours or ecological relationships may
significantly limit such approaches. As you know, there are various types of
ecological relationships, such as competition, mutualism, etc. However, for many
of them, species interactions have not yet been studied as much as for trophic
relationships (e.g., Kéfi et al. 2012, Ecol Lett). Furthermore, a recent study
implied that, even in a community that has been studied as a prey-predator
system, various unanticipated interactions (these might not be trophic ones)
play a significant role in its dynamics (Kawatsu et al. 2021, Ecol Lett).
Applying the approach proposed in this paper to such a system would result in
biased predictions. Please discuss this point in the manuscript.

> We do explicitly state that we are focusing on Food Web prediction but this
> point does potentially segue into a nice discussion as we can touch on
> different applications. We could also point to Stock and friends 2021 that do
> something similar but for pollinator networks