%!TEX TS-program = xelatex
%DIF LATEXDIFF DIFFERENCE FILE
%DIF DEL reviews/01_ELE/01_draft.tex           Thu Oct 28 19:00:33 2021
%DIF ADD reviews/01_ELE/01_draft_revised.tex   Thu Oct 28 19:00:32 2021
\documentclass[11pt]{article}

\usepackage[english]{babel}

\usepackage{amsmath,amssymb,amsfonts}
\usepackage[utf8]{inputenc}
\usepackage[T1]{fontenc}
\usepackage{stix2}
\usepackage[scaled]{helvet}
\usepackage[scaled]{inconsolata}

\usepackage{lastpage}

\usepackage{setspace}

\usepackage{ccicons}

\usepackage[hang,flushmargin]{footmisc}

\usepackage{geometry}

\setlength{\parindent}{0pt}
\setlength{\parskip}{6pt plus 2pt minus 1pt}

\usepackage{fancyhdr}
\renewcommand{\headrulewidth}{0pt}\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}

\makeatletter
\newcounter{tableno}
\newenvironment{tablenos:no-prefix-table-caption}{
  \caption@ifcompatibility{}{
    \let\oldthetable\thetable
    \let\oldtheHtable\theHtable
    \renewcommand{\thetable}{tableno:\thetableno}
    \renewcommand{\theHtable}{tableno:\thetableno}
    \stepcounter{tableno}
    \captionsetup{labelformat=empty}
  }
}{
  \caption@ifcompatibility{}{
    \captionsetup{labelformat=default}
    \let\thetable\oldthetable
    \let\theHtable\oldtheHtable
    \addtocounter{table}{-1}
  }
}
\makeatother

\usepackage{array}
\newcommand{\PreserveBackslash}[1]{\let\temp=\\#1\let\\=\temp}
\let\PBS=\PreserveBackslash

\usepackage[breaklinks=true]{hyperref}
\hypersetup{colorlinks,%
citecolor=blue,%
filecolor=blue,%
linkcolor=blue,%
urlcolor=blue}
\usepackage{url}

\usepackage{caption}
\setcounter{secnumdepth}{0}
\usepackage{cleveref}

\usepackage{graphicx}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth
\else\Gin@nat@width\fi}
\makeatother
\let\Oldincludegraphics\includegraphics
\renewcommand{\includegraphics}[1]{\Oldincludegraphics[width=\maxwidth]{#1}}

\usepackage{longtable}
\usepackage{booktabs}

\usepackage{color}
\usepackage{fancyvrb}
\newcommand{\VerbBar}{|}
\newcommand{\VERB}{\Verb[commandchars=\\\{\}]}
\DefineVerbatimEnvironment{Highlighting}{Verbatim}{commandchars=\\\{\}}
% Add ',fontsize=\small' for more characters per line
\usepackage{framed}
\definecolor{shadecolor}{RGB}{248,248,248}
\newenvironment{Shaded}{\begin{snugshade}}{\end{snugshade}}
\newcommand{\KeywordTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\DataTypeTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{#1}}
\newcommand{\DecValTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\BaseNTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\FloatTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\ConstantTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\CharTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\SpecialCharTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\StringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\VerbatimStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\SpecialStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\ImportTok}[1]{#1}
\newcommand{\CommentTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\DocumentationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\AnnotationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\CommentVarTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\OtherTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{#1}}
\newcommand{\FunctionTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\VariableTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\ControlFlowTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\OperatorTok}[1]{\textcolor[rgb]{0.81,0.36,0.00}{\textbf{#1}}}
\newcommand{\BuiltInTok}[1]{#1}
\newcommand{\ExtensionTok}[1]{#1}
\newcommand{\PreprocessorTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\AttributeTok}[1]{\textcolor[rgb]{0.77,0.63,0.00}{#1}}
\newcommand{\RegionMarkerTok}[1]{#1}
\newcommand{\InformationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\WarningTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\AlertTok}[1]{\textcolor[rgb]{0.94,0.16,0.16}{#1}}
\newcommand{\ErrorTok}[1]{\textcolor[rgb]{0.64,0.00,0.00}{\textbf{#1}}}
\newcommand{\NormalTok}[1]{#1}

\newlength{\cslhangindent}
\setlength{\cslhangindent}{1.5em}
\newlength{\csllabelwidth}
\setlength{\csllabelwidth}{3em}
\newenvironment{CSLReferences}[3] % #1 hanging-ident, #2 entry spacing
 {% don't indent paragraphs
  \setlength{\parindent}{0pt}
  % turn on hanging indent if param 1 is 1
  \ifodd #1 \everypar{\setlength{\hangindent}{\cslhangindent}}\ignorespaces\fi
  % set entry spacing
  \ifnum #2 > 0
  \setlength{\parskip}{#2\baselineskip}
  \fi
 }%
 {}
\usepackage{calc} % for \widthof, \maxof
\newcommand{\CSLBlock}[1]{#1\hfill\break}
\newcommand{\CSLLeftMargin}[1]{\parbox[t]{\maxof{\widthof{#1}}{\csllabelwidth}}{#1}}
\newcommand{\CSLRightInline}[1]{\parbox[t]{\linewidth}{#1}}
\newcommand{\CSLIndent}[1]{\hspace{\cslhangindent}#1}\geometry{verbose,letterpaper,tmargin=2.2cm,bmargin=2.2cm,lmargin=2.2cm,rmargin=2.2cm}

\usepackage{lineno}
\usepackage[nolists,noheads]{endfloat}

\pagestyle{plain}

\tolerance=1
\emergencystretch=\maxdimen
\hyphenpenalty=10000
\hbadness=10000

\doublespacing

\fancypagestyle{normal}
{
  \fancyhf{}
  \fancyfoot[R]{\footnotesize\sffamily\thepage\ of \pageref*{LastPage}}
}
%DIF PREAMBLE EXTENSION ADDED BY LATEXDIFF
%DIF BOLD PREAMBLE %DIF PREAMBLE
\DeclareOldFontCommand{\bf}{\normalfont\bfseries}{\mathbf} %DIF PREAMBLE
\providecommand{\DIFaddtex}[1]{{\bf #1}} %DIF PREAMBLE
\providecommand{\DIFdeltex}[1]{} %DIF PREAMBLE
%DIF COLOR PREAMBLE %DIF PREAMBLE
\RequirePackage{color} %DIF PREAMBLE
\providecommand{\DIFaddbegin}{\protect\color{blue}} %DIF PREAMBLE
\providecommand{\DIFaddend}{\protect\color{black}} %DIF PREAMBLE
\providecommand{\DIFdelbegin}{\protect\color{red}} %DIF PREAMBLE
\providecommand{\DIFdelend}{\protect\color{black}} %DIF PREAMBLE
\providecommand{\DIFmodbegin}{} %DIF PREAMBLE
\providecommand{\DIFmodend}{} %DIF PREAMBLE
%DIF FLOATSAFE PREAMBLE %DIF PREAMBLE
\providecommand{\DIFaddFL}[1]{\DIFadd{#1}} %DIF PREAMBLE
\providecommand{\DIFdelFL}[1]{\DIFdel{#1}} %DIF PREAMBLE
\providecommand{\DIFaddbeginFL}{} %DIF PREAMBLE
\providecommand{\DIFaddendFL}{} %DIF PREAMBLE
\providecommand{\DIFdelbeginFL}{} %DIF PREAMBLE
\providecommand{\DIFdelendFL}{} %DIF PREAMBLE
%DIF HYPERREF PREAMBLE %DIF PREAMBLE
\providecommand{\DIFadd}[1]{\texorpdfstring{\DIFaddtex{#1}}{#1}} %DIF PREAMBLE
\providecommand{\DIFdel}[1]{\texorpdfstring{\DIFdeltex{#1}}{}} %DIF PREAMBLE
\newcommand{\DIFscaledelfig}{0.5}
%DIF HIGHLIGHTGRAPHICS PREAMBLE %DIF PREAMBLE
\RequirePackage{settobox} %DIF PREAMBLE
\RequirePackage{letltxmacro} %DIF PREAMBLE
\newsavebox{\DIFdelgraphicsbox} %DIF PREAMBLE
\newlength{\DIFdelgraphicswidth} %DIF PREAMBLE
\newlength{\DIFdelgraphicsheight} %DIF PREAMBLE
% store original definition of \includegraphics %DIF PREAMBLE
\LetLtxMacro{\DIFOincludegraphics}{\includegraphics} %DIF PREAMBLE
\newcommand{\DIFaddincludegraphics}[2][]{{\color{blue}\fbox{\DIFOincludegraphics[#1]{#2}}}} %DIF PREAMBLE
\newcommand{\DIFdelincludegraphics}[2][]{% %DIF PREAMBLE
\sbox{\DIFdelgraphicsbox}{\DIFOincludegraphics[#1]{#2}}% %DIF PREAMBLE
\settoboxwidth{\DIFdelgraphicswidth}{\DIFdelgraphicsbox} %DIF PREAMBLE
\settoboxtotalheight{\DIFdelgraphicsheight}{\DIFdelgraphicsbox} %DIF PREAMBLE
\scalebox{\DIFscaledelfig}{% %DIF PREAMBLE
\parbox[b]{\DIFdelgraphicswidth}{\usebox{\DIFdelgraphicsbox}\\[-\baselineskip] \rule{\DIFdelgraphicswidth}{0em}}\llap{\resizebox{\DIFdelgraphicswidth}{\DIFdelgraphicsheight}{% %DIF PREAMBLE
\setlength{\unitlength}{\DIFdelgraphicswidth}% %DIF PREAMBLE
\begin{picture}(1,1)% %DIF PREAMBLE
\thicklines\linethickness{2pt} %DIF PREAMBLE
{\color[rgb]{1,0,0}\put(0,0){\framebox(1,1){}}}% %DIF PREAMBLE
{\color[rgb]{1,0,0}\put(0,0){\line( 1,1){1}}}% %DIF PREAMBLE
{\color[rgb]{1,0,0}\put(0,1){\line(1,-1){1}}}% %DIF PREAMBLE
\end{picture}% %DIF PREAMBLE
}\hspace*{3pt}}} %DIF PREAMBLE
} %DIF PREAMBLE
\LetLtxMacro{\DIFOaddbegin}{\DIFaddbegin} %DIF PREAMBLE
\LetLtxMacro{\DIFOaddend}{\DIFaddend} %DIF PREAMBLE
\LetLtxMacro{\DIFOdelbegin}{\DIFdelbegin} %DIF PREAMBLE
\LetLtxMacro{\DIFOdelend}{\DIFdelend} %DIF PREAMBLE
\DeclareRobustCommand{\DIFaddbegin}{\DIFOaddbegin \let\includegraphics\DIFaddincludegraphics} %DIF PREAMBLE
\DeclareRobustCommand{\DIFaddend}{\DIFOaddend \let\includegraphics\DIFOincludegraphics} %DIF PREAMBLE
\DeclareRobustCommand{\DIFdelbegin}{\DIFOdelbegin \let\includegraphics\DIFdelincludegraphics} %DIF PREAMBLE
\DeclareRobustCommand{\DIFdelend}{\DIFOaddend \let\includegraphics\DIFOincludegraphics} %DIF PREAMBLE
\LetLtxMacro{\DIFOaddbeginFL}{\DIFaddbeginFL} %DIF PREAMBLE
\LetLtxMacro{\DIFOaddendFL}{\DIFaddendFL} %DIF PREAMBLE
\LetLtxMacro{\DIFOdelbeginFL}{\DIFdelbeginFL} %DIF PREAMBLE
\LetLtxMacro{\DIFOdelendFL}{\DIFdelendFL} %DIF PREAMBLE
\DeclareRobustCommand{\DIFaddbeginFL}{\DIFOaddbeginFL \let\includegraphics\DIFaddincludegraphics} %DIF PREAMBLE
\DeclareRobustCommand{\DIFaddendFL}{\DIFOaddendFL \let\includegraphics\DIFOincludegraphics} %DIF PREAMBLE
\DeclareRobustCommand{\DIFdelbeginFL}{\DIFOdelbeginFL \let\includegraphics\DIFdelincludegraphics} %DIF PREAMBLE
\DeclareRobustCommand{\DIFdelendFL}{\DIFOaddendFL \let\includegraphics\DIFOincludegraphics} %DIF PREAMBLE
%DIF LISTINGS PREAMBLE %DIF PREAMBLE
\RequirePackage{listings} %DIF PREAMBLE
\RequirePackage{color} %DIF PREAMBLE
\lstdefinelanguage{DIFcode}{ %DIF PREAMBLE
%DIF DIFCODE_BOLD %DIF PREAMBLE
  % unfortunately \bfseries cannot be combined with ttfamily without extra packages %DIF PREAMBLE
  % also morecomment=[il] is broken as of v1.5b of listings at least %DIF PREAMBLE
  % workaround: plot in white with tiny font %DIF PREAMBLE
  % morecomment=[il]{\%DIF\ <\ }, %DIF PREAMBLE
  moredelim=[il][\color{white}\tiny]{\%DIF\ <\ }, %DIF PREAMBLE
  moredelim=[il][\sffamily\bfseries]{\%DIF\ >\ } %DIF PREAMBLE
} %DIF PREAMBLE
\lstdefinestyle{DIFverbatimstyle}{ %DIF PREAMBLE
	language=DIFcode, %DIF PREAMBLE
	basicstyle=\ttfamily, %DIF PREAMBLE
	columns=fullflexible, %DIF PREAMBLE
	keepspaces=true %DIF PREAMBLE
} %DIF PREAMBLE
\lstnewenvironment{DIFverbatim}{\lstset{style=DIFverbatimstyle}}{} %DIF PREAMBLE
\lstnewenvironment{DIFverbatim*}{\lstset{style=DIFverbatimstyle,showspaces=true}}{} %DIF PREAMBLE
%DIF END PREAMBLE EXTENSION ADDED BY LATEXDIFF

\begin{document}
\raggedright
\thispagestyle{empty}
{\Large\bfseries\sffamily Food web reconstruction through phylogenetic
transfer of low-rank network representation}
\vskip 5em

%
\href{https://orcid.org/0000-0001-6067-1349}{Tanya\,Strydom}%
%
\,\textsuperscript{1,2,‡}\quad %
\href{https://orcid.org/0000-0003-0193-5441}{Salomé\,Bouskila}%
%
\,\textsuperscript{1,‡}\quad %
\href{https://orcid.org/0000-0001-9051-0597}{Francis\,Banville}%
%
\,\textsuperscript{1,3,2}\quad %
\href{https://orcid.org/0000-0003-4036-977X}{Ceres\,Barros}%
%
\,\textsuperscript{4}\quad %
\href{https://orcid.org/0000-0002-2151-6693}{Dominique\,Caron}%
%
\,\textsuperscript{5,2}\quad %
\href{https://orcid.org/0000-0003-0452-6993}{Maxwell J\,Farrell}%
%
\,\textsuperscript{6}\quad %
\href{https://orcid.org/0000-0002-9935-1366}{Marie-Josée\,Fortin}%
%
\,\textsuperscript{6}\quad %
\href{https://orcid.org/0000-0003-3220-6161}{Victoria\,Hemming}%
%
\,\textsuperscript{\DIFdelbegin \DIFdel{4}\DIFdelend \DIFaddbegin \DIFadd{7}\DIFaddend }\quad %
\href{https://orcid.org/0000-0002-4104-9463}{Benjamin\,Mercier}%
%
\,\textsuperscript{3,2}\quad %
\href{https://orcid.org/0000-0002-6004-4027}{Laura J.\,Pollock}%
%
\,\textsuperscript{5,2}\quad %
\href{https://orcid.org/0000-0001-5785-8321}{Rogini\,Runghen}%
%
\,\textsuperscript{\DIFdelbegin \DIFdel{7}\DIFdelend \DIFaddbegin \DIFadd{8}\DIFaddend }\quad %
\href{https://orcid.org/0000-0002-3454-0633}{Giulio V.\,Dalla Riva}%
%
\,\textsuperscript{\DIFdelbegin \DIFdel{8}\DIFdelend \DIFaddbegin \DIFadd{9}\DIFaddend }\quad %
\href{https://orcid.org/0000-0002-0735-5184}{Timothée\,Poisot}%
%
\,\textsuperscript{1,2}

\textsuperscript{1}\,Département de Sciences Biologiques, Université de
Montréal, Montréal, Canada\quad \textsuperscript{2}\,\DIFdelbegin \DIFdel{Québec }\DIFdelend \DIFaddbegin \DIFadd{Quebec }\DIFaddend Centre for
Biodiversity \DIFdelbegin \DIFdel{Sciences}\DIFdelend \DIFaddbegin \DIFadd{Science}\DIFaddend , Montréal,
Canada\quad \textsuperscript{3}\,\DIFaddbegin \DIFadd{Département de Biologie, }\DIFaddend Université de
Sherbrooke, Sherbrooke, Canada\quad \textsuperscript{4}\,Department of
Forest Resources Management, University of British Columbia, Vancouver,
\DIFaddbegin \DIFadd{B.C., }\DIFaddend Canada\quad \textsuperscript{5}\,Department of Biology, McGill
University, Montréal, Canada\quad \textsuperscript{6}\,Department of
Ecology \& Evolutionary Biology, University of Toronto, Toronto,
Canada\quad \textsuperscript{7}\,\DIFaddbegin \DIFadd{Department of Forest and Conservation
Sciences, University of British Columbia, Vancouver,
Canada}\quad \DIFadd{\textsuperscript{8}\,}\DIFaddend Centre for Integrative Ecology, School
of Biological Sciences, University of Canterbury, Canterbury, New
Zealand\quad \textsuperscript{\DIFdelbegin \DIFdel{8}\DIFdelend \DIFaddbegin \DIFadd{9}\DIFaddend }\,School of Mathematics and Statistics,
University of Canterbury, Canterbury, New Zealand

\textsuperscript{‡}\,These authors contributed equally to the work\\

\textbf{Correspondance to:}\\
Timothée Poisot --- \texttt{timothee.poisot@umontreal.ca}\\

\vfill
This work is released by its authors under a CC-BY 4.0 license\hfill\ccby\\
Last revision: \emph{\today}

\clearpage
\thispagestyle{empty}

\vfill
Despite their importance in many ecological processes, collecting data
and information on ecological interactions is an exceedingly challenging
task. For this reason, large parts of the world have a data deficit when
it comes to species interactions, and how the resulting networks are
structured. As data collection alone is unlikely to be sufficient,
community ecologists must adopt predictive methods. Here\DIFaddbegin \DIFadd{, }\DIFaddend we develop
such a method \DIFdelbegin \DIFdel{, }\DIFdelend relying on graph embedding and transfer learning to
assemble a predicted list of trophic interactions between Canadian
mammals. This interaction list is derived from the European food web,
despite sharing \DIFdelbegin \DIFdel{only 5}\DIFdelend \DIFaddbegin \DIFadd{4}\DIFaddend \% of common species with Canada. The results of the
predictive model are compared against databases of recorded pairwise
interactions, showing that we correctly recover \DIFdelbegin \DIFdel{over 95}\DIFdelend \DIFaddbegin \DIFadd{91}\DIFaddend \% of known
interactions. We provide guidance on how this method can be adapted by
substituting some approaches or predictors in order to make it more
generally applicable.



\vfill

\clearpage
\linenumbers
\pagestyle{normal}

\hypertarget{introduction}{%
\section{Introduction}\label{introduction}}

There are two core challenges we are faced with in furthering our
understanding of ecological networks across space, particularly at
macro-ecologically relevant scales (\emph{e.g.} Trøjelsgaard \& Olesen
2016). First, \DIFaddbegin \DIFadd{ecological }\DIFaddend networks within a location are difficult to
sample properly (Jordano 2016a, b), resulting in a widespread ``Eltonian
shortfall'' (Hortal \emph{et al.} 2015), \emph{i.e.} a lack of knowledge
about \DIFdelbegin \DIFdel{inter and intra }\DIFdelend \DIFaddbegin \DIFadd{inter- and intra- }\DIFaddend specific relationships. This first challenge has
been, in large part, addressed by the recent emergence of a suite of
methods aiming to predict interactions within \emph{existing} networks,
many of which are reviewed in Strydom \emph{et al.} (2021a). Second,
recent analyses based on collected data (Poisot \emph{et al.} 2021a) or
metadata (Cameron \emph{et al.} 2019) highlight that ecological networks
are currently studied in a biased subset of space and bioclimates, which
impedes our ability to generalize any local understanding of network
structure. Meaning that, although the framework to address
incompleteness \emph{within} networks exists, there would still be
regions for which, due to a \emph{lack} of local interaction data, we
are unable to infer potential species interactions. Having a general
solution for inferring \DIFdelbegin \DIFdel{a }\DIFdelend \emph{\DIFdelbegin \DIFdel{plausible}\DIFdelend \DIFaddbegin \DIFadd{potential}\DIFaddend } \DIFdelbegin \DIFdel{metaweb }\DIFdelend \DIFaddbegin \DIFadd{interactions }\DIFaddend (despite the
unavailability of interaction data) could be the catalyst for
significant breakthroughs in our ability to start thinking about species
interaction networks over large spatial scales. \DIFaddbegin \DIFadd{In a recent overview of
the field of ecological network prediction, Strydom \emph{et al.}
(2021a) identified two challenges of interest to the prediction of
interactions at large scales. First, there is a relative scarcity of
relevant data in most places globally -- paradoxically, this restricts
our ability to infer interactions to locations where inference is
perhaps the least required; second, accurate predictions often demand
accurate predictors, and the lack of methods that can leverage small
amount of data is a serious impediment to our predictive ability
globally.
}\DIFaddend 

Here, we present a general method \DIFdelbegin \DIFdel{for }\DIFdelend \DIFaddbegin \DIFadd{to recommend potential trophic
interactions, relying on }\DIFaddend the transfer learning of network
representations, \DIFdelbegin \DIFdel{relying on the }\DIFdelend \DIFaddbegin \DIFadd{specifically by using }\DIFaddend similarities of species in a
biologically/ecologically relevant proxy space (\emph{e.g.} shared
morphology or ancestry). Transfer learning is a machine learning
methodology that uses the knowledge gained from solving one problem and
applying it to a related (destination) problem (Pan \& Yang 2010; Torrey
\& Shavlik 2010). In this instance, we solve the problem of predicting
trophic interactions between species, based on knowledge extracted from
another species pool for which interactions are known by using
phylogenetic structure as a medium for transfer. \DIFdelbegin \DIFdel{This allows us to
construct a \emph{probabilistic} metaweb for a community for which we
have \emph{no} prior trophic interaction data for the desired species
pool. Our methodology is outlined in fig.~\ref{fig:concept}, where we
provide an illustration based on learning the embedding of a metaweb of
trophic interactions for European mammals (known interactions; Maiorano
\emph{et al.} 2020b, a) and, based on phylogenetic relationships between
mammals globally (\emph{i.e.}, phylogenetic tree Upham \emph{et al.}
2019), infer a metaweb for the Canadian mammalian species pool
(interactions are treated as unknown in this instance).
}%DIFDELCMD < 

%DIFDELCMD < %%%
\DIFdelend There is a plurality of
measures of species similarities that can be used for \DIFaddbegin \DIFadd{inferring
\emph{potential} species interactions \emph{i.e.} }\DIFaddend metaweb reconstruction
(see \emph{e.g.} Morales-Castilla \emph{et al.} 2015); however,
phylogenetic proximity has several desirable properties when working at
large scales. Gerhold \emph{et al.} (2015) made the point that
phylogenetic signal captures diversification of characters (large
macro-evolutionary process), but not necessarily community assembly
(fine ecological process); Dormann \emph{et al.} (2010) previously found
very similar conclusions. Interactions tend \DIFaddbegin \DIFadd{to }\DIFaddend reflect a phylogenetic
signal because they have a conserved pattern of evolutionary convergence
that encompasses a wide range of ecological and evolutionary mechanisms
(Cavender-Bares \emph{et al.} 2009; Mouquet \emph{et al.} 2012), and -
most importantly - retain this signal even when it is not detectable at
the community scale (Hutchinson \emph{et al.} 2017; Poisot \& Stouffer
2018). Finally, species interactions at macro-ecological scales seem to
respond mostly to macro-evolutionary processes (Price 2003); which is
evidenced by the presence of conserved backbones in food webs (Dalla
Riva \& Stouffer 2016\DIFaddbegin \DIFadd{; Mora \emph{et al.} 2018}\DIFaddend ), strong evolutionary
signature on prey choice (Stouffer \emph{et al.} 2012), and strong
phylogenetic signature in food web intervality (Eklöf \& Stouffer 2016).
Phylogenetic reconstruction has also previously been used within the
context of ecological networks, namely understanding ancestral
plant-insect interactions (Braga \emph{et al.} 2021). Taken together,
these considerations suggest that phylogenies can reliably be used to
transfer knowledge on species interactions.

Our \DIFaddbegin \DIFadd{methodology is outlined in fig.~\ref{fig:concept}, where we provide
an illustration based on learning the embedding of a metaweb of trophic
interactions for European mammals (known interactions; Maiorano \emph{et
al.} 2020b, a) and, based on phylogenetic relationships between mammals
globally (\emph{i.e.}, phylogenetic tree Upham \emph{et al.} 2019),
infer a metaweb for the Canadian mammalian species pool (interactions
are treated as unknown in this instance). Following the definition of
Dunne (2006), a metaweb is a network analogue to the regional species
pool; specifically, it is an inventory of all \emph{potential}
interactions between species from a spatially delimited area (and so
captures the \(\gamma\) diversity of interactions). The metaweb is,
therefore, \emph{not} a prediction of the food web at a specific locale
within the spatial area it covers, and will have a different structure
(notably by having a larger connectance; see \emph{e.g.} Wood \emph{et
al.} 2015). These local food webs (which captures the \(\alpha\)
diversity of interactions) are a subset of the metaweb's species and
interactions, and have been called ``metaweb realizations'' (Poisot
\emph{et al.} 2015). Differences between local food web and their
metaweb are due to chance, species abundance and co-occurrence, local
environmental conditions, and local distribution of functional traits,
among others.
}

\DIFadd{Because the metaweb represents the joint effect of functional,
phylogenetic, and macroecological processes (Morales-Castilla \emph{et
al.} 2015), it holds valuable ecological information. Specifically, it
is the ``upper bounds'' on what the composition of the local networks
can be (see \emph{e.g.} McLeod \emph{et al.} 2021). These local
networks, in turn, can be reconstructed given appropriate knowledge of
local species composition, providing information on structure of food
webs at finer spatial scales. This has been done for example for
tree-galler-parasitoid systems (Gravel \emph{et al.} 2018), fish trophic
interactions (Albouy \emph{et al.} 2019), tetrapod trophic interactions
(O'Connor \emph{et al.} 2020), and crop-pest networks (Grünig \emph{et
al.} 2020). Whereas the original metaweb definition, and indeed most
past uses of metawebs, was based on the presence/absence of
interactions, we focus on \emph{probabilistic} metawebs where
interactions are represented as the chance of success of a Bernoulli
trial (see \emph{e.g.} Poisot \emph{et al.} 2016); therefore, not only
does our method recommend interactions that may exist, it gives each
interaction a score, allowing us to properly weigh them.
}

\DIFadd{Our }\DIFaddend case study shows that phylogenetic transfer learning is \DIFdelbegin \DIFdel{indeed }\DIFdelend an effective
approach to \DIFdelbegin \DIFdel{predict the Canadian mammalian metaweb}\DIFdelend \DIFaddbegin \DIFadd{the generation of probabilistic metawebs}\DIFaddend . This showcases
that although the components (species) that make up the Canadian and
European communities may be \emph{minimally} shared \DIFaddbegin \DIFadd{(the overall species
overlap is less than 4\%)}\DIFaddend , if the medium (proxy space) selected in the
transfer step is biologically plausible, we can still effectively learn
from the known network and make biologically relevant predictions of
interactions. \DIFaddbegin \DIFadd{Indeed, as we detail in the results, when validated
against known but fractional data of trophic interactions between
Canadian mammals, our model achieves a predictive accuracy of
approximately 91\%. }\DIFaddend It should be reiterated that the framework presented
in fig.~\ref{fig:concept} is amenable to changes; notably, the measure
of similarity may not be phylogeny, and can be replaced by information
on foraging (Beckerman \emph{et al.} 2006), cell-level mechanisms
(Boeckaerts \emph{et al.} 2021), or a combination of traits and
phylogenetic structure (Stock 2021). \DIFaddbegin \DIFadd{Most importantly, although we focus
on a trophic system, it is an established fact that different
(non-trophic) interactions do themselves interact with and influence the
outcome of trophic interactions (Kéfi \emph{et al.} 2012; see
\emph{e.g.} Kawatsu \emph{et al.} 2021). Future development of metaweb
inference techniques should cover the prediction of multiple interaction
types.
}\DIFaddend 

\hypertarget{data-used-for-the-case-study}{%
\section{Data used for the case
study}\label{data-used-for-the-case-study}}

We use data from the European metaweb assembled by Maiorano \emph{et
al.} (2020b)\DIFdelbegin \DIFdel{, following the definition of the metaweb first introduced
by Dunne (2006), \emph{i.e.} an inventory of all possible interactions
within species from
a spatially delimited pool. Notably the metaweb is
not a prediction of the food web at any specific locale within the
frontiers of the species pool -- in fact,
these local food webs are
expected to have a subset of both the species and }\DIFdelend \DIFaddbegin \DIFadd{. This was assembled using data extracted from scientific
literature (including published papers, books, and grey literature) from
}\DIFaddend the \DIFdelbegin \DIFdel{interactions of
their metaweb (Poisot \emph{et al.} 2012) . This being said, as the
metaweb represents the total of functional, phylogenetic, and
macroecological processes (Morales-Castilla }\DIFdelend \DIFaddbegin \DIFadd{last 50 years and includes all terrestrial tetrapods (mammals,
breeding birds, reptiles and amphibians) occurring on the European
sub-continent (and Turkey) - with the caveat that only species
introduced in historical times and currently naturalized being included.
This metaweb itself is a network of binary (}\DIFaddend \emph{\DIFdelbegin \DIFdel{et al}\DIFdelend \DIFaddbegin \DIFadd{i.e}\DIFaddend .}
\DIFdelbegin \DIFdel{2015), it is
thus still worthy of ecological attention.
We deduced the }\DIFdelend \DIFaddbegin \DIFadd{presence/absence), potential two-way interactions between species pairs.
}

\DIFadd{We filtered down the European metaweb to create a }\DIFaddend subgraph corresponding
to all mammals by matching species names in the original network to the
\DIFdelbegin \DIFdel{GBIF }\DIFdelend \DIFaddbegin \DIFadd{Global Biodiversity Information Facility (GBIF) }\DIFaddend taxonomic backbone (GBIF
Secretariat 2021) and retaining all those who matched to mammals. This
serves a dual purpose 1) to extract only mammals from the European
network and 2) to match and standardize species names when aggregating
the different data sources further downstream (which is an important
consideration when combining datasets (Grenié \emph{et al.} 2021)). All
nodes had valid matches to GBIF at this step, and so this backbone is
used for all name reconciliation steps as outlined below.

The European metaweb represents the knowledge we want to learn and
transfer; the phylogenetic similarity of mammals here represents the
information for transfer \DIFaddbegin \DIFadd{(\emph{i.e.} the transfer medium)}\DIFaddend . We used the
mammalian consensus supertree by Upham \emph{et al.} (2019), for which
all approximatively 6000 names have been similarly matched to their GBIF
valid names. This step allows us to place each node of the mammalian
European metaweb in the phylogeny.

The destination problem to which we want to transfer knowledge is the
trophic interactions between mammals in Canada. We obtained the list of
extant species from the \DIFdelbegin \DIFdel{IUCN }\DIFdelend \DIFaddbegin \DIFadd{International Union for Conservation of Nature
(IUCN) }\DIFaddend checklist, and selected the terrestrial and semi-aquatic species
(this corresponds to the same selection that was applied by Maiorano
\emph{et al.} (2020b) in the European metaweb). The IUCN names were, as
previously, reconciled against GBIF to have an exact match to the
taxonomy.

After taxonomic cleaning and reconciliation as outlined in the following
sections, the mammalian European metaweb has 260 species, and the
Canadian species pool has 163; of these, 17 (about 4\% of the total) are
shared, and 89 species from Canada (54\%) had at least one congeneric
species in Europe. The similarity for both species pools predictably
increases with higher taxonomic order, with 19\% of shared genera, 47\%
of shared families, and 75\% of shared orders; for the last point,
Canada and Europe each had a single unique order (\emph{Didelphimorphia}
for Canada, \emph{Erinaceomorpha} for Europe).

In the following sections, we describe the representational learning
step applied to European data, the transfer step through phylogenetic
similarity, and the generation of a probabilistic metaweb for the
destination species pool.

\hypertarget{method-description}{%
\section{Method description}\label{method-description}}

The \DIFdelbegin \DIFdel{crux of the }\DIFdelend \DIFaddbegin \DIFadd{core point of our }\DIFaddend method is the transfer of knowledge of a known
\DIFaddbegin \DIFadd{ecological }\DIFaddend network, in order to predict interactions between species
from another location \DIFaddbegin \DIFadd{at which the network is unknown (or partially
known)}\DIFaddend . In fig.~\ref{fig:concept}, we give a high-level overview of the
approach; in the example around which this manuscript is built
(leveraging detailed knowledge about binary trophic interactions between
Mammalia in Europe to predict the less known trophic interactions
between closely phylogenetically related Mammalia in Canada), we use a
series of specific steps for network embedding, trait inference, network
prediction and thresholding.

Specifically, our approach can be summarized as follows: from the known
network in Europe, we use a truncated Singular Value Decomposition
(t-SVD; Halko \emph{et al.} 2011) to generate latent traits representing
a low-dimensional embedding of the network\DIFdelbegin \DIFdel{; these }\DIFdelend \DIFaddbegin \DIFadd{. As an aside, most ecologists
are indirectly familiar with SVD: Principal Component Analysis is a
special case of SVD, which is more sensitive to numerical instabilities
(see notably Shlens 2014). The latent }\DIFaddend traits give an unbiased estimate
of the node's position in the latent feature spaces \DIFdelbegin \DIFdel{.
Then, we map these latent traits }\DIFdelend \DIFaddbegin \DIFadd{and can be mapped
}\DIFaddend onto a reference phylogeny (other distance-based measures of species
proximity that allow for the inference of features in the latent space
can be used, for example the dissimilarity in functional traits). Based
on the reconstructed latent traits for species in the destination
species pool, a Random Dot Product Graph model (hereafter RDPG; Young \&
Scheinerman 2007) predicts the interaction between species through a
function of the nodes' features through matrix multiplication. Thus,
from latent traits and node position, we can infer interactions.

\DIFaddbegin \DIFadd{The method we develop is, ecologically speaking, a ``black box,''
\emph{i.e.} an algorithm that can be understood mathematically, but
whose component parts are not always directly tied to ecological
processes. There is a growing realization in machine learning that
(unintentional) black box algorithms are not necessarily a bad thing
(Holm 2019), as long as their constituent parts can be examined (which
is the case with our method). But more importantly, data hold more
information than we might think; as such, even algorithms that are
disconnected from the model can make correct guesses most of the time
(Halevy \emph{et al.} 2009); in fact, in an instance of ecological
forecasting of spatio-temporal systems, model-free approaches
(\emph{i.e.} drawing all of their information from the data)
outperformed model-informed ones (Perretti \emph{et al.} 2013).
}

\DIFaddend \hypertarget{implementation-and-code-availability}{%
\subsection{Implementation and code
availability}\label{implementation-and-code-availability}}

The entire pipeline is implemented in \emph{Julia} 1.6 (Bezanson
\emph{et al.} 2017) and is available under the permissive MIT License at
\href{https://osf.io/2zwqm/}{\texttt{https://osf.io/2zwqm/}}. The
taxonomic cleanup steps are done using \texttt{GBIF.jl} (Dansereau \&
Poisot 2021). The network embedding and analysis is done using
\texttt{EcologicalNetworks.jl} (Poisot \emph{et al.} 2019; Banville
\emph{et al.} 2021). The phylogenetic simulations are done using
\texttt{PhyloNetworks.jl} (Solís-Lemus \emph{et al.} 2017) and
\texttt{Phylo.jl} (Reeve \emph{et al.} 2016). A complete
\texttt{Project.toml} file specifying the full tree of dependencies is
available alongside the code. This material also includes a fully
annotated copy of the entire code required to run this project
(describing both the intent of the code and discussing some technical
implementation details), a vignette for every step of the process, and a
series of Jupyter notebooks with the text and code. The pipeline can be
executed on a laptop in a matter of minutes, and therefore does not
require extensive computational power.

\hypertarget{step-1-learning-the-origin-network-representation}{%
\subsection{Step 1: Learning the origin network
representation}\label{step-1-learning-the-origin-network-representation}}

The first step in transfer learning is to learn the structure of the
original dataset. In order to do so, we rely on an approach inspired
from representational learning, where we learn a \emph{representation}
of the metaweb (in the form of the latent subspaces), rather than a list
of interactions (species \emph{a} eats \emph{b}). This approach is
conceptually different from other metaweb-scale predictions (\emph{e.g.}
Albouy \emph{et al.} 2019), in that the metaweb representation is easily
transferable. Specifically, we use RDPG to create a number of latent
variables that can be combined into an approximation of the network
adjacency matrix. RDPG results are known to have strong phylogenetic
signal, and to capture the evolutionary backbone of food webs (Dalla
Riva \& Stouffer 2016)\DIFaddbegin \DIFadd{; in other words, the latent variables of an RDPG
can be mapped onto a phylogenetic tree, and phylogenetically similar
predators should share phylogenetically similar preys}\DIFaddend . In addition,
recent advances show that the latent variables produced this way can be
used to predict \emph{de novo} network edges\DIFaddbegin \DIFadd{. Interestingly, the latent
variables do not need to be produced by decomposing the network itself;
in a recent contribution, Runghen \emph{et al.} }\DIFaddend (\DIFdelbegin \DIFdel{\emph{i.e.} interactions;
Runghen \emph{et al.} }\DIFdelend 2021) \DIFaddbegin \DIFadd{showed that deep
artificial neural networks are able to reconstruct the left and right
subspaces of an RDPG, in order to predict human movement networks from
individual/location metadata. This is an exciting opportunity, as it
opens up the possibility of using additional metadata as predictors}\DIFaddend .

The latent variables are created by performing a truncated Singular
Value Decomposition (t-SVD) on the adjacency matrix. SVD is an
appropriate embedding of ecological networks, which has recently been
shown to both capture their complex, emerging properties (Strydom
\emph{et al.} 2021b) and to allow highly accurate prediction of the
interactions within a single network (Poisot \emph{et al.} 2021b). Under
SVD, an adjacency matrix \(\mathbf{A}\) (where
\(\mathbf{A}_{m,n}\in\mathbb{B}\) where 1 indicates predation and 0 an
absence thereof) is decomposed into three components resulting in
\DIFdelbegin \DIFdel{\(\mathbf{A} = \mathbf{L}\mathbf{\Sigma}\mathbf{R'}.\) }\DIFdelend \DIFaddbegin \DIFadd{\(\mathbf{A} = \mathbf{U}\mathbf{\Sigma}\mathbf{V'}.\) }\DIFaddend Here,
\(\mathbf{\Sigma}\) is a \(m \times n\) diagonal matrix and contains
only singular (\(\sigma\)) values along its diagonal, \DIFdelbegin \DIFdel{\(\mathbf{L}\) }\DIFdelend \DIFaddbegin \DIFadd{\(\mathbf{U}\) }\DIFaddend is
a \(m \times m\) unitary matrix, and \DIFdelbegin \DIFdel{\(\mathbf{R}'\) }\DIFdelend \DIFaddbegin \DIFadd{\(\mathbf{V}'\) }\DIFaddend a \(n \times n\)
unitary matrix. Truncating the SVD removes additional noise in the
dataset by omitting non-zero and/or smaller \(\sigma\) values from
\(\mathbf{\Sigma}\) using the rank of the matrix. Under a t-SVD
\(\mathbf{A}_{m,n}\) is decomposed so that \(\mathbf{\Sigma}\) is a
square \(r \times r\) diagonal matrix (whith \(1 \le r \le r_{full}\)
where \(r_{full}\) is the full rank of \(\mathbf{A}\) and \(r\) the rank
at which we truncate the matrix) containing only non-zero \(\sigma\)
values. Additionally, \DIFdelbegin \DIFdel{\(\mathbf{L}\) }\DIFdelend \DIFaddbegin \DIFadd{\(\mathbf{U}\) }\DIFaddend is now a \(m \times r\) semi
unitary matrix and \DIFdelbegin \DIFdel{\(\mathbf{R}'\) }\DIFdelend \DIFaddbegin \DIFadd{\(\mathbf{V}'\) }\DIFaddend a \(n \times r\) semi-unitary matrix.

The specific rank at which the SVD ought to be truncated is a difficult
question. The purpose of SVD is to remove the noise (expressed at high
dimensions) and to focus on the signal, (expressed at low dimensions).
In datasets with a clear signal/noise demarcation, a scree plot of
\(\mathbf{\Sigma}\) can show a sharp drop at the rank where noise starts
(Zhu \& Ghodsi 2006). Because the European metaweb is almost entirely
known, the amount of noise (uncertainty) is low; this is reflected in
fig.~\ref{fig:scree} (left), where the scree plot shows no important
drop, and in fig.~\ref{fig:scree} (right) where the proportion of
variance explained increases smoothly at higher dimensions. For this
reason, we default back to a threshold that explains 60\% of the
variance in the underlying data, corresponding to 12 dimensions -
\emph{i.e.} a tradeoff between accuracy and a reduced number of
features.

\DIFdelbegin \DIFdel{A }\DIFdelend \DIFaddbegin \DIFadd{An }\DIFaddend RDPG estimates the probability of observing interactions between
nodes (species) as a function of the nodes' latent variables\DIFaddbegin \DIFadd{, and is a
way to turn a SVD (which decompose one matrix into three) into two
matrices that can be multiplied to provide an approximation of the
network}\DIFaddend . The latent variables used for the RDPG, called the left and
right subspaces, are defined as
\DIFdelbegin \DIFdel{\(\mathscr{L} = \mathbf{L}\sqrt{\mathbf{\Sigma}}\), and
\(\mathscr{R} = \sqrt{\mathbf{\Sigma}}\mathbf{R}\) }\DIFdelend \DIFaddbegin \DIFadd{\(\mathscr{L} = \mathbf{U}\sqrt{\mathbf{\Sigma}}\), and
\(\mathscr{R} = \sqrt{\mathbf{\Sigma}}\mathbf{V}'\) }\DIFaddend -- using the full
rank of \(\mathbf{A}\), \DIFdelbegin \DIFdel{\(\mathscr{L}\mathscr{R}' = \mathbf{A}\)}\DIFdelend \DIFaddbegin \DIFadd{\(\mathscr{L}\mathscr{R} = \mathbf{A}\)}\DIFaddend , and
using any smaller rank results in
\DIFdelbegin \DIFdel{\(\mathscr{L}\mathscr{R}' \approx \mathbf{A}\)}\DIFdelend \DIFaddbegin \DIFadd{\(\mathscr{L}\mathscr{R} \approx \mathbf{A}\)}\DIFaddend . Using a rank of 1 for the
t-SVD provides a first-order approximation of the network. \DIFaddbegin \DIFadd{One advantage
of using a RDPG rather than a SVD is that the number of components to
estimate decreases; notably, one does not have to estimate the singular
values of the SVD. Furthermore, the two subspaces can be directly
multiplied to yield a network.
}\DIFaddend 

Because RDPG relies on matrix multiplication, the higher dimensions
essentially serve to make specific interactions converge towards 0 or 1;
therefore, for reasonably low ranks, there is no guarantee that the
values in the reconstructed network will be within the unit range. In
order to determine what constitutes an appropriate threshold for
probability, we performed the RDPG approach on the European metaweb, and
evaluated the probability threshold by treating this as a binary
classification problem, specifically assuming that both 0 and 1 in the
European metaweb are all true. Given the methodological details given in
Maiorano \emph{et al.} (2020b) and O'Connor \emph{et al.} (2020), this
seems like a reasonable assumption, although one that does not hold for
all metawebs. We used the thresholding approach presented in Poisot
\emph{et al.} (2021b), and picked a cutoff that maximized Youden's \(J\)
statistic (a measure of the informedness (trust) of predictions; Youden
(1950)); the resulting cutoff was 0.22, and gave an accuracy above 0.99.
\DIFaddbegin \DIFadd{In Supp. Mat. 1, we provide several lines of evidence that using the
entire network to estimate the threshold does not lead to overfitting;
that using a subset of species would yield the same threshold; that
decreasing the quality of the original data by adding of removing
interactions would minimally affect the predictive accuracy of RDPG
applied to the European metaweb; and that the networks reconstructed
from artificially modified data are reconstructed with the correct
ecological properties.
}\DIFaddend 

The left and right subspaces for the European metaweb, accompanied by
the threshold for prediction, represent the knowledge we seek to
transfer. In the next section, we explain how we rely on phylogenetic
similarity to do so.

\hypertarget{steps-2-and-3-transfer-learning-through-phylogenetic-relatedness}{%
\subsection{Steps 2 and 3: Transfer learning through phylogenetic
relatedness}\label{steps-2-and-3-transfer-learning-through-phylogenetic-relatedness}}

In order to transfer the knowledge from the European metaweb to the
Canadian species pool, we performed ancestral character estimation using
a Brownian motion model, which is a conservative approach in the absence
of strong hypotheses about the nature of phylogenetic signal in the
network decomposition (Litsios \& Salamin 2012). This uses the estimated
feature vectors for the European mammals to create a state
reconstruction for all species (conceptually something akin to a
trait-based mammalian phylogeny using \DIFaddbegin \DIFadd{latent }\DIFaddend generality and
vulnerability traits) and allows us to impute the missing (latent) trait
data for the Canadian species that are not already in the European
network; as we are focused on predicting contemporary interactions, we
only retained the values for the tips of the tree. We assumed that all
traits (\emph{i.e.} the feature vectors for the left and right
subspaces) were independent, which is a reasonable assumption as every
trait/dimension added to the t-SVD has an \emph{additive} effect to the
one before it. Note that the Upham \emph{et al.} (2019) tree itself has
some uncertainty associated to inner nodes of the phylogeny. In this
case study, we have decided to not propagate this uncertainty, as it
would complexify the process. The Brownian motion algorithm returns the
\emph{average} value of the trait, and its upper and lower bounds.
Because we do not estimate other parameters of the traits'
distributions, we considered that every species trait is represented as
a uniform distribution between these bounds\DIFdelbegin \DIFdel{; in a situation where the algorithm would return point values for all simulations, one could in theory either estimate the
parameters of a
distribution for each tip, or draw randomly from the outputs. In all
cases, }\DIFdelend \DIFaddbegin \DIFadd{. The choice of the uniform
distribution was made because the algorithm returns a minimum and
maximum point estimate for the value, and given this information, }\DIFaddend the
\DIFaddbegin \DIFadd{uniform distribution is the one with maximum entropy. Had all mean
parameters estimates been positive, the exponential distribution would
have been an alternative, but this is not the case for the subspaces of
an RDPG. In order to examine the consequences of the choice of
distribution, we estimated the variance per latent variable per node to
use a Normal distribution; as we show in Supp. Mat. 2, this decision
results in dramatically over-estimating the number and probability of
interactions, and therefore we keep the discussions in the main text to
the uniform case. The }\DIFaddend inferred left and right \DIFdelbegin \DIFdel{sub-spaces }\DIFdelend \DIFaddbegin \DIFadd{subspaces }\DIFaddend for the Canadian
species pool (\(\hat{\mathscr{L}}\) and \(\hat{\mathscr{R}}\)) have
entries that are distributions, representing the range of values for a
given species at a given dimension.

These objects represent the transferred knowledge, which we can use for
prediction of the Canadian metaweb.

\hypertarget{step-4-probabilistic-prediction-of-the-destination-network}{%
\subsection{Step 4: Probabilistic prediction of the destination
network}\label{step-4-probabilistic-prediction-of-the-destination-network}}

The phylogenetic reconstruction of \(\hat{\mathscr{L}}\) and
\(\hat{\mathscr{R}}\) has an associated uncertainty, represented by the
breadth of the uniform distribution associated to each of their entries.
Therefore, we can use this information to assemble a
\emph{probabilistic} metaweb in the sense of Poisot \emph{et al.}
(2016), \emph{i.e.} in which every interaction is represented as a
single, independent, Bernoulli event of probability \(p\).

Specifically, we have adopted the following approach. For every entry in
\(\hat{\mathscr{L}}\) and \(\hat{\mathscr{R}}\), we draw a value from
its distribution. This results in one instance of the possible left
(\(\hat{\mathscr{l}}\)) and right (\(\hat{\mathscr{r}}\)) subspaces for
the Canadian metaweb. These can be multiplied, to produce one matrix of
real values. Because the entries in \(\hat{\mathscr{l}}\) and
\(\hat{\mathscr{r}}\) are in the same space where \(\mathscr{L}\) and
\(\mathscr{R}\) were originally predicted, it follows that the threshold
\(\rho\) estimated for the European metaweb also applies. We use this
information to produce one random Canadian metaweb,
\DIFdelbegin \DIFdel{\(N = \hat{\mathscr{L}}\)\(\hat{\mathscr{R}}' \ge \rho\)}\DIFdelend \DIFaddbegin \DIFadd{\(N = \hat{\mathscr{L}}\hat{\mathscr{R}}' \ge \rho\)}\DIFaddend . As we can see in
(fig.~\ref{fig:subspaces})\DIFaddbegin \DIFadd{, }\DIFaddend the European and Canadian metawebs are
structurally similar (as would be expected given the biogeographic
similarities) and the two (left and right) subspaces are distinct
\emph{i.e.} capturing predation (generality) and prey (vulnerability)
\DIFaddbegin \DIFadd{latent }\DIFaddend traits.

Because the intervals around some trait values can be broad (in fact,
probably broader than what they would actually be, see \emph{e.g.}
Garland \emph{et al.} 1999), we repeat the above process
\(2\times 10^5\) times, which results in a probabilistic metaweb \(P\),
where the probability of an interaction (here conveying our degree of
trust that it exists given the inferred trait distributions) is given by
the number of times where it appears across all random draws \(N\),
divided by the number of samples. An interaction with \(P_{i,j} = 1\)
means that these two species were predicted to interact in all
\(2\times 10^5\) random draws.

\DIFaddbegin \DIFadd{It must be noted that despite bringing in a large amount of information
from the European species pool and interactions, the Canadian metaweb
has distinct structural properties. Following an approach similar to
Vermaat \emph{et al.} (2009), we show in Supp. Mat. 3 that not only can
we observe differences in a multivariate space between the European and
Canadian metaweb, we can also observe differences in the same space
between random subgraphs from these networks. These results line up with
the studies spatializing metawebs that have been discussed in the
introduction: changes in the species pool are driving local structural
changes in the networks.
}

\DIFaddend \hypertarget{data-cleanup-discovery-validation-and-thresholding}{%
\subsection{Data cleanup, discovery, validation, and
thresholding}\label{data-cleanup-discovery-validation-and-thresholding}}

Once the probabilistic metaweb for Canada has been produced, we followed
a number of data inflation steps to finalize it. This step is external
to the actual transfer learning framework but rather serves as a way to
augment and validate the predicted metaweb.

First, we extracted the subgraph corresponding to the 17 species shared
between the European and Canadian pools and replaced these interactions
with a probability of 0 (non-interaction) or 1 (interaction), according
to their value in the European metaweb. This represents a minute
modification of the inferred network (about 0.8\% of all species pairs
from the Canadian web), but ensures that we are directly re-using
knowledge from Europe.

Second, we looked for all species in the Canadian pool known to the
Global Biotic Interactions (GLoBI) database (Poelen \emph{et al.} 2014),
and extracted their known interactions. Because \DIFdelbegin \DIFdel{GLOBI }\DIFdelend \DIFaddbegin \DIFadd{GLoBI }\DIFaddend aggregates
observed interactions, it is not a \emph{networks} data source, and
therefore the only information we can reliably extract from it is that a
species pair \emph{was reported to interact at least once}. This last
statement should yet be taken with caution, as some sources in \DIFdelbegin \DIFdel{GLOBI
}\DIFdelend \DIFaddbegin \DIFadd{GLoBI
}\DIFaddend (\emph{e.g.} Thessen \& Parr 2014) are produced through text analysis,
and therefore may not document direct evidence of the interaction.
Nevertheless, should the predictive model work, we would expect that a
majority of interactions known to \DIFdelbegin \DIFdel{GLOBI }\DIFdelend \DIFaddbegin \DIFadd{GLoBI }\DIFaddend would also be predicted. \DIFdelbegin \DIFdel{After
performing this check, we set the probability of all interactions known
to GLOBI (}\DIFdelend \DIFaddbegin \DIFadd{We
retrieved }\DIFaddend 366 \DIFdelbegin \DIFdel{in total}\DIFdelend \DIFaddbegin \DIFadd{interactions between mammals from the Canadian species
pool from GLoBI}\DIFaddend , 33 of which were not predicted by the model\DIFdelbegin \DIFdel{, for
}\DIFdelend \DIFaddbegin \DIFadd{; this
results in }\DIFaddend a success rate of 91\%\DIFdelbegin \DIFdel{) }\DIFdelend \DIFaddbegin \DIFadd{. After performing this check, we set
the probability of all interactions known to GLoBI }\DIFaddend to 1.

Finally, we downloaded the data from Strong \& Leroux (2014), who mined
various literature sources to identify trophic interactions in
Newfoundland. This dataset documented 25 interactions between mammals,
only two of which were not part of our (Canada-level) predictions,
resulting in a success rate of 92\%. These two interactions were added
to our predicted metaweb with a probability of 1. A table listing all
interactions in the predicted Canadian metaweb can be found in the
supplementary material.

Because the confidence intervals on the inferred trait space are
probably over-estimates, we decided to apply a thresholding step to the
interactions after the data inflation (fig.~\ref{fig:thresholds}).
Cirtwill \& Hambäck (2021) proposed a number of strategies to threshold
probabilistic networks. Their methods assume the underlying data to be
tag-based sequencing, which represents interactions as co-occurrences of
predator and prey within the same tags; this is conceptually identical
to our Bernoulli-trial based reconstruction of a probabilistic network.
We performed a full analysis of the effect of various cutoffs, and as
they either resulted in removing too few interactions, or removing
enough interactions that species started to be disconnected from the
network, we set this threshold for a probability equivalent to 0 to the
largest possible value that still allowed all species to have at least
one interaction with a non-zero probability. The need for this slight
deviation from the Cirtwill \& Hambäck (2021) method highlights the need
for additional development on network thresholding.

\hypertarget{results-and-discussion-of-the-case-study}{%
\section{Results and discussion of the case
study}\label{results-and-discussion-of-the-case-study}}

In fig.~\ref{fig:thresholds}, we examine the effect of varying the
cutoff on \(P(i \rightarrow j)\) on the number of links, species, and
connectance. Determining a cutoff using the maximum curvature, or
central difference approximation of the second order partial derivative,
as suggested by \emph{e.g.} Cirtwill \& Hambäck (2021), results in
species being lost, or almost all links being kept. We therefore settled
on the value that allowed all species to remain with at least one
interaction. This result, in and of itself, suggests that additional
methodological developments for the thresholding of probabilistic
networks are required.


The t-SVD embedding is able to learn relevant ecological features for
the network. fig.~\ref{fig:degree} shows that the first rank correlates
linearly with generality and vulnerability (Schoener 1989), \emph{i.e.}
the number of preys and predators \DIFaddbegin \DIFadd{for each species}\DIFaddend . Importantly, this
implies that a rank 1 approximation represents the configuration model
for the metaweb, \emph{i.e.} a set of random networks generated from a
given degree sequence (Park \& Newman 2004). Accounting for the
probabilistic nature of the degrees, the rank 1 approximation also
represents the \emph{soft} configuration model (van der Hoorn \emph{et
al.} 2018). Both models are maximum entropy graph models (Garlaschelli
\emph{et al.} 2018), with sharp (all network realizations satisfy the
specified degree sequence) and soft (network realizations satisfy the
degree sequence on average) local constraints, respectively. The (soft)
configuration model is an unbiased random graph model widely used by
ecologists in the context of null hypothesis significance testing of
network structure (\emph{e.g.} Bascompte \emph{et al.} 2003) and can
provide informative priors for Bayesian inference of network structure
(\emph{e.g.} Young \emph{et al.} 2021). It is noteworthy that for this
metaweb, the relevant information was extracted at the first rank.
Because the first rank corresponds to the leading singular value of the
system, the results of fig.~\ref{fig:degree} have a straightforward
interpretation: degree-based processes are the most important in
structuring the mammalian food web.

\hypertarget{discussion}{%
\section{Discussion}\label{discussion}}

One important aspect in which Europe and Canada differ (despite their
comparable bioclimatic conditions) is the degree of the legacy of human
impacts, which have been much longer in Europe. Nenzén \emph{et al.}
(2014) showed that even at small scales (the Iberian peninsula), mammal
food webs retain the signal of both climate change and human activity,
even when this human activity was orders of magnitude less important
than it is now. Similarly, Yeakel \emph{et al.} (2014) showed that
changes in human occupation over several centuries can lead to food web
collapse. Megafauna in particular seems to be very sensitive to human
arrival (Pires \emph{et al.} 2015). In short, there is
well-substantiated support for the idea that human footprint affects
more than the risk of species extinction (Marco \emph{et al.} 2018), and
can lead to changes in interaction structure. Yet, owing to the inherent
plasticity of interactions, there have been documented instances of food
webs undergoing rapid collapse/recovery cycles over short periods of
time (Pedersen \emph{et al.} 2017). The embedding of a network, in a
sense, embeds its macro-evolutionary history, especially as RDPG
captures ecological signal (Dalla Riva \& Stouffer 2016); at this point,
it is important to recall that a metaweb is intended as a catalogue of
all \DIFdelbegin \DIFdel{possible }\DIFdelend \DIFaddbegin \DIFadd{potential }\DIFaddend interactions, which should then be filtered
(Morales-Castilla \emph{et al.} 2015). In practice (and in this
instance) the reconstructed metaweb will predict interactions that are
plausible based on the species' evolutionary history, however some
interactions would\DIFaddbegin \DIFadd{/would }\DIFaddend not be realized due to human impact.

\DIFdelbegin \DIFdel{Cirtwill \emph{et al.} (}\DIFdelend \DIFaddbegin \DIFadd{Dallas \emph{et al.} (2017) suggested that most links in ecological
networks may be cryptic, \emph{i.e.} uncommon or otherwise hard to
observe. This argument essentially echoes Jordano (2016b): the sampling
of ecological interactions is difficult because it requires first the
joint observation of two species, and then the observation of their
interaction. In addition, it is generally expected that weak or rare
links would be more common in networks (Csermely 2004), compared to
strong, persistent links; this is notably the case in food chains,
wherein many weaker links are key to the stability of a system (Neutel
\emph{et al.} 2002). In the light of these observations, the results in
fig.~\ref{fig:inflation} are not particularly surprising: we expect to
see a surge in these low-probability interactions under a model that has
a good predictive accuracy. Because the predictions we generate are by
design probabilistic, then one can weigh these rare links appropriately.
In a sense, that most ecological interactions are elusive can call for a
slightly different approach to sampling: once the common interactions
are documented, the effort required in documenting each rare interaction
may increase exponentially. Recent proposals suggest that machine
learning algorithms, in these situations, can act as data generators
(Hoffmann \emph{et al.} }\DIFaddend 2019)\DIFdelbegin \DIFdel{previously made the point }\DIFdelend \DIFaddbegin \DIFadd{: in this perspective, high quality
observational data can be supplemented with synthetic data coming from
predictive models, which increases the volume of information available
for inference. Indeed, Strydom \emph{et al.} (2021a) suggested that
knowing the metaweb may render the prediction of local networks easier,
because it fixes an ``upper bound'' on which interactions can exist;
indeed, with a probabilistic metaweb, we can consider that the metaweb
represents an aggregation of informative priors on the interactions.
}

\DIFadd{Related to the last point, Cirtwill \emph{et al.} (2019) showed }\DIFaddend that
network inference techniques based on Bayesian approaches would perform
far better in the presence of an interaction-level informative prior;
the desirable properties of such a prior would be that it is expressed
as a probability, preferably representing a Bernoulli event, the value
of which would be representative of relevant biological processes
(probability of predation in this case). We argue that the probability
returned at the very last step of our framework may serve as this
informative prior; indeed, the output of our analysis can be used in
subsequent steps, also possibly involving expert elicitation to validate
some of the most strongly recommended interactions. One important
\emph{caveat} to keep in mind when working with interaction inference is
that interactions can never really be true negatives (in the current
state of our methodological framework and data collection limitations);
this renders the task of validating a model through the usual
application of binary classification statistics very difficult (although
see Strydom \emph{et al.} 2021a for a discussion of alternative
suggestions). The other way through which our framework can be improved
is by substituting the predictors that are used for transfer. For
example, in the presence of information on species traits that are known
to be predictive of species interactions, one might want to rely on
functional rather than phylogenetic distances -- in food webs, body size
(and allometrically related variables) has been established as such a
variable (Brose \emph{et al.} 2006); the identification of relevant
functional traits is facilitated by recent methodological developments
(Rosado \emph{et al.} 2013). It should be noted that Xing \& Fayle
(2021) highlight phylogenetic relatedness as one of the core components
of network comparison at the global scale. In this case study, we have
embedded the original metaweb using t-SVD, because it lends itself to \DIFdelbegin \DIFdel{a
}\DIFdelend \DIFaddbegin \DIFadd{an
}\DIFaddend RDPG reconstruction, which is known to capture the consequences of
evolutionary processes (Dalla Riva \& Stouffer 2016); this being said,
there are \DIFdelbegin \DIFdel{others }\DIFdelend \DIFaddbegin \DIFadd{other }\DIFaddend ways to embed graphs (Cai \emph{et al.} 2017; Arsov \&
Mirceva 2019; Cao \emph{et al.} 2019), which can be used as
alternatives.

As Herbert (1965) rightfully pointed out, ``{[}y{]}ou can't draw neat
lines around planet-wide problems''; in this regard, our approach \DIFaddbegin \DIFadd{(and
indeed, any inference of a metaweb at large scales) }\DIFaddend must contend with
\DIFdelbegin \DIFdel{two interesting }\DIFdelend \DIFaddbegin \DIFadd{several interesting and interwoven families of }\DIFaddend problems. The first is
the limit of the metaweb to embed and transfer. If the initial metaweb
is too narrow in scope, notably from a taxonomic point of view, the
chances of finding another area with enough related species to make a
reliable inference \DIFdelbegin \DIFdel{decrease. This }\DIFdelend \DIFaddbegin \DIFadd{decreases; this would likely be indicated by large
confidence intervals during ancestral character estimation, but the lack
of well documented metawebs is currently preventing the development of
more concrete guidelines. The question of phylogenetic relatedness and
dispersal }\DIFaddend is notably true if the metaweb is assembled in an area with
mostly endemic species\DIFaddbegin \DIFadd{, and as with every predictive algorithm, there is
room for the application of our best ecological judgement}\DIFaddend . Conversely,
the metaweb should be reliably filled, which assumes that the \(S^2\)
interactions in a pool of \(S\) species have been examined, either
through literature surveys or expert elicitation. \DIFdelbegin \DIFdel{The second problem is to determine }\DIFdelend \DIFaddbegin \DIFadd{Supp. Mat. 1 provides
some guidance as to the type of sampling effort that should be
prioritized. Although RDPG was able to maintain very high predictive
power when interactions were missing, the addition of false positive
interactions was immediately detected; this suggests that it may be
appropriate to err on the side of ``too many'' interactions when
constructing the initial metaweb to be transferred. The second series of
problems are related to determining }\DIFaddend which area should be used to infer
the new metaweb in, as this determines the species pool that must be
used. In our application, we focused on the mammals of Canada. The
upside of this approach is that information at the country level is
likely to be required by policy makers and stakeholders for their
biodiversity assessment, as each country tends to set goals at the
national level (Buxton \emph{et al.} 2021) for which quantitative
instruments are designed (Turak \emph{et al.} 2017), with specific
strategies often enacted at smaller scales (Ray \emph{et al.} 2021). \DIFdelbegin \DIFdel{Yet
these }\DIFdelend \DIFaddbegin \DIFadd{And
yet, we do not really have a satisfying answer to the question of
``where does a food web stop?''; the current most satisfying solutions
involve examining the spatial consistency of network area relationships
(see \emph{e.g.} Galiana \emph{et al.} 2018, 2019, 2021; Fortin \emph{et
al.} 2021), which is of course impossible in the absence of enough
information about the network itself. This suggests that an \emph{a
posteriori} refinement of the results may be required, based on a
downscaling of the metaweb. The final family of problems relates less to
the availability of data or quantitative tools, and more to the praxis
of spatial ecology. Operating under the context of }\DIFaddend national divisions,
in large parts of the world, \DIFdelbegin \DIFdel{reflect nothing except for }\DIFdelend \DIFaddbegin \DIFadd{reflects nothing more than }\DIFaddend the legacy of
settler colonialism\DIFdelbegin \DIFdel{, and operating under them
must be done under the clear realization that
they }\DIFdelend \DIFaddbegin \DIFadd{. Indeed, the use of ecological data is not an
apolitical act (Nost \& Goldstein 2021), as data infrastructures tend to
be designed to answer questions within national boundaries, and their
use both draws upon and reinforces territorial statecraft; as per Machen
\& Nost (2021), this is particularly true when the output of
``algorithmic thinking'' (\emph{e.g.} relying on machine learning to
generate knowledge) can be re-used for governance (\emph{e.g.} enacting
conservation decisions at the national scale). We therefore recognize
that methods such as we propose operate under the framework that
}\DIFaddend contributed to the ongoing biodiversity crisis (Adam 2014), \DIFdelbegin \DIFdel{can reinforce }\DIFdelend \DIFaddbegin \DIFadd{reinforced
}\DIFaddend environmental injustice (Choudry 2013; Domínguez \& Luoma 2020), and on
Turtle Island especially, \DIFdelbegin \DIFdel{will probably end up being }\DIFdelend \DIFaddbegin \DIFadd{should be }\DIFaddend replaced by Indigenous principles of
land management (Eichhorn \emph{et al.} 2019; No'kmaq \emph{et al.}
2021). \DIFaddbegin \DIFadd{As we see AI/ML being increasingly mobilized to generate
knowledge that is lacking for conservation decisions (\emph{e.g.} Lamba
\emph{et al.} 2019; Mosebo Fernandes \emph{et al.} 2020), our discussion
of these tools need to go beyond the technical, and into the governance
consequences they can have.
}\DIFaddend 

\textbf{Acknowledgements:} We acknowledge that this study was conducted
on land within the traditional unceded territory of the Saint Lawrence
Iroquoian, Anishinabewaki, Mohawk, Huron-Wendat, and Omàmiwininiwak
nations. TP, TS, DC, and LP received funding from the Canadian Institue
for Ecology \& Evolution. FB is funded by the \DIFdelbegin \DIFdel{Institut de Valorisation
des Données}\DIFdelend \DIFaddbegin \DIFadd{Institute for Data
Valorization (IVADO)}\DIFaddend . TS, SB, and TP are funded by a donation from the
Courtois Foundation. CB was awarded a Mitacs Elevate Fellowship no.
IT12391, in partnership with fRI Research, and also acknowledges funding
from Alberta Innovates and the Forest Resources Improvement Association
of Alberta. M-JF acknowledges funding from NSERC Discovery Grant and
NSERC CRC. RR is funded by New Zealand's Biological Heritage Ngā Koiora
Tuku Iho National Science Challenge, administered by New Zealand
Ministry of Business, Innovation, and Employment. BM is funded by the
NSERC Alexander Graham Bell Canada Graduate Scholarship and the FRQNT
master's scholarship. LP acknowledges funding from NSERC Discovery Grant
(NSERC RGPIN-2019-05771). TP acknowledges financial support from NSERC
through the Discovery Grants and Discovery Accelerator Supplement
programs.

\hypertarget{references}{%
\section*{References}\label{references}}
\addcontentsline{toc}{section}{References}

\hypertarget{refs}{}
\begin{CSLReferences}{1}{0}
\leavevmode\hypertarget{ref-Adam2014EleTre}{}%
Adam, R. (2014). \emph{Elephant treaties: The Colonial legacy of the
biodiversity crisis}. UPNE.

\leavevmode\hypertarget{ref-Albouy2019MarFis}{}%
Albouy, C., Archambault, P., Appeltans, W., Araújo, M.B., Beauchesne,
D., Cazelles, K., \emph{et al.} (2019). The marine fish food web is
globally connected. \emph{Nature Ecology \& Evolution}, 3, 1153--1161.

\leavevmode\hypertarget{ref-Arsov2019NetEmb}{}%
Arsov, N. \& Mirceva, G. (2019). \DIFdelbegin \DIFdel{Network Embedding: An Overview. \emph{arXiv:1911.11726 {[}cs, stat{]}}}\DIFdelend \DIFaddbegin \DIFadd{\emph{Network Embedding: An Overview}.
Available at: }\url{http://arxiv.org/abs/1911.11726}\DIFadd{. Last accessed}\DIFaddend .

\leavevmode\hypertarget{ref-Banville2021ManJl}{}%
Banville, F., Vissault, S. \& Poisot, T. (2021). Mangal.jl and
EcologicalNetworks.jl: Two complementary packages for analyzing
ecological networks in Julia. \emph{Journal of Open Source Software}, 6,
2721.

\leavevmode\hypertarget{ref-Bascompte2003NesAss}{}%
Bascompte, J., Jordano, P., Melian, C.J. \& Olesen, J.M. (2003). The
nested assembly of plant-animal mutualistic networks. \emph{Proceedings
of the National Academy of Sciences}, 100, 9383--9387.

\leavevmode\hypertarget{ref-Beckerman2006ForBio}{}%
Beckerman, A.P., Petchey, O.L. \& Warren, P.H. (2006). Foraging biology
predicts food web complexity. \emph{Proceedings of the National Academy
of Sciences}, 103, 13745--13749.

\leavevmode\hypertarget{ref-Bezanson2017JulFre}{}%
Bezanson, J., Edelman, A., Karpinski, S. \& Shah, V. (2017). Julia: A
Fresh Approach to Numerical Computing. \emph{SIAM Review}, 59, 65--98.

\leavevmode\hypertarget{ref-Boeckaerts2021PreBac}{}%
Boeckaerts, D., Stock, M., Criel, B., Gerstmans, H., De Baets, B. \&
Briers, Y. (2021). Predicting bacteriophage hosts based on sequences of
annotated receptor-binding proteins. \emph{Scientific Reports}, 11,
1467.

\leavevmode\hypertarget{ref-Braga2021PhyRec}{}%
Braga, M.P., Janz, N., Nylin, S., Ronquist, F. \& Landis, M.J. (2021).
Phylogenetic reconstruction of ancestral ecological networks through
time for pierid butterflies and their host plants. \emph{Ecology
Letters}, n/a.

\leavevmode\hypertarget{ref-Brose2006ConRes}{}%
Brose, U., Jonsson, T., Berlow, E.L., Warren, P., Banasek-Richter, C.,
Bersier, L.-F., \emph{et al.} (2006). ConsumerResource Body-Size
Relationships in Natural Food Webs. \emph{Ecology}, 87, 2411--2417.

\leavevmode\hypertarget{ref-Buxton2021KeyInf}{}%
Buxton, R.T., Bennett, J.R., Reid, A.J., Shulman, C., Cooke, S.J.,
Francis, C.M., \emph{et al.} (2021). Key information needs to move from
knowledge to action for biodiversity conservation in Canada.
\emph{Biological Conservation}, 256, 108983.

\leavevmode\hypertarget{ref-Cai2017ComSur}{}%
Cai, H., Zheng, V.W. \& Chang, K.C.-C. (2017). \DIFdelbegin \DIFdel{A Comprehensive Survey of
Graph Embedding: Problems, Techniques and Applications. \emph{arXiv
preprint arXiv:1709.07604}}\DIFdelend \DIFaddbegin \DIFadd{\emph{A Comprehensive
Survey of Graph Embedding: Problems, Techniques and Applications}.
Available at: }\url{http://arxiv.org/abs/1709.07604}\DIFadd{. Last accessed}\DIFaddend .

\leavevmode\hypertarget{ref-Cameron2019UneGlo}{}%
Cameron, E.K., Sundqvist, M.K., Keith, S.A., CaraDonna, P.J., Mousing,
E.A., Nilsson, K.A., \emph{et al.} (2019). Uneven global distribution of
food web studies under climate change. \emph{Ecosphere}, 10, e02645.

\leavevmode\hypertarget{ref-Cao2019NetEmb}{}%
Cao, R.-M., Liu, S.-Y. \& Xu, X.-K. (2019). Network embedding for link
prediction: The pitfall and improvement. \emph{Chaos: An
Interdisciplinary Journal of Nonlinear Science}, 29, 103102.

\leavevmode\hypertarget{ref-Cavender-Bares2009MerCom}{}%
Cavender-Bares, J., Kozak, K.H., Fine, P.V.A. \& Kembel, S.W. (2009).
The merging of community ecology and phylogenetic biology. \emph{Ecology
Letters}, 12, 693--715.

\leavevmode\hypertarget{ref-Choudry2013SavBio}{}%
Choudry, A. (2013). Saving biodiversity, for whom and for what?
Conservation NGOs, complicity, colonialism and conquest in an era of
capitalist globalization. In: \emph{NGOization: Complicity,
contradictions and prospects}. Bloomsbury Publishing, pp. 24--44.

\leavevmode\hypertarget{ref-Cirtwill2019QuaFra}{}%
Cirtwill, A.R., \DIFdelbegin \DIFdel{Eklf}\DIFdelend \DIFaddbegin \DIFadd{Ekl}\DIFaddend , A., Roslin, T., Wootton, K. \& Gravel, D. (2019). A
quantitative framework for investigating the reliability of empirical
network construction. \emph{Methods in Ecology and Evolution}, 0.

\leavevmode\hypertarget{ref-Cirtwill2021BuiFoo}{}%
Cirtwill, A.R. \& Hambäck, P. (2021). Building food networks from
molecular data: Bayesian or fixed-number thresholds for including links.
\emph{Basic and Applied Ecology}, 50, 67--76.

\leavevmode\DIFaddbegin \hypertarget{ref-Csermely2004StrLin}{}%DIF > 
\DIFadd{Csermely, P. (2004). Strong links are important, but weak links
stabilize them. \emph{Trends in Biochemical Sciences}, 29, 331--334.
}

\leavevmode\DIFaddend \hypertarget{ref-DallaRiva2016ExpEvo}{}%
Dalla Riva, G.V. \& Stouffer, D.B. (2016). Exploring the evolutionary
signature of food webs' backbones using functional traits. \emph{Oikos},
125, 446--456.

\leavevmode\DIFaddbegin \hypertarget{ref-Dallas2017PreCry}{}%DIF > 
\DIFadd{Dallas, T., Park, A.W. \& Drake, J.M. (2017). Predicting cryptic links
in host-parasite networks. \emph{PLOS Computational Biology}, 13,
e1005557.
}

\leavevmode\DIFaddend \hypertarget{ref-Dansereau2021SimJl}{}%
Dansereau, G. \& Poisot, T. (2021). SimpleSDMLayers.jl and GBIF.jl: A
Framework for Species Distribution Modeling in Julia. \emph{Journal of
Open Source Software}, 6, 2872.

\leavevmode\hypertarget{ref-Dominguez2020DecCon}{}%
Domínguez, L. \& Luoma, C. (2020). Decolonising Conservation Policy: How
Colonial Land and Conservation Ideologies Persist and Perpetuate
Indigenous Injustices at the Expense of the Environment. \emph{Land}, 9,
65.

\leavevmode\hypertarget{ref-Dormann2010EvoCli}{}%
Dormann, C.F., Gruber, B., Winter, M. \& Herrmann, D. (2010). Evolution
of climate niches in European mammals? \emph{Biology Letters}, 6,
229--232.

\leavevmode\hypertarget{ref-Dunne2006NetStr}{}%
Dunne, J.A. (2006). The Network Structure of Food Webs. In:
\emph{Ecological networks: Linking structure and dynamics} (eds. Dunne,
J.A. \& Pascual, M.). Oxford University Press, pp. 27--86.

\leavevmode\hypertarget{ref-Eichhorn2019SteDec}{}%
Eichhorn, M.P., Baker, K. \& Griffiths, M. (2019). Steps towards
decolonising biogeography. \emph{Frontiers of Biogeography}, 12, 1--7.

\leavevmode\hypertarget{ref-Eklof2016PhyCom}{}%
Eklöf, A. \& Stouffer, D.B. (2016). The phylogenetic component of food
web structure and intervality. \emph{Theoretical Ecology}, 9, 107--115.

\leavevmode\DIFaddbegin \hypertarget{ref-Fortin2021NetEco}{}%DIF > 
\DIFadd{Fortin, M.-J., Dale, M.R.T. \& Brimacombe, C. (2021). Network ecology in
dynamic landscapes. \emph{Proceedings of the Royal Society B: Biological
Sciences}, 288, rspb.2020.1889, 20201889.
}

\leavevmode\hypertarget{ref-Galiana2021SpaSca}{}%DIF > 
\DIFadd{Galiana, N., Barros, C., Braga, J., Ficetola, G.F., Maiorano, L.,
Thuiller, W., \emph{et al.} (2021). The spatial scaling of food web
structure across European biogeographical regions. \emph{Ecography},
n/a.
}

\leavevmode\hypertarget{ref-Galiana2019GeoVar}{}%DIF > 
\DIFadd{Galiana, N., Hawkins, B.A. \& Montoya, J.M. (2019). The geographical
variation of network structure is scale dependent: Understanding the
biotic specialization of hostparasitoid networks. \emph{Ecography}, 42,
1175--1187.
}

\leavevmode\hypertarget{ref-Galiana2018SpaSca}{}%DIF > 
\DIFadd{Galiana, N., Lurgi, M., Claramunt-López, B., Fortin, M.-J., Leroux, S.,
Cazelles, K., \emph{et al.} (2018). The spatial scaling of species
interaction networks. \emph{Nature Ecology \& Evolution}, 2, 782--790.
}

\leavevmode\DIFaddend \hypertarget{ref-Garland1999IntPhy}{}%
Garland, T., JR., Midford, P.E. \& Ives, A.R. (1999). An Introduction to
Phylogenetically Based Statistical Methods, with a New Method for
Confidence Intervals on Ancestral Values1. \emph{American Zoologist},
39, 374--388.

\leavevmode\hypertarget{ref-Garlaschelli2018CovStr}{}%
Garlaschelli, D., Hollander, F. den \& Roccaverde, A. (2018). Covariance
structure behind breaking of ensemble equivalence in random graphs.
\emph{Journal of Statistical Physics}, 173, 644--662.

\leavevmode\hypertarget{ref-GBIFSecretariat2021GbiBac}{}%
GBIF Secretariat. (2021). GBIF Backbone Taxonomy.

\leavevmode\hypertarget{ref-Gerhold2015PhyPat}{}%
Gerhold, P., Cahill, J.F., Winter, M., Bartish, I.V. \& Prinzing, A.
(2015). Phylogenetic patterns are not proxies of community assembly
mechanisms (they are far better). \emph{Functional Ecology}, 29,
600--614.

\leavevmode\DIFaddbegin \hypertarget{ref-Gravel2018BriElt}{}%DIF > 
\DIFadd{Gravel, D., Baiser, B., Dunne, J.A., Kopelke, J.-P., Martinez, N.D.,
Nyman, T., \emph{et al.} (2018). Bringing Elton and Grinnell together: A
quantitative framework to represent the biogeography of ecological
interaction networks. \emph{Ecography}, 0.
}

\leavevmode\DIFaddend \hypertarget{ref-Grenie2021HarTax}{}%
Grenié, M., Berti, E., Carvajal-Quintero, J.D., Winter, M. \& Sagouis,
A. (2021). Harmonizing taxon names in biodiversity data: A review of
tools, databases, and best practices.

\leavevmode\DIFaddbegin \hypertarget{ref-Grunig2020CroFor}{}%DIF > 
\DIFadd{Grünig, M., Mazzi, D., Calanca, P., Karger, D.N. \& Pellissier, L.
(2020). Crop and forest pest metawebs shift towards increased linkage
and suitability overlap under climate change. \emph{Communications
Biology}, 3, 1--10.
}

\leavevmode\hypertarget{ref-Halevy2009UnrEff}{}%DIF > 
\DIFadd{Halevy, A., Norvig, P. \& Pereira, F. (2009). The Unreasonable
Effectiveness of Data. \emph{IEEE Intelligent Systems}, 24, 8--12.
}

\leavevmode\DIFaddend \hypertarget{ref-Halko2011FinStr}{}%
Halko, N., Martinsson, P.G. \& Tropp, J.A. (2011). Finding Structure
with Randomness: Probabilistic Algorithms for Constructing Approximate
Matrix Decompositions. \emph{SIAM Review}, 53, 217--288.

\leavevmode\hypertarget{ref-Herbert1965Dun}{}%
Herbert, F. (1965). \emph{Dune}. \DIFdelbegin \DIFdel{First}\DIFdelend \DIFaddbegin \DIFadd{1st edn}\DIFaddend . Chilton Book Company,
Philadelphia.

\leavevmode\DIFaddbegin \hypertarget{ref-Hoffmann2019MacLea}{}%DIF > 
\DIFadd{Hoffmann, J., Bar-Sinai, Y., Lee, L.M., Andrejevic, J., Mishra, S.,
Rubinstein, S.M., \emph{et al.} (2019). Machine learning in a
data-limited regime: Augmenting experiments with synthetic data uncovers
order in crumpled sheets. \emph{Science Advances}, 5, eaau6792.
}

\leavevmode\hypertarget{ref-Holm2019DefBla}{}%DIF > 
\DIFadd{Holm, E.A. (2019). In defense of the black box. \emph{Science}, 364,
26--27.
}

\leavevmode\DIFaddend \hypertarget{ref-Hortal2015SevSho}{}%
Hortal, J., de Bello, F., Diniz-Filho, J.A.F., Lewinsohn, T.M., Lobo,
J.M. \& Ladle, R.J. (2015). Seven Shortfalls that Beset Large-Scale
Knowledge of Biodiversity. \emph{Annual Review of Ecology, Evolution,
and Systematics}, 46, 523--549.

\leavevmode\hypertarget{ref-Hutchinson2017CopSig}{}%
Hutchinson, M.C., Cagua, E.F. \& Stouffer, D.B. (2017). Cophylogenetic
signal is detectable in pollination interactions across ecological
scales. \emph{Ecology}, n/a--n/a.

\leavevmode\hypertarget{ref-Jordano2016ChaEco}{}%
Jordano, P. (2016a). Chasing Ecological Interactions. \emph{PLOS Biol},
14, e1002559.

\leavevmode\hypertarget{ref-Jordano2016SamNet}{}%
Jordano, P. (2016b). Sampling networks of ecological interactions.
\emph{Functional Ecology}, 30, 1883--1893.

\leavevmode\DIFaddbegin \hypertarget{ref-Kawatsu2021AreNet}{}%DIF > 
\DIFadd{Kawatsu, K., Ushio, M., van Veen, F.J.F. \& Kondoh, M. (2021). Are
networks of trophic interactions sufficient for understanding the
dynamics of multi-trophic communities? Analysis of a tri-trophic insect
food-web time-series. \emph{Ecology Letters}, 24, 543--552.
}

\leavevmode\hypertarget{ref-Kefi2012MorMea}{}%DIF > 
\DIFadd{Kéfi, S., Berlow, E.L., Wieters, E.A., Navarrete, S.A., Petchey, O.L.,
Wood, S.A., \emph{et al.} (2012). More than a meal\ldots{} integrating
non-feeding interactions into food webs: More than a meal \ldots.
\emph{Ecology Letters}, 15, 291--300.
}

\leavevmode\hypertarget{ref-Lamba2019DeeLea}{}%DIF > 
\DIFadd{Lamba, A., Cassey, P., Segaran, R.R. \& Koh, L.P. (2019). Deep learning
for environmental conservation. \emph{Current Biology}, 29, R977--R982.
}

\leavevmode\DIFaddend \hypertarget{ref-Litsios2012EffPhy}{}%
Litsios, G. \& Salamin, N. (2012). Effects of Phylogenetic Signal on
Ancestral State Reconstruction. \emph{Systematic Biology}, 61, 533--538.

\leavevmode\DIFaddbegin \hypertarget{ref-Machen2021ThiAlg}{}%DIF > 
\DIFadd{Machen, R. \& Nost, E. (2021). Thinking algorithmically: The making of
hegemonic knowledge in climate governance. \emph{Transactions of the
Institute of British Geographers}, 46, 555--569.
}

\leavevmode\DIFaddend \hypertarget{ref-Maiorano2020DatTet}{}%
Maiorano, L., Montemaggiori, A., Ficetola, G.F., O'Connor, L. \&
Thuiller, W. (2020a). Data from: Tetra-EU 1.0: A species-level trophic
meta-web of European tetrapods.

\leavevmode\DIFdelbegin %DIFDELCMD < \hypertarget{ref-Maiorano2020TetEu}{}%%%
\DIFdelend \DIFaddbegin \hypertarget{ref-Maiorano2020Tet10}{}\DIFaddend %
Maiorano, L., Montemaggiori, A., Ficetola, G.F., O'Connor, L. \&
Thuiller, W. (2020b). TETRA-EU 1.0: A species-level trophic metaweb of
European tetrapods. \emph{Global Ecology and Biogeography}, 29,
1452--1457.

\leavevmode\hypertarget{ref-Marco2018ChaHum}{}%
Marco, M.D., Venter, O., Possingham, H.P. \& Watson, J.E.M. (2018).
Changes in human footprint drive changes in species extinction risk.
\emph{Nature Communications}, 9, 4621.

\leavevmode\DIFaddbegin \hypertarget{ref-McLeod2021SamAsy}{}%DIF > 
\DIFadd{McLeod, A., Leroux, S.J., Gravel, D., Chu, C., Cirtwill, A.R., Fortin,
M.-J., \emph{et al.} (2021). Sampling and asymptotic network properties
of spatial multi-trophic networks. \emph{Oikos}, n/a.
}

\leavevmode\hypertarget{ref-Mora2018IdeCom}{}%DIF > 
\DIFadd{Mora, B.B., Gravel, D., Gilarranz, L.J., Poisot, T. \& Stouffer, D.B.
(2018). Identifying a common backbone of interactions underlying food
webs from different ecosystems. \emph{Nature Communications}, 9, 2603.
}

\leavevmode\DIFaddend \hypertarget{ref-Morales-Castilla2015InfBio}{}%
Morales-Castilla, I., Matias, M.G., Gravel, D. \& Araújo, M.B. (2015).
Inferring biotic interactions from proxies. \emph{Trends in Ecology \&
Evolution}, 30, 347--356.

\leavevmode\DIFaddbegin \hypertarget{ref-MoseboFernandes2020MacLea}{}%DIF > 
\DIFadd{Mosebo Fernandes, A.C., Quintero Gonzalez, R., Lenihan-Clarke, M.A.,
Leslie Trotter, E.F. \& Jokar Arsanjani, J. (2020). Machine Learning for
Conservation Planning in a Changing Climate. \emph{Sustainability}, 12,
7657.
}

\leavevmode\DIFaddend \hypertarget{ref-Mouquet2012EcoAdv}{}%
Mouquet, N., Devictor, V., Meynard, C.N., Munoz, F., Bersier, L.-F.,
Chave, J., \emph{et al.} (2012). Ecophylogenetics: Advances and
perspectives. \emph{Biological Reviews}, 87, 769--785.

\leavevmode\hypertarget{ref-Nenzen2014Imp850}{}%
Nenzén, H.K., Montoya, D. \& Varela, S. (2014). The Impact of 850,000
Years of Climate Changes on the Structure and Dynamics of Mammal Food
Webs. \emph{PLOS ONE}, 9, e106651.

\leavevmode\DIFaddbegin \hypertarget{ref-Neutel2002StaRea}{}%DIF > 
\DIFadd{Neutel, A.-M., Heesterbeek, J.A.P. \& de Ruiter, P.C. (2002). Stability
in Real Food Webs: Weak Links in Long Loops. \emph{Science}, 296,
1120--1123.
}

\leavevmode\DIFaddend \hypertarget{ref-Nokmaq2021AwaSle}{}%
No'kmaq, M., Marshall, A., Beazley, K.F., Hum, J., joudry, shalan,
Papadopoulos, A., \emph{et al.} (2021). {``Awakening the sleeping
giant''}: Re-Indigenization principles for transforming biodiversity
conservation in Canada and beyond. \emph{FACETS}, 6, 839--869.

\leavevmode\DIFaddbegin \hypertarget{ref-Nost2021PolEco}{}%DIF > 
\DIFadd{Nost, E. \& Goldstein, J.E. (2021). A political ecology of data.
\emph{Environment and Planning E: Nature and Space}, 25148486211043503.
}

\leavevmode\DIFaddend \hypertarget{ref-OConnor2020UnvFoo}{}%
O'Connor, L.M.J., Pollock, L.J., Braga, J., Ficetola, G.F., Maiorano,
L., Martinez-Almoyna, C., \emph{et al.} (2020). Unveiling the food webs
of tetrapods across Europe through the prism of the Eltonian niche.
\emph{Journal of Biogeography}, 47, 181--192.

\leavevmode\hypertarget{ref-Pan2010SurTra}{}%
Pan, S.J. \& Yang, Q. (2010). A Survey on Transfer Learning. \emph{IEEE
Transactions on Knowledge and Data Engineering}, 22, 1345--1359.

\leavevmode\hypertarget{ref-Park2004StaMec}{}%
Park, J. \& Newman, M.E.J. (2004). Statistical mechanics of networks.
\emph{Physical Review E}, 70, 066117.

\leavevmode\hypertarget{ref-Pedersen2017SigCol}{}%
Pedersen, E.J., Thompson, P.L., Ball, R.A., Fortin, M.-J., Gouhier,
T.C., Link, H., \emph{et al.} (2017). Signatures of the collapse and
incipient recovery of an overexploited marine ecosystem. \emph{Royal
Society Open Science}, 4, 170215.

\leavevmode\DIFaddbegin \hypertarget{ref-Perretti2013ModFor}{}%DIF > 
\DIFadd{Perretti, C.T., Munch, S.B. \& Sugihara, G. (2013). Model-free
forecasting outperforms the correct mechanistic model for simulated and
experimental data. \emph{Proceedings of the National Academy of
Sciences}, 110, 5253--5257.
}

\leavevmode\DIFaddend \hypertarget{ref-Pires2015PleMeg}{}%
Pires, M.M., Koch, P.L., Fariña, R.A., de Aguiar, M.A.M., dos Reis, S.F.
\& Guimarães, P.R. (2015). Pleistocene megafaunal interaction networks
became more vulnerable after human arrival. \emph{Proceedings of the
Royal Society B: Biological Sciences}, 282, 20151367.

\leavevmode\hypertarget{ref-Poelen2014GloBio}{}%
Poelen, J.H., Simons, J.D. \& Mungall, C.J. (2014). Global biotic
interactions: An open infrastructure to share and analyze
species-interaction datasets. \emph{Ecological Informatics}, 24,
148--159.

\leavevmode\hypertarget{ref-Poisot2019EcoJl}{}%
Poisot, T., Belisle, Z., Hoebeke, L., Stock, M. \& Szefer, P. (2019).
EcologicalNetworks.jl - analysing ecological networks. \emph{Ecography}.

\leavevmode\hypertarget{ref-Poisot2021GloKno}{}%
Poisot, T., Bergeron, G., Cazelles, K., Dallas, T., Gravel, D.,
MacDonald, A., \emph{et al.} (2021a). Global knowledge gaps in species
interaction networks data. \emph{Journal of Biogeography}, n/a.

\leavevmode\DIFdelbegin %DIFDELCMD < \hypertarget{ref-Poisot2012DisSpe}{}%%%
%DIF < 
\DIFdel{Poisot, T., Canard, E., Mouillot, D., Mouquet, N. \& Gravel, D. (2012).
The dissimilarity of species interaction networks. \emph{Ecology
Letters}, 15, 1353--1361.
}%DIFDELCMD < 

%DIFDELCMD < \leavevmode%%%
\DIFdelend \hypertarget{ref-Poisot2016StrPro}{}%
Poisot, T., Cirtwill, A.R., Cazelles, K., Gravel, D., Fortin, M.-J. \&
Stouffer, D.B. (2016). The structure of probabilistic networks.
\emph{Methods in Ecology and Evolution}, 7, 303--312.

\leavevmode\hypertarget{ref-Poisot2021ImpMam}{}%
Poisot, T., Ouellet, M.-A., Mollentze, N., Farrell, M.J., Becker, D.J.,
Albery, G.F., \emph{et al.} (2021b). \DIFdelbegin \DIFdel{Imputing the mammalian virome with
linear filtering and singular value decomposition. \emph{arXiv:2105.14973 {[}q-bio{]}}}\DIFdelend \DIFaddbegin \DIFadd{\emph{Imputing the mammalian virome
with linear filtering and singular value decomposition}. Available at:
}\url{http://arxiv.org/abs/2105.14973}\DIFadd{. Last accessed}\DIFaddend .

\leavevmode\hypertarget{ref-Poisot2018IntRet}{}%
Poisot, T. \& Stouffer, D.B. (2018). Interactions retain the
co-phylogenetic matching that communities lost. \emph{Oikos}, 127,
230--238.

\leavevmode\DIFaddbegin \hypertarget{ref-Poisot2015SpeWhy}{}%DIF > 
\DIFadd{Poisot, T., Stouffer, D.B. \& Gravel, D. (2015). Beyond species: Why
ecological interaction networks vary through space and time.
\emph{Oikos}, 124, 243--251.
}

\leavevmode\DIFaddend \hypertarget{ref-Price2003MacThe}{}%
Price, P.W. (2003). \emph{Macroevolutionary theory on macroecological
patterns}. Cambridge University Press.

\leavevmode\hypertarget{ref-Ray2021BioCri}{}%
Ray, J.C., Grimm, J. \& Olive, A. (2021). The biodiversity crisis in
Canada: Failures and challenges of federal and sub-national strategic
and legal frameworks. \emph{FACETS}, 6, 1044--1068.

\leavevmode\hypertarget{ref-Reeve2016HowPar}{}%
Reeve, R., Leinster, T., Cobbold, C.A., Thompson, J., Brummitt, N.,
Mitchell, S.N., \emph{et al.} (2016). \DIFdelbegin \DIFdel{How to partition diversity.
\emph{arXiv:1404.6520 {[}q-bio{]}}}\DIFdelend \DIFaddbegin \DIFadd{\emph{How to partition diversity}.
Available at: }\url{http://arxiv.org/abs/1404.6520}\DIFadd{. Last accessed}\DIFaddend .

\leavevmode\hypertarget{ref-Rosado2013GoiBac}{}%
Rosado, B.H.P., Dias, A. \& de Mattos, E. (2013). Going Back to Basics:
Importance of Ecophysiology when Choosing Functional Traits for Studying
Communities and Ecosystems. \emph{Natureza \&
conservaç\textasciitilde ao revista brasileira de
conservaç\textasciitilde ao da natureza}, 11, 15--22.

\leavevmode\hypertarget{ref-Runghen2021ExpNod}{}%
Runghen, R., Stouffer, D.B. \& Dalla Riva, G.V. (2021). Exploiting node
metadata to predict interactions in large networks using graph embedding
and neural networks.

\leavevmode\hypertarget{ref-Schoener1989FooWeb}{}%
Schoener, T.W. (1989). Food webs from the small to the large.
\emph{Ecology}, 70, 1559--1589.

\leavevmode\DIFaddbegin \hypertarget{ref-Shlens2014TutPri}{}%DIF > 
\DIFadd{Shlens, J. (2014). \emph{A Tutorial on Principal Component Analysis}.
Available at: }\url{http://arxiv.org/abs/1404.1100}\DIFadd{. Last accessed.
}

\leavevmode\DIFaddend \hypertarget{ref-Solis-Lemus2017PhyPac}{}%
Solís-Lemus, C., Bastide, P. \& Ané, C. (2017). PhyloNetworks: A Package
for Phylogenetic Networks. \emph{Molecular Biology and Evolution}, 34,
3292--3298.

\leavevmode\hypertarget{ref-Stock2021PaiLea}{}%
Stock, M. (2021). Pairwise learning for predicting pollination
interactions based on traits and phylogeny. \emph{Ecological Modelling},
14.

\leavevmode\hypertarget{ref-Stouffer2012EvoCon}{}%
Stouffer, D.B., Sales-Pardo, M., Sirer, M.I. \& Bascompte, J. (2012).
Evolutionary Conservation of Species' Roles in Food Webs.
\emph{Science}, 335, 1489--1492.

\leavevmode\hypertarget{ref-Strong2014ImpNon}{}%
Strong, J.S. \& Leroux, S.J. (2014). Impact of Non-Native Terrestrial
Mammals on the Structure of the Terrestrial Mammal Food Web of
Newfoundland, Canada. \emph{PLOS ONE}, 9, e106264.

\leavevmode\hypertarget{ref-Strydom2021RoaPre}{}%
Strydom, T., Catchen, M.D., Banville, F., Caron, D., Dansereau, G.,
Desjardins-Proulx, P., \emph{et al.} (2021a). \DIFdelbegin \DIFdel{\emph{A Roadmap Toward
Predicting Species Interaction Networks (Across Space and Time)}
(Preprint).
EcoEvoRxiv.
}\DIFdelend \DIFaddbegin \DIFadd{A roadmap towards
predicting species interaction networks (across space and time).
\emph{Philosophical Transactions of the Royal Society B: Biological
Sciences}, 376, 20210063.
}\DIFaddend 

\leavevmode\hypertarget{ref-Strydom2021SvdEnt}{}%
Strydom, T., Dalla Riva, G.V. \& Poisot, T. (2021b). SVD Entropy Reveals
the High Complexity of Ecological Networks. \emph{Frontiers in Ecology
and Evolution}, 9.

\leavevmode\hypertarget{ref-Thessen2014KnoExt}{}%
Thessen, A.E. \& Parr, C.S. (2014). Knowledge extraction and semantic
annotation of text from the encyclopedia of life. \emph{PloS one}, 9,
e89550.

\leavevmode\hypertarget{ref-Torrey2010TraLea}{}%
Torrey, L. \& Shavlik, J. (2010). Transfer learning. In: \emph{Handbook
of research on machine learning applications and trends: Algorithms,
methods, and techniques}. IGI global, pp. 242--264.

\leavevmode\hypertarget{ref-Trojelsgaard2016EcoNet}{}%
Trøjelsgaard, K. \& Olesen, J.M. (2016). Ecological networks in motion:
Micro- and macroscopic variability across scales. \emph{Functional
Ecology}, 30, 1926--1935.

\leavevmode\hypertarget{ref-Turak2017UsiEss}{}%
Turak, E., Brazill-Boast, J., Cooney, T., Drielsma, M., DelaCruz, J.,
Dunkerley, G., \emph{et al.} (2017). Using the essential biodiversity
variables framework to measure biodiversity change at national scale.
\emph{Biological Conservation}, SI:measures of biodiversity, 213,
264--271.

\leavevmode\hypertarget{ref-Upham2019InfMam}{}%
Upham, N.S., Esselstyn, J.A. \& Jetz, W. (2019). Inferring the mammal
tree: Species-level sets of phylogenies for questions in ecology,
evolution, and conservation. \emph{PLOS Biology}, 17, e3000494.

\leavevmode\hypertarget{ref-vanderHoorn2018SpaMax}{}%
van der Hoorn, P., Lippner, G. \& Krioukov, D. (2018). Sparse
Maximum-Entropy Random Graphs with a Given Power-Law Degree
Distribution. \emph{Journal of Statistical Physics}, 173, 806--844\DIFaddbegin \DIFadd{.
}

\leavevmode\hypertarget{ref-Vermaat2009MajDim}{}%DIF > 
\DIFadd{Vermaat, J.E., Dunne, J.A. \& Gilbert, A.J. (2009). Major dimensions in
food-web structure properties. \emph{Ecology}, 90, 278--282.
}

\leavevmode\hypertarget{ref-Wood2015EffSpa}{}%DIF > 
\DIFadd{Wood, S.A., Russell, R., Hanson, D., Williams, R.J. \& Dunne, J.A.
(2015). Effects of spatial scale of sampling on food web structure.
\emph{Ecology and Evolution}, 5, 3769--3782}\DIFaddend .

\leavevmode\hypertarget{ref-Xing2021RisEco}{}%
Xing, S. \& Fayle, T.M. (2021). The rise of ecological network
meta-analyses: Problems and prospects. \emph{Global Ecology and
Conservation}, 30, e01805.

\leavevmode\hypertarget{ref-Yeakel2014ColEco}{}%
Yeakel, J.D., Pires, M.M., Rudolf, L., Dominy, N.J., Koch, P.L.,
Guimarães, P.R., \emph{et al.} (2014). Collapse of an ecological network
in Ancient Egypt. \emph{PNAS}, 111, 14472--14477.

\leavevmode\hypertarget{ref-Youden1950IndRat}{}%
Youden, W.J. (1950). Index for rating diagnostic tests. \emph{Cancer},
3, 32--35.

\leavevmode\hypertarget{ref-Young2021BayInf}{}%
Young, J.-G., Cantwell, G.T. \& Newman, M.E.J. (2021). Bayesian
inference of network structure from unreliable data. \emph{Journal of
Complex Networks}, 8.

\leavevmode\hypertarget{ref-Young2007RanDot}{}%
Young, S.J. \& Scheinerman, E.R. (2007). Random Dot Product Graph Models
for Social Networks. In: \emph{Algorithms and Models for the Web-Graph},
Lecture Notes in Computer Science (eds. Bonato, A. \& Chung, F.R.K.).
Springer, Berlin, Heidelberg, pp. 138--149.

\leavevmode\hypertarget{ref-Zhu2006AutDim}{}%
Zhu, M. \& Ghodsi, A. (2006). Automatic dimensionality selection from
the scree plot via the use of profile likelihood. \emph{Computational
Statistics \& Data Analysis}, 51, 918--930.

\end{CSLReferences}

\end{document}
